--- dc400b26-src/source/AppHdr.h.bak	2005-08-27 14:58:35.368013112 +0100
+++ dc400b26-src/source/AppHdr.h	2005-08-27 14:58:16.124938504 +0100
@@ -57,7 +57,7 @@
     #define PLAIN_TERM
     // #define CHARACTER_SET           A_ALTCHARSET
     #define CHARACTER_SET           0
-    #define USE_ASCII_CHARACTERS
+    // #define USE_ASCII_CHARACTERS
     #define MULTIUSER
 
     // Set curses include file if you don't want the default curses.h
