--- source/makefile.lnx.bak	2005-02-18 22:07:35.000000000 +0000
+++ source/makefile.lnx	2005-02-19 00:57:27.675676064 +0000
@@ -15,19 +15,13 @@
 COPY = cp
 OS_TYPE = LINUX
 
-CFLAGS = -Wall -Wwrite-strings -Wstrict-prototypes \
-            -Wmissing-prototypes -Wmissing-declarations \
-            -g -D$(OS_TYPE) $(EXTRA_FLAGS)
-
 # LDFLAGS = -static
 MCHMOD = 2755
 # INSTALLDIR = /usr/games
+# Gentoo ebuild note: The INSTALLDIR is never actually used here because we install files ourselves.
 INSTALLDIR = /opt/crawl/bin
 LIB = -lncurses
 
-# Include for Linux
-INCLUDES = -I/usr/include/ncurses
-
 all: $(GAME)
 
 install: $(GAME)
