--- source/religion.cc.bak	2005-02-18 22:02:56.283569632 +0000
+++ source/religion.cc	2005-02-18 22:03:39.665974504 +0000
@@ -53,18 +53,18 @@
 #include "stuff.h"
 
 const char *sacrifice[] = {
-    {" glows silver and disappears."},
-    {" glows a brilliant golden colour and disappears."},
-    {" rots away in an instant."},
-    {" crumbles to dust."},
-    {" is eaten by a bug."},    /* Xom - no sacrifices */
-    {" explodes into nothingness."},
-    {" is consumed in a burst of flame."},
-    {" is consumed in a roaring column of flame."},
-    {" glows faintly for a moment, then is gone."},
-    {" is consumed in a roaring column of flame."},
-    {" glows with a rainbow of weird colours and disappears."},
-    {" evaporates."}
+    " glows silver and disappears.",
+    " glows a brilliant golden colour and disappears.",
+    " rots away in an instant.",
+    " crumbles to dust.",
+    " is eaten by a bug.",    /* Xom - no sacrifices */
+    " explodes into nothingness.",
+    " is consumed in a burst of flame.",
+    " is consumed in a roaring column of flame.",
+    " glows faintly for a moment, then is gone.",
+    " is consumed in a roaring column of flame.",
+    " glows with a rainbow of weird colours and disappears.",
+    " evaporates."
 };
 
 void altar_prayer(void);
