--- /root/orig-crawl-source/menu.cc	2005-07-20 17:21:45.730562704 +0100
+++ menu.cc	2005-07-20 17:33:29.204618392 +0100
@@ -80,8 +80,9 @@
             nav = true;
             repaint = page_down();
             if (!repaint && flags && !is_set(MF_EASY_EXIT)) {
-                repaint = first_entry != 0;
-                first_entry = 0;
+                nav = false;
+                lastch = keyin;
+                return false;
             }
             break;
         case CK_PGUP: case '<': case ';':
