diff -Nur a/ggml/src/ggml-blas/CMakeLists.txt b/ggml/src/ggml-blas/CMakeLists.txt
--- a/ggml/src/ggml-blas/CMakeLists.txt	2024-12-23 19:05:27.000000000 -0800
+++ b/ggml/src/ggml-blas/CMakeLists.txt	2024-12-23 23:07:42.279475783 -0800
@@ -54,6 +54,7 @@
         endif()
         if (DepBLAS_FOUND)
             set(BLAS_INCLUDE_DIRS ${DepBLAS_INCLUDE_DIRS})
+            set(BLAS_LIBRARIES ${DepBLAS_LINK_LIBRARIES})
         else()
             message(WARNING "BLAS_INCLUDE_DIRS neither been provided nor been automatically"
             " detected by pkgconfig, trying to find cblas.h from possible paths...")
