diff -Nur a/cmake/common.cmake b/cmake/common.cmake
--- a/cmake/common.cmake	2025-01-27 10:36:10.000000000 -0800
+++ b/cmake/common.cmake	2025-02-02 17:33:51.405319315 -0800
@@ -20,7 +20,7 @@
             list(APPEND C_FLAGS   ${WARNING_FLAGS})
             list(APPEND CXX_FLAGS ${WARNING_FLAGS})
 
-            ggml_get_flags(${CMAKE_CXX_COMPILER_ID} ${CMAKE_CXX_COMPILER_VERSION})
+            #ggml_get_flags(${CMAKE_CXX_COMPILER_ID} ${CMAKE_CXX_COMPILER_VERSION})
 
             add_compile_options("$<$<COMPILE_LANGUAGE:C>:${C_FLAGS};${GF_C_FLAGS}>"
                                 "$<$<COMPILE_LANGUAGE:CXX>:${CXX_FLAGS};${GF_CXX_FLAGS}>")
diff -Nur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2025-01-27 10:36:10.000000000 -0800
+++ b/CMakeLists.txt	2025-02-02 17:30:35.155312361 -0800
@@ -143,8 +143,8 @@
 #
 # 3rd-party
 #
-
-if (NOT TARGET ggml)
+find_package(ggml REQUIRED)
+if (NOT TARGET ggml::ggml)
     add_subdirectory(ggml)
     # ... otherwise assume ggml is added by a parent CMakeLists.txt
 endif()
diff -Nur a/examples/gguf/CMakeLists.txt b/examples/gguf/CMakeLists.txt
--- a/examples/gguf/CMakeLists.txt	2025-01-27 10:36:10.000000000 -0800
+++ b/examples/gguf/CMakeLists.txt	2025-02-02 17:37:41.195327404 -0800
@@ -1,5 +1,5 @@
 set(TARGET llama-gguf)
 add_executable(${TARGET} gguf.cpp)
 install(TARGETS ${TARGET} RUNTIME)
-target_link_libraries(${TARGET} PRIVATE ggml ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(${TARGET} PRIVATE ggml::all ${CMAKE_THREAD_LIBS_INIT})
 target_compile_features(${TARGET} PRIVATE cxx_std_17)
diff -Nur a/examples/gguf-hash/CMakeLists.txt b/examples/gguf-hash/CMakeLists.txt
--- a/examples/gguf-hash/CMakeLists.txt	2025-01-27 10:36:10.000000000 -0800
+++ b/examples/gguf-hash/CMakeLists.txt	2025-02-02 17:36:28.578658202 -0800
@@ -18,5 +18,5 @@
 add_library(sha256 OBJECT deps/sha256/sha256.c deps/sha256/sha256.h)
 target_link_libraries(${TARGET} PRIVATE sha256)
 
-target_link_libraries(${TARGET} PRIVATE ggml ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(${TARGET} PRIVATE ggml::all ${CMAKE_THREAD_LIBS_INIT})
 target_compile_features(${TARGET} PRIVATE cxx_std_17)
diff -Nur a/examples/quantize-stats/CMakeLists.txt b/examples/quantize-stats/CMakeLists.txt
--- a/examples/quantize-stats/CMakeLists.txt	2025-01-27 10:36:10.000000000 -0800
+++ b/examples/quantize-stats/CMakeLists.txt	2025-02-02 17:41:04.255334567 -0800
@@ -1,6 +1,6 @@
 set(TARGET llama-quantize-stats)
 add_executable(${TARGET} quantize-stats.cpp)
 install(TARGETS ${TARGET} RUNTIME)
-target_link_libraries(${TARGET} PRIVATE llama build_info ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(${TARGET} PRIVATE llama build_info ggml::all ${CMAKE_THREAD_LIBS_INIT})
 target_include_directories(${TARGET} PRIVATE ../../common)
 target_compile_features(${TARGET} PRIVATE cxx_std_17)
diff -Nur a/examples/rpc/CMakeLists.txt b/examples/rpc/CMakeLists.txt
--- a/examples/rpc/CMakeLists.txt	2025-01-27 10:36:10.000000000 -0800
+++ b/examples/rpc/CMakeLists.txt	2025-02-02 17:39:19.261997514 -0800
@@ -1,2 +1,2 @@
 add_executable(rpc-server rpc-server.cpp)
-target_link_libraries(rpc-server PRIVATE ggml llama)
+target_link_libraries(rpc-server PRIVATE ggml::all llama)
diff -Nur a/src/CMakeLists.txt b/src/CMakeLists.txt
--- a/src/CMakeLists.txt	2025-01-27 10:36:10.000000000 -0800
+++ b/src/CMakeLists.txt	2025-02-02 17:43:23.862006169 -0800
@@ -32,7 +32,7 @@
 target_include_directories(llama PUBLIC . ../include ../common)
 target_compile_features   (llama PUBLIC cxx_std_17) # don't bump
 
-target_link_libraries(llama PUBLIC ggml)
+target_link_libraries(llama PUBLIC ggml::all)
 
 if (BUILD_SHARED_LIBS)
     set_target_properties(llama PROPERTIES POSITION_INDEPENDENT_CODE ON)
