diff -Nur a/cmake/common.cmake b/cmake/common.cmake
--- a/cmake/common.cmake	2025-03-08 08:26:00.000000000 -0800
+++ b/cmake/common.cmake	2025-03-08 15:35:03.580480656 -0800
@@ -20,7 +20,7 @@
             list(APPEND C_FLAGS   ${WARNING_FLAGS})
             list(APPEND CXX_FLAGS ${WARNING_FLAGS})
 
-            ggml_get_flags(${CMAKE_CXX_COMPILER_ID} ${CMAKE_CXX_COMPILER_VERSION})
+            #ggml_get_flags(${CMAKE_CXX_COMPILER_ID} ${CMAKE_CXX_COMPILER_VERSION})
 
             add_compile_options("$<$<COMPILE_LANGUAGE:C>:${C_FLAGS};${GF_C_FLAGS}>"
                                 "$<$<COMPILE_LANGUAGE:CXX>:${CXX_FLAGS};${GF_CXX_FLAGS}>")
diff -Nur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2025-03-08 08:26:00.000000000 -0800
+++ b/CMakeLists.txt	2025-03-08 15:35:58.823815944 -0800
@@ -144,11 +144,7 @@
 #
 # 3rd-party
 #
-
-if (NOT TARGET ggml)
-    add_subdirectory(ggml)
-    # ... otherwise assume ggml is added by a parent CMakeLists.txt
-endif()
+find_package(ggml REQUIRED)
 
 #
 # build the library
diff -Nur a/common/CMakeLists.txt b/common/CMakeLists.txt
--- a/common/CMakeLists.txt	2025-03-08 08:26:00.000000000 -0800
+++ b/common/CMakeLists.txt	2025-03-08 17:55:45.090777839 -0800
@@ -138,4 +138,4 @@
 
 target_include_directories(${TARGET} PUBLIC .)
 target_compile_features   (${TARGET} PUBLIC cxx_std_17)
-target_link_libraries     (${TARGET} PRIVATE ${LLAMA_COMMON_EXTRA_LIBS} PUBLIC llama Threads::Threads)
+target_link_libraries     (${TARGET} PRIVATE ${LLAMA_COMMON_EXTRA_LIBS} PUBLIC llama ggml-base Threads::Threads)
diff -Nur a/examples/convert-llama2c-to-ggml/CMakeLists.txt b/examples/convert-llama2c-to-ggml/CMakeLists.txt
--- a/examples/convert-llama2c-to-ggml/CMakeLists.txt	2025-03-08 08:26:00.000000000 -0800
+++ b/examples/convert-llama2c-to-ggml/CMakeLists.txt	2025-03-08 18:06:19.494133447 -0800
@@ -1,5 +1,5 @@
 set(TARGET llama-convert-llama2c-to-ggml)
 add_executable(${TARGET} convert-llama2c-to-ggml.cpp)
 install(TARGETS ${TARGET} RUNTIME)
-target_link_libraries(${TARGET} PRIVATE common llama ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(${TARGET} PRIVATE common llama ggml-cpu ${CMAKE_THREAD_LIBS_INIT})
 target_compile_features(${TARGET} PRIVATE cxx_std_17)
diff -Nur a/examples/cvector-generator/CMakeLists.txt b/examples/cvector-generator/CMakeLists.txt
--- a/examples/cvector-generator/CMakeLists.txt	2025-03-08 08:26:00.000000000 -0800
+++ b/examples/cvector-generator/CMakeLists.txt	2025-03-08 18:04:45.637463471 -0800
@@ -1,5 +1,5 @@
 set(TARGET llama-cvector-generator)
 add_executable(${TARGET} cvector-generator.cpp pca.hpp)
 install(TARGETS ${TARGET} RUNTIME)
-target_link_libraries(${TARGET} PRIVATE common llama ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(${TARGET} PRIVATE common llama ggml-cpu ${CMAKE_THREAD_LIBS_INIT})
 target_compile_features(${TARGET} PRIVATE cxx_std_17)
diff -Nur a/examples/export-lora/CMakeLists.txt b/examples/export-lora/CMakeLists.txt
--- a/examples/export-lora/CMakeLists.txt	2025-03-08 08:26:00.000000000 -0800
+++ b/examples/export-lora/CMakeLists.txt	2025-03-08 18:01:39.830790279 -0800
@@ -1,5 +1,5 @@
 set(TARGET llama-export-lora)
 add_executable(${TARGET} export-lora.cpp)
 install(TARGETS ${TARGET} RUNTIME)
-target_link_libraries(${TARGET} PRIVATE common llama ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(${TARGET} PRIVATE common llama ggml-cpu ${CMAKE_THREAD_LIBS_INIT})
 target_compile_features(${TARGET} PRIVATE cxx_std_17)
diff -Nur a/examples/gguf/CMakeLists.txt b/examples/gguf/CMakeLists.txt
--- a/examples/gguf/CMakeLists.txt	2025-03-08 08:26:00.000000000 -0800
+++ b/examples/gguf/CMakeLists.txt	2025-03-08 15:38:10.880487279 -0800
@@ -1,5 +1,5 @@
 set(TARGET llama-gguf)
 add_executable(${TARGET} gguf.cpp)
 install(TARGETS ${TARGET} RUNTIME)
-target_link_libraries(${TARGET} PRIVATE ggml ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(${TARGET} PRIVATE ggml ggml::all ${CMAKE_THREAD_LIBS_INIT})
 target_compile_features(${TARGET} PRIVATE cxx_std_17)
diff -Nur a/examples/gguf-hash/CMakeLists.txt b/examples/gguf-hash/CMakeLists.txt
--- a/examples/gguf-hash/CMakeLists.txt	2025-03-08 08:26:00.000000000 -0800
+++ b/examples/gguf-hash/CMakeLists.txt	2025-03-08 15:39:15.963822915 -0800
@@ -18,5 +18,5 @@
 add_library(sha256 OBJECT deps/sha256/sha256.c deps/sha256/sha256.h)
 target_link_libraries(${TARGET} PRIVATE sha256)
 
-target_link_libraries(${TARGET} PRIVATE ggml ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(${TARGET} PRIVATE ggml ggml::all ${CMAKE_THREAD_LIBS_INIT})
 target_compile_features(${TARGET} PRIVATE cxx_std_17)
diff -Nur a/examples/imatrix/CMakeLists.txt b/examples/imatrix/CMakeLists.txt
--- a/examples/imatrix/CMakeLists.txt	2025-03-08 08:26:00.000000000 -0800
+++ b/examples/imatrix/CMakeLists.txt	2025-03-08 17:47:47.974094478 -0800
@@ -1,5 +1,5 @@
 set(TARGET llama-imatrix)
 add_executable(${TARGET} imatrix.cpp)
 install(TARGETS ${TARGET} RUNTIME)
-target_link_libraries(${TARGET} PRIVATE common llama ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(${TARGET} PRIVATE common llama ggml ${CMAKE_THREAD_LIBS_INIT})
 target_compile_features(${TARGET} PRIVATE cxx_std_17)
diff -Nur a/examples/llava/CMakeLists.txt b/examples/llava/CMakeLists.txt
--- a/examples/llava/CMakeLists.txt	2025-03-08 08:26:00.000000000 -0800
+++ b/examples/llava/CMakeLists.txt	2025-03-08 17:59:36.940785966 -0800
@@ -5,7 +5,7 @@
             clip.h
             )
 
-target_link_libraries(llava PRIVATE ggml llama ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(llava PRIVATE ggml llama ggml-cpu ${CMAKE_THREAD_LIBS_INIT})
 
 target_include_directories(llava PUBLIC .)
 target_include_directories(llava PUBLIC ../..)
diff -Nur a/examples/lookup/CMakeLists.txt b/examples/lookup/CMakeLists.txt
--- a/examples/lookup/CMakeLists.txt	2025-03-08 08:26:00.000000000 -0800
+++ b/examples/lookup/CMakeLists.txt	2025-03-08 17:47:22.814093596 -0800
@@ -13,11 +13,11 @@
 set(TARGET llama-lookup-merge)
 add_executable(${TARGET} lookup-merge.cpp)
 install(TARGETS ${TARGET} RUNTIME)
-target_link_libraries(${TARGET} PRIVATE common llama ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(${TARGET} PRIVATE common llama ggml::all ${CMAKE_THREAD_LIBS_INIT})
 target_compile_features(${TARGET} PRIVATE cxx_std_17)
 
 set(TARGET llama-lookup-stats)
 add_executable(${TARGET} lookup-stats.cpp)
 install(TARGETS ${TARGET} RUNTIME)
-target_link_libraries(${TARGET} PRIVATE common llama ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(${TARGET} PRIVATE common llama ggml::all ${CMAKE_THREAD_LIBS_INIT})
 target_compile_features(${TARGET} PRIVATE cxx_std_17)
diff -Nur a/examples/quantize-stats/CMakeLists.txt b/examples/quantize-stats/CMakeLists.txt
--- a/examples/quantize-stats/CMakeLists.txt	2025-03-08 08:26:00.000000000 -0800
+++ b/examples/quantize-stats/CMakeLists.txt	2025-03-08 17:40:32.770745834 -0800
@@ -1,6 +1,6 @@
 set(TARGET llama-quantize-stats)
 add_executable(${TARGET} quantize-stats.cpp)
 install(TARGETS ${TARGET} RUNTIME)
-target_link_libraries(${TARGET} PRIVATE llama build_info ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(${TARGET} PRIVATE llama ggml::all build_info ${CMAKE_THREAD_LIBS_INIT})
 target_include_directories(${TARGET} PRIVATE ../../common)
 target_compile_features(${TARGET} PRIVATE cxx_std_17)
diff -Nur a/examples/rpc/CMakeLists.txt b/examples/rpc/CMakeLists.txt
--- a/examples/rpc/CMakeLists.txt	2025-03-08 08:26:00.000000000 -0800
+++ b/examples/rpc/CMakeLists.txt	2025-03-08 15:41:38.207161272 -0800
@@ -1,2 +1,2 @@
 add_executable(rpc-server rpc-server.cpp)
-target_link_libraries(rpc-server PRIVATE ggml llama)
+target_link_libraries(rpc-server PRIVATE ggml ggml::all llama)
diff -Nur a/examples/simple/CMakeLists.txt b/examples/simple/CMakeLists.txt
--- a/examples/simple/CMakeLists.txt	2025-03-08 08:26:00.000000000 -0800
+++ b/examples/simple/CMakeLists.txt	2025-03-08 17:15:30.827359691 -0800
@@ -1,5 +1,5 @@
 set(TARGET llama-simple)
 add_executable(${TARGET} simple.cpp)
 install(TARGETS ${TARGET} RUNTIME)
-target_link_libraries(${TARGET} PRIVATE llama ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(${TARGET} PRIVATE llama ggml ggml::all ${CMAKE_THREAD_LIBS_INIT})
 target_compile_features(${TARGET} PRIVATE cxx_std_17)
diff -Nur a/examples/simple-chat/CMakeLists.txt b/examples/simple-chat/CMakeLists.txt
--- a/examples/simple-chat/CMakeLists.txt	2025-03-08 08:26:00.000000000 -0800
+++ b/examples/simple-chat/CMakeLists.txt	2025-03-08 17:16:49.104029116 -0800
@@ -1,5 +1,5 @@
 set(TARGET llama-simple-chat)
 add_executable(${TARGET} simple-chat.cpp)
 install(TARGETS ${TARGET} RUNTIME)
-target_link_libraries(${TARGET} PRIVATE llama ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(${TARGET} PRIVATE llama ggml ggml::all ${CMAKE_THREAD_LIBS_INIT})
 target_compile_features(${TARGET} PRIVATE cxx_std_17)
diff -Nur a/pocs/vdot/CMakeLists.txt b/pocs/vdot/CMakeLists.txt
--- a/pocs/vdot/CMakeLists.txt	2025-03-08 08:26:00.000000000 -0800
+++ b/pocs/vdot/CMakeLists.txt	2025-03-08 18:09:35.224140296 -0800
@@ -1,9 +1,9 @@
 set(TARGET llama-vdot)
 add_executable(${TARGET} vdot.cpp)
-target_link_libraries(${TARGET} PRIVATE common llama ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(${TARGET} PRIVATE common llama ggml-cpu ${CMAKE_THREAD_LIBS_INIT})
 target_compile_features(${TARGET} PRIVATE cxx_std_17)
 
 set(TARGET llama-q8dot)
 add_executable(${TARGET} q8dot.cpp)
-target_link_libraries(${TARGET} PRIVATE common llama ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(${TARGET} PRIVATE common llama ggml-cpu ${CMAKE_THREAD_LIBS_INIT})
 target_compile_features(${TARGET} PRIVATE cxx_std_17)
