diff -pNura lua-nginx-module.org/src/ngx_http_lua_module.c lua-nginx-module/src/ngx_http_lua_module.c
--- lua-nginx-module.org/src/ngx_http_lua_module.c	2022-06-27 22:37:24.000000000 -0700
+++ lua-nginx-module/src/ngx_http_lua_module.c	2022-09-06 17:54:35.236664131 -0700
@@ -856,36 +856,6 @@ ngx_http_lua_init(ngx_conf_t *cf)
     if (lmcf->lua == NULL) {
         dd("initializing lua vm");
 
-#ifndef OPENRESTY_LUAJIT
-        if (ngx_process != NGX_PROCESS_SIGNALLER && !ngx_test_config) {
-            ngx_log_error(NGX_LOG_ALERT, cf->log, 0,
-                          "detected a LuaJIT version which is not OpenResty's"
-                          "; many optimizations will be disabled and "
-                          "performance will be compromised (see "
-                          "https://github.com/openresty/luajit2 for "
-                          "OpenResty's LuaJIT or, even better, consider using "
-                          "the OpenResty releases from https://openresty.org/"
-                          "en/download.html)");
-        }
-#else
-#   if !defined(HAVE_LUA_RESETTHREAD)
-        ngx_log_error(NGX_LOG_ALERT, cf->log, 0,
-                      "detected an old version of OpenResty's LuaJIT missing "
-                      "the lua_resetthread API and thus the "
-                      "performance will be compromised; please upgrade to the "
-                      "latest version of OpenResty's LuaJIT: "
-                      "https://github.com/openresty/luajit2");
-#   endif
-#   if !defined(HAVE_LUA_EXDATA2)
-        ngx_log_error(NGX_LOG_ALERT, cf->log, 0,
-                      "detected an old version of OpenResty's LuaJIT missing "
-                      "the exdata2 API and thus the "
-                      "performance will be compromised; please upgrade to the "
-                      "latest version of OpenResty's LuaJIT: "
-                      "https://github.com/openresty/luajit2");
-#   endif
-#endif
-
         ngx_http_lua_content_length_hash =
                                   ngx_http_lua_hash_literal("content-length");
         ngx_http_lua_location_hash = ngx_http_lua_hash_literal("location");
