diff -pNura stream-lua-nginx-module.orig/src/ngx_stream_lua_module.c stream-lua-nginx-module/src/ngx_stream_lua_module.c
--- stream-lua-nginx-module.orig/src/ngx_stream_lua_module.c	2023-12-27 00:51:50.000000000 -0000
+++ stream-lua-nginx-module/src/ngx_stream_lua_module.c	2024-01-16 18:47:11.613333303 -0000
@@ -594,27 +594,6 @@ ngx_stream_lua_init(ngx_conf_t *cf)
     if (lmcf->lua == NULL) {
         dd("initializing lua vm");
 
-#ifndef OPENRESTY_LUAJIT
-        if (ngx_process != NGX_PROCESS_SIGNALLER && !ngx_test_config) {
-            ngx_log_error(NGX_LOG_ALERT, cf->log, 0,
-                          "detected a LuaJIT version which is not OpenResty's"
-                          "; many optimizations will be disabled and "
-                          "performance will be compromised (see "
-                          "https://github.com/openresty/luajit2 for "
-                          "OpenResty's LuaJIT or, even better, consider using "
-                          "the OpenResty releases from https://openresty.org/"
-                          "en/download.html)");
-        }
-#else
-#   if !defined(HAVE_LUA_EXDATA2)
-        ngx_log_error(NGX_LOG_ALERT, cf->log, 0,
-                      "detected an old version of OpenResty's LuaJIT missing "
-                      "the exdata2 API and thus the "
-                      "performance will be compromised; please upgrade to the "
-                      "latest version of OpenResty's LuaJIT: "
-                      "https://github.com/openresty/luajit2");
-#   endif
-#endif
 
 
         rc = ngx_stream_lua_init_vm(&lmcf->lua, NULL, cf->cycle, cf->pool,
