diff -Nur a/Makefile b/Makefile
--- a/Makefile	2026-05-13 06:57:01.000000000 -0700
+++ b/Makefile	2026-05-24 13:23:21.556396859 -0700
@@ -1,6 +1,6 @@
 KERNEL_VERSION = linux-6.12.87
 KERNEL_REMOTE = https://cdn.kernel.org/pub/linux/kernel/v6.x/$(KERNEL_VERSION).tar.xz
-KERNEL_TARBALL = tarballs/$(KERNEL_VERSION).tar.xz
+KERNEL_TARBALL ?= tarballs/$(KERNEL_VERSION).tar.xz
 KERNEL_SOURCES = $(KERNEL_VERSION)
 KERNEL_PATCHES = $(shell find patches/ -name "0*.patch" | sort)
 KERNEL_C_BUNDLE = kernel.c
@@ -28,7 +28,7 @@
 else
     OS := $(shell uname -s)
 endif
-ifeq ($(ARCH),)
+ifeq ($(ARCH),x86)
 	GUESTARCH := $(HOSTARCH)
 	STRIP := strip
 else ifeq ($(ARCH),arm64)
@@ -146,9 +146,9 @@
 	$(CC) -shared -DABI_VERSION=$(ABI_VERSION) -O2 -o $@ $(KERNEL_C_BUNDLE) $(QBOOT_C_BUNDLE) $(INITRD_C_BUNDLE) -Wl,--kill-at -Wl,--nxcompat
 else
 	$(CC) -fPIC -DABI_VERSION=$(ABI_VERSION) -shared $(SONAME_$(OS)) -o $@ $(KERNEL_C_BUNDLE) $(QBOOT_C_BUNDLE) $(INITRD_C_BUNDLE)
-ifeq ($(OS),Linux)
-	$(STRIP) $(KRUNFW_BINARY_$(OS))
-endif
+#ifeq ($(OS),Linux)
+#	$(STRIP) $(KRUNFW_BINARY_$(OS))
+#endif
 endif
 
 install:
