--- a/hopper/epilogue_fwd.hpp
+++ b/hopper/epilogue_fwd.hpp
@@ -97,7 +97,7 @@ using StrideO = cute::Stride<int64_t, _1, int64_t, int64_t, int64_t>;
     using CopyOpR2S = std::conditional_t<
         ArchTag::kMinComputeCapability >= 90,
         // cute::SM90_U32x4_STSM_N if Element size is 2 bytes (fp16, bf16)
-        decltype(cutlass::epilogue::collective::detail::sm90_get_smem_store_op_for_accumulator<StrideO, Element>()),
+        decltype(cutlass::epilogue::collective::detail::sm90_get_smem_store_op_for_accumulator<StrideO, Element, decltype(select<0, 1>(TileShape_MNK_PV{}))>()),
         AutoVectorizingCopyWithAssumedAlignment<128>
     >;
