--- a/hopper/epilogue_fwd.hpp
+++ b/hopper/epilogue_fwd.hpp
@@ -97,7 +97,7 @@
     using CopyOpR2S = std::conditional_t<
         ArchTag::kMinComputeCapability >= 90,
         // cute::SM90_U32x4_STSM_N if Element size is 2 bytes (fp16, bf16)
-        decltype(cutlass::epilogue::collective::detail::sm90_get_smem_store_op_for_accumulator<StrideO, Element>()),
+        decltype(cutlass::epilogue::collective::detail::sm90_get_smem_store_op_for_accumulator<StrideO, Element, decltype(select<0, 1>(TileShape_MNK_PV{}))>()),
         AutoVectorizingCopyWithAssumedAlignment<128>
     >;
     using SmemCopyAtomO = Copy_Atom<CopyOpR2S, Element>;
--- a/hopper/utils.h
+++ b/hopper/utils.h
@@ -412,19 +412,12 @@
 auto
 to_tiled_mma_sm100_ts(
     TiledMMA<MMA_Atom<
-      MMA_Traits<SM100_MMA_F8F6F4_SS, a_type, b_type, c_type,
-                    cute::C<M>, cute::C<N>,
-                    cute::integral_constant<UMMA::Major, a_major>,
-                    cute::integral_constant<UMMA::Major, b_major>,
-                    cute::integral_constant<UMMA::ScaleIn, a_neg>,
-                    cute::integral_constant<UMMA::ScaleIn, b_neg>>,
+      SM100_MMA_F8F6F4_SS<a_type, b_type, c_type, M, N, a_major, b_major, a_neg, b_neg>,
       TAs...>, TMs...>) {

   return TiledMMA<MMA_Atom<
-    MMA_Traits<SM100_MMA_F8F6F4_TS<a_type, b_type, c_type,
-                                M, N,
-                                a_major, b_major,
-                                a_neg, b_neg, UMMA::Saturate::False>>,
+    SM100_MMA_F8F6F4_TS<a_type, b_type, c_type, M, N, a_major, b_major,
+                         a_neg, b_neg, UMMA::Saturate::False>,
     TAs...>, TMs...>{};
 }
