diff -Nur a/setup.py b/setup.py
--- a/setup.py	2025-05-30 08:50:58.000000000 -0700
+++ b/setup.py	2025-06-03 10:43:28.284856254 -0700
@@ -203,7 +203,10 @@
             build_tool = []
         # Make sure we use the nvcc from CUDA_HOME
         if _is_cuda():
-            cmake_args += [f'-DCMAKE_CUDA_COMPILER={CUDA_HOME}/bin/nvcc']
+            cmake_args += [
+                f'-DCMAKE_CUDA_COMPILER={CUDA_HOME}/bin/nvcc',
+                '-DUSE_SYSTEM_NVTX=ON',
+            ]
         subprocess.check_call(
             ['cmake', ext.cmake_lists_dir, *build_tool, *cmake_args],
             cwd=self.build_temp)
