--- pyneighborhood-0.5.4.orig/setup.py	2011-03-22 20:51:25.000000000 +0100
+++ pyneighborhood-0.5.4/setup.py	2011-04-18 19:13:32.637996266 +0200
@@ -19,6 +19,7 @@
 import sys
 import subprocess
 import stat
+import glob
 from distutils.core import setup
 
 setup(name='pyneighborhood',
@@ -30,5 +31,8 @@
       maintainer_email='info@stefan-betz.net',
       url='http://launchpad.net/pyneighborhood',
       scripts=['pyNeighborhood','pyNeighborhood-debug'],
-      packages=['pyneighborhood']
+      packages=['pyneighborhood'],
+      data_files=[ ('/usr/share/applications', ['pyNeighborhood.desktop']),
+                  ('/usr/share/pyneighborhood/icons', glob.glob("icons/*.png")),
+                  ('/usr/share/pyneighborhood/ui', glob.glob("ui/*.glade"))]
      )
