commit cf17b8b3b1454872a583af4d4ca32790661c4bb5
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Fri May 13 01:14:08 2011 -0400

    stuff for bionics crippled locale.h

diff --git a/form/fty_num.c b/form/fty_num.c
index 4bd7132..a1540f0 100644
--- a/form/fty_num.c
+++ b/form/fty_num.c
@@ -95,12 +95,7 @@ Generic_This_Type(void *arg)
 	  argn->precision = args->precision;
 	  argn->low = args->low;
 	  argn->high = args->high;
-
-#if HAVE_LOCALE_H
-	  argn->L = localeconv();
-#else
 	  argn->L = NULL;
-#endif
 	}
     }
   return (void *)argn;
@@ -247,7 +242,7 @@ Check_This_Field(FIELD *field, const void *argp)
 	    break;
 	  bp++;
 	}
-      if (isDecimalPoint(*bp))
+      if (*bp=='.')
 	{
 	  bp++;
 	  while (*bp)
@@ -294,12 +289,11 @@ static bool
 Check_This_Character(int c, const void *argp)
 {
   const thisARG *argn = (const thisARG *)argp;
-  struct lconv *L = argn->L;
 
-  return ((isDigit(c) ||
+  return (((c >= '0' && c <= '9') ||
 	   c == '+' ||
 	   c == '-' ||
-	   isDecimalPoint(c))
+	   c == '.')
 	  ? TRUE
 	  : FALSE);
 }
