diff -rup amdbrook/platform/brcc/src/amdhlsl/AMDhlsl/ILCodeGen/HeaderFiles/ilregister.hpp amdbrook.new/platform/brcc/src/amdhlsl/AMDhlsl/ILCodeGen/HeaderFiles/ilregister.hpp
--- amdbrook/platform/brcc/src/amdhlsl/AMDhlsl/ILCodeGen/HeaderFiles/ilregister.hpp	2008-12-03 01:12:53.000000000 +0100
+++ amdbrook.new/platform/brcc/src/amdhlsl/AMDhlsl/ILCodeGen/HeaderFiles/ilregister.hpp	2009-02-02 17:47:42.000000000 +0100
@@ -89,6 +89,7 @@ jurisdiction and venue of these courts.
 #ifndef AMD_ILREGISTER_HPP
 #define AMD_ILREGISTER_HPP
 
+#include <cstring>
 #include "convert.hpp"
 
 #ifndef IL_BIT
diff -rup amdbrook/platform/brcc/src/amdhlsl/InterfaceUtils/SourceFiles/calutAMDhlslCompiler.cpp amdbrook.new/platform/brcc/src/amdhlsl/InterfaceUtils/SourceFiles/calutAMDhlslCompiler.cpp
--- amdbrook/platform/brcc/src/amdhlsl/InterfaceUtils/SourceFiles/calutAMDhlslCompiler.cpp	2008-12-03 01:12:53.000000000 +0100
+++ amdbrook.new/platform/brcc/src/amdhlsl/InterfaceUtils/SourceFiles/calutAMDhlslCompiler.cpp	2009-02-02 18:02:53.000000000 +0100
@@ -95,6 +95,8 @@ jurisdiction and venue of these courts.
 #include <assert.h>
 #include "amuABIMultiBinary.h"
 #include "hlfrontendapi.hpp"
+#include <cstdlib>
+#include <cstring>
 
 static ErrorState     amdHLSLerrst;
 
diff -rup amdbrook/platform/brcc/src/Deprecated/ps2arb_lexer.cpp amdbrook.new/platform/brcc/src/Deprecated/ps2arb_lexer.cpp
--- amdbrook/platform/brcc/src/Deprecated/ps2arb_lexer.cpp	2008-12-03 01:12:53.000000000 +0100
+++ amdbrook.new/platform/brcc/src/Deprecated/ps2arb_lexer.cpp	2009-02-02 18:05:10.000000000 +0100
@@ -27,6 +27,7 @@
 #define YY_FLEX_MINOR_VERSION 5
 
 #include <stdio.h>
+#include <cstring>
 
 
 /* cfront 1.2 defines "c_plusplus" instead of "__cplusplus" */
diff -rup amdbrook/platform/brcc/src/HighLevelFE/Parser/lexer.cpp amdbrook.new/platform/brcc/src/HighLevelFE/Parser/lexer.cpp
--- amdbrook/platform/brcc/src/HighLevelFE/Parser/lexer.cpp	2008-12-03 01:12:53.000000000 +0100
+++ amdbrook.new/platform/brcc/src/HighLevelFE/Parser/lexer.cpp	2009-02-02 18:06:31.000000000 +0100
@@ -9,6 +9,7 @@
 #define YY_FLEX_MINOR_VERSION 5
 
 #include <stdio.h>
+#include <cstring>
 
 
 /* cfront 1.2 defines "c_plusplus" instead of "__cplusplus" */
diff -rup amdbrook/platform/include/brook/CPU/brtvector.hpp amdbrook.new/platform/include/brook/CPU/brtvector.hpp
--- amdbrook/platform/include/brook/CPU/brtvector.hpp	2008-12-03 01:12:53.000000000 +0100
+++ amdbrook.new/platform/include/brook/CPU/brtvector.hpp	2009-02-02 17:45:24.000000000 +0100
@@ -185,7 +185,7 @@ template <class T> typename GetValueOf<T
 template <class T> static typename GetValueOf<T>::type GetAt (const T& in,int i) {
     return in.getAt(i);
 }
-#define SPECIALGETAT(TYP) template <> static TYP GetAt (const TYP& in,int UNUSED(i)) {return in;}
+#define SPECIALGETAT(TYP) template <> TYP GetAt (const TYP& in,int UNUSED(i)) {return in;}
 
 SPECIALGETAT(int)
 SPECIALGETAT(unsigned int)
diff -rup amdbrook/platform/Makefile amdbrook.new/platform/Makefile
--- amdbrook/platform/Makefile	2008-12-03 01:12:53.000000000 +0100
+++ amdbrook.new/platform/Makefile	2009-02-03 19:40:28.000000000 +0100
@@ -33,7 +33,7 @@
 #
 
 ROOTDIR := ..
-SUBDIRS := brcc runtime utils
+SUBDIRS := brcc runtime
 SDKDIR := $(ROOTDIR)/sdk
 
 # We don't use the default "all" rule here because we want to build the Brook
