diff -rup amdcal/samples/app/compute_matmult/compute_matmult.cpp amdcal.new/samples/app/compute_matmult/compute_matmult.cpp
--- amdcal/samples/app/compute_matmult/compute_matmult.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/app/compute_matmult/compute_matmult.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -244,7 +244,7 @@ CALint SetupData ( CALmodule *module, CA
 }
 
     int
-main(CALuint argc, CALchar** argv)
+main(CALint argc, CALchar** argv)
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/app/double_matmult/double_matmult.cpp amdcal.new/samples/app/double_matmult/double_matmult.cpp
--- amdcal/samples/app/double_matmult/double_matmult.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/app/double_matmult/double_matmult.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -393,7 +393,7 @@ isDoubleEqual(CALdouble a, CALdouble b)
 
 
     int
-main(CALuint argc, CALchar** argv)
+main(CALint argc, CALchar** argv)
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/app/hellocal/hellocal.cpp amdcal.new/samples/app/hellocal/hellocal.cpp
--- amdcal/samples/app/hellocal/hellocal.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/app/hellocal/hellocal.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -100,7 +100,7 @@ jurisdiction and venue of these courts.
 /**
  * \brief Main function that shows the basics needed to setup and run a CAL program
  */
-int main(CALuint argc, CALchar** argv)
+int main(CALint argc, CALchar** argv)
 {
 	// Validate the available CAL runtime version 
 	if(!ValidateCALRuntime())
diff -rup amdcal/samples/app/lds_transpose/lds_transpose.cpp amdcal.new/samples/app/lds_transpose/lds_transpose.cpp
--- amdcal/samples/app/lds_transpose/lds_transpose.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/app/lds_transpose/lds_transpose.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -282,7 +282,7 @@ CALint SetupData ( CALmodule *module, CA
 }
 
     int
-main(CALuint argc, CALchar** argv)
+main(CALint argc, CALchar** argv)
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/app/matrix_transpose/matrix_transpose.cpp amdcal.new/samples/app/matrix_transpose/matrix_transpose.cpp
--- amdcal/samples/app/matrix_transpose/matrix_transpose.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/app/matrix_transpose/matrix_transpose.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -330,7 +330,7 @@ CALint SetupData (CALmodule *module, CAL
 }
 
     int
-main(CALuint argc, CALchar** argv)
+main(CALint argc, CALchar** argv)
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/app/memexport_matmult/memexport_matmult.cpp amdcal.new/samples/app/memexport_matmult/memexport_matmult.cpp
--- amdcal/samples/app/memexport_matmult/memexport_matmult.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/app/memexport_matmult/memexport_matmult.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -255,7 +255,7 @@ CALint SetupData ( CALmodule *module, CA
 
 
     int
-main(CALuint argc, CALchar** argv)
+main(CALint argc, CALchar** argv)
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/app/memimport_matmult/memimport_matmult.cpp amdcal.new/samples/app/memimport_matmult/memimport_matmult.cpp
--- amdcal/samples/app/memimport_matmult/memimport_matmult.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/app/memimport_matmult/memimport_matmult.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -253,7 +253,7 @@ CALint SetupData ( CALmodule *module, CA
 
 
     int
-main(CALuint argc, CALchar** argv)
+main(CALint argc, CALchar** argv)
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/app/NLM_Denoise/nlm_denoise.cpp amdcal.new/samples/app/NLM_Denoise/nlm_denoise.cpp
--- amdcal/samples/app/NLM_Denoise/nlm_denoise.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/app/NLM_Denoise/nlm_denoise.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -337,7 +337,7 @@ CALint SetupData ( CALmodule *module, CA
 }
 
     int
-main(CALuint argc, CALchar** argv)
+main(CALint argc, CALchar** argv)
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/app/NLM_Denoise_Compute/NLM_Denoise_Compute.cpp amdcal.new/samples/app/NLM_Denoise_Compute/NLM_Denoise_Compute.cpp
--- amdcal/samples/app/NLM_Denoise_Compute/NLM_Denoise_Compute.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/app/NLM_Denoise_Compute/NLM_Denoise_Compute.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -338,7 +338,7 @@ CALint SetupData(CALmodule *module, CALr
 }
 
     int
-main(CALuint argc, CALchar** argv)
+main(CALint argc, CALchar** argv)
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/app/simple_matmult/simple_matmult.cpp amdcal.new/samples/app/simple_matmult/simple_matmult.cpp
--- amdcal/samples/app/simple_matmult/simple_matmult.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/app/simple_matmult/simple_matmult.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -242,7 +242,7 @@ CALint SetupData ( CALmodule *module, CA
 
 
     int
-main(CALuint argc, CALchar** argv)
+main(CALint argc, CALchar** argv)
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/languages/IL/bursting_CS/bursting_CS.cpp amdcal.new/samples/languages/IL/bursting_CS/bursting_CS.cpp
--- amdcal/samples/languages/IL/bursting_CS/bursting_CS.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/languages/IL/bursting_CS/bursting_CS.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -232,7 +232,7 @@ CALint SetupData ( CALmodule *module, CA
 }
 
     int
-main(CALuint argc, CALchar** argv)
+main(CALint argc, CALchar** argv)
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/languages/IL/bursting_IL/bursting_IL.cpp amdcal.new/samples/languages/IL/bursting_IL/bursting_IL.cpp
--- amdcal/samples/languages/IL/bursting_IL/bursting_IL.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/languages/IL/bursting_IL/bursting_IL.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -234,7 +234,7 @@ CALint SetupData ( CALmodule *module, CA
 }
 
     int
-main(CALuint argc, CALchar** argv)
+main(CALint argc, CALchar** argv)
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/languages/IL/comparison_instr_IL/comparison_instr_IL.cpp amdcal.new/samples/languages/IL/comparison_instr_IL/comparison_instr_IL.cpp
--- amdcal/samples/languages/IL/comparison_instr_IL/comparison_instr_IL.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/languages/IL/comparison_instr_IL/comparison_instr_IL.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -251,7 +251,7 @@ CALint SetupData(CALmodule* module, CALr
 
 
 
-int main(CALuint argc, CALchar** argv)
+int main(CALint argc, CALchar** argv)
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/languages/IL/constbuf_IL/constbuf_IL.cpp amdcal.new/samples/languages/IL/constbuf_IL/constbuf_IL.cpp
--- amdcal/samples/languages/IL/constbuf_IL/constbuf_IL.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/languages/IL/constbuf_IL/constbuf_IL.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -226,7 +226,7 @@ CALint SetupData(CALmodule* module, CALr
 
 
     int
-main(CALuint argc, CALchar** argv)
+main(CALint argc, CALchar** argv)
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/languages/IL/float_instr_IL/float_instr_IL.cpp amdcal.new/samples/languages/IL/float_instr_IL/float_instr_IL.cpp
--- amdcal/samples/languages/IL/float_instr_IL/float_instr_IL.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/languages/IL/float_instr_IL/float_instr_IL.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -241,7 +241,7 @@ CALint SetupData ( CALmodule *module, CA
 
 
 
-int main(CALuint argc, CALchar** argv)
+int main(CALint argc, CALchar** argv)
 {
     if (!ParseCommandLine(argc, argv, &Info))
     {
diff -rup amdcal/samples/languages/IL/gather_IL/gather_IL.cpp amdcal.new/samples/languages/IL/gather_IL/gather_IL.cpp
--- amdcal/samples/languages/IL/gather_IL/gather_IL.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/languages/IL/gather_IL/gather_IL.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -265,7 +265,7 @@ CALint SetupData(CALmodule* module, CALr
 }
 
     int
-main( CALuint argc, CALchar** argv )
+main( CALint argc, CALchar** argv )
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/languages/IL/input_IL/input_IL.cpp amdcal.new/samples/languages/IL/input_IL/input_IL.cpp
--- amdcal/samples/languages/IL/input_IL/input_IL.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/languages/IL/input_IL/input_IL.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -240,7 +240,7 @@ CALint SetupData(CALmodule* module, CALr
 }
 
 
-int main(CALuint argc, CALchar** argv)
+int main(CALint argc, CALchar** argv)
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/languages/IL/int_instr_IL/int_instr_IL.cpp amdcal.new/samples/languages/IL/int_instr_IL/int_instr_IL.cpp
--- amdcal/samples/languages/IL/int_instr_IL/int_instr_IL.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/languages/IL/int_instr_IL/int_instr_IL.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -225,7 +225,7 @@ CALint SetupData(CALmodule* module, CALr
 
 
     int
-main( CALuint argc, CALchar** argv )
+main( CALint argc, CALchar** argv )
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/languages/IL/loop_IL/loop_IL.cpp amdcal.new/samples/languages/IL/loop_IL/loop_IL.cpp
--- amdcal/samples/languages/IL/loop_IL/loop_IL.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/languages/IL/loop_IL/loop_IL.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -212,7 +212,7 @@ CALint SetupData(CALmodule* module, CALr
     return 1;
 }
 
-int main(CALuint argc, CALchar** argv)
+int main(CALint argc, CALchar** argv)
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/languages/IL/output_IL/output_IL.cpp amdcal.new/samples/languages/IL/output_IL/output_IL.cpp
--- amdcal/samples/languages/IL/output_IL/output_IL.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/languages/IL/output_IL/output_IL.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -203,7 +203,7 @@ CALint SetupData ( CALmodule *module, CA
 
 
     int
-main( CALuint argc, CALchar** argv )
+main( CALint argc, CALchar** argv )
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/languages/IL/scatter_IL/scatter_IL.cpp amdcal.new/samples/languages/IL/scatter_IL/scatter_IL.cpp
--- amdcal/samples/languages/IL/scatter_IL/scatter_IL.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/languages/IL/scatter_IL/scatter_IL.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -243,7 +243,7 @@ CALint SetupData(CALmodule* module, CALr
 }
 
     int
-main(CALuint argc, CALchar** argv)
+main(CALint argc, CALchar** argv)
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/languages/IL/scratch_buffer_IL/scratch_buffer_IL.cpp amdcal.new/samples/languages/IL/scratch_buffer_IL/scratch_buffer_IL.cpp
--- amdcal/samples/languages/IL/scratch_buffer_IL/scratch_buffer_IL.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/languages/IL/scratch_buffer_IL/scratch_buffer_IL.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -229,7 +229,7 @@ CALint SetupData ( CALmodule *module, CA
 }
 
     int
-main( CALuint argc, CALchar** argv )
+main( CALint argc, CALchar** argv )
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/runtime/cachespeed/cachespeed.cpp amdcal.new/samples/runtime/cachespeed/cachespeed.cpp
--- amdcal/samples/runtime/cachespeed/cachespeed.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/runtime/cachespeed/cachespeed.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -401,7 +401,7 @@ CALint run()
 }
 
     int
-main(CALuint argc, CALchar** argv)
+main(CALint argc, CALchar** argv)
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/runtime/cachespeed_CS/cachespeed_CS.cpp amdcal.new/samples/runtime/cachespeed_CS/cachespeed_CS.cpp
--- amdcal/samples/runtime/cachespeed_CS/cachespeed_CS.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/runtime/cachespeed_CS/cachespeed_CS.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -412,7 +412,7 @@ Data* data = new Data[numInputs + numOut
 }
 
     int
-main( CALuint argc, CALchar** argv )
+main( CALint argc, CALchar** argv )
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/runtime/domain/domain.cpp amdcal.new/samples/runtime/domain/domain.cpp
--- amdcal/samples/runtime/domain/domain.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/runtime/domain/domain.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -393,7 +393,7 @@ static CALuint RunProgram ( CALcontext *
 
 
     int
-main(CALuint argc, CALchar** argv)
+main(CALint argc, CALchar** argv)
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/runtime/export_burst_perf/export_burst_perf.cpp amdcal.new/samples/runtime/export_burst_perf/export_burst_perf.cpp
--- amdcal/samples/runtime/export_burst_perf/export_burst_perf.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/runtime/export_burst_perf/export_burst_perf.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -414,7 +414,7 @@ CALint run()
 }
 
     int
-main(CALuint argc, CALchar** argv)
+main(CALint argc, CALchar** argv)
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/runtime/exportspeed/exportspeed.cpp amdcal.new/samples/runtime/exportspeed/exportspeed.cpp
--- amdcal/samples/runtime/exportspeed/exportspeed.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/runtime/exportspeed/exportspeed.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -389,7 +389,7 @@ CALuint run()
 }
 
     int
-main(CALuint argc, CALchar** argv)
+main(CALint argc, CALchar** argv)
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/runtime/exportspeed_CS/exportspeed_CS.cpp amdcal.new/samples/runtime/exportspeed_CS/exportspeed_CS.cpp
--- amdcal/samples/runtime/exportspeed_CS/exportspeed_CS.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/runtime/exportspeed_CS/exportspeed_CS.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -371,7 +371,7 @@ CALuint run()
 }
 
     int
-main( CALuint argc, CALchar** argv )
+main( CALint argc, CALchar** argv )
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/runtime/importspeed/importspeed.cpp amdcal.new/samples/runtime/importspeed/importspeed.cpp
--- amdcal/samples/runtime/importspeed/importspeed.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/runtime/importspeed/importspeed.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -410,7 +410,7 @@ CALuint run()
 }
 
     int
-main(CALuint argc, CALchar** argv)
+main(CALint argc, CALchar** argv)
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/runtime/importspeed_CS/importspeed_CS.cpp amdcal.new/samples/runtime/importspeed_CS/importspeed_CS.cpp
--- amdcal/samples/runtime/importspeed_CS/importspeed_CS.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/runtime/importspeed_CS/importspeed_CS.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -375,7 +375,7 @@ CALuint run()
 }
 
     int
-main(CALuint argc, CALchar** argv)
+main(CALint argc, CALchar** argv)
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/runtime/inputspeed/inputspeed.cpp amdcal.new/samples/runtime/inputspeed/inputspeed.cpp
--- amdcal/samples/runtime/inputspeed/inputspeed.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/runtime/inputspeed/inputspeed.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -403,7 +403,7 @@ CALuint run()
 }
 
     int
-main(CALuint argc, CALchar** argv)
+main(CALint argc, CALchar** argv)
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/runtime/inputspeed_CS/inputspeed_CS.cpp amdcal.new/samples/runtime/inputspeed_CS/inputspeed_CS.cpp
--- amdcal/samples/runtime/inputspeed_CS/inputspeed_CS.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/runtime/inputspeed_CS/inputspeed_CS.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -385,7 +385,7 @@ CALint run()
     return 0;
 }
 
-int main(CALuint argc, CALchar** argv)
+int main(CALint argc, CALchar** argv)
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/runtime/ldsread_CS/ldsread_CS.cpp amdcal.new/samples/runtime/ldsread_CS/ldsread_CS.cpp
--- amdcal/samples/runtime/ldsread_CS/ldsread_CS.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/runtime/ldsread_CS/ldsread_CS.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -380,7 +380,7 @@ CALuint run()
 
     return 0;
 }
-int main(CALuint argc, CALchar** argv)
+int main(CALint argc, CALchar** argv)
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/runtime/ldswrite_CS/ldswrite_CS.cpp amdcal.new/samples/runtime/ldswrite_CS/ldswrite_CS.cpp
--- amdcal/samples/runtime/ldswrite_CS/ldswrite_CS.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/runtime/ldswrite_CS/ldswrite_CS.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -363,7 +363,7 @@ CALuint run()
     return 0;
 }
 
-int main(CALuint argc, CALchar** argv)
+int main(CALint argc, CALchar** argv)
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/runtime/memtiming/memtiming.cpp amdcal.new/samples/runtime/memtiming/memtiming.cpp
--- amdcal/samples/runtime/memtiming/memtiming.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/runtime/memtiming/memtiming.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -302,7 +302,7 @@ Shutdown(CALdevice dev, CALcontext ctx)
 }
 
     int
-main(CALuint argc, CALchar** argv)
+main(CALint argc, CALchar** argv)
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/runtime/nonblocking_map/nonblocking_map.cpp amdcal.new/samples/runtime/nonblocking_map/nonblocking_map.cpp
--- amdcal/samples/runtime/nonblocking_map/nonblocking_map.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/runtime/nonblocking_map/nonblocking_map.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -435,7 +435,7 @@ CleanupBuffers(BufferInfo* buffers, CALc
 }
 
     int
-main(CALuint argc, CALchar** argv )
+main(CALint argc, CALchar** argv )
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/runtime/outputspeed/outputspeed.cpp amdcal.new/samples/runtime/outputspeed/outputspeed.cpp
--- amdcal/samples/runtime/outputspeed/outputspeed.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/runtime/outputspeed/outputspeed.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -351,7 +351,7 @@ CALuint run()
 }
 
     int
-main( CALuint argc, CALchar** argv )
+main( CALint argc, CALchar** argv )
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/runtime/perf_counters/perf_counters.cpp amdcal.new/samples/runtime/perf_counters/perf_counters.cpp
--- amdcal/samples/runtime/perf_counters/perf_counters.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/runtime/perf_counters/perf_counters.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -144,7 +144,7 @@ CALint SetupData(CALmodule* module, CALr
 }
 
 
-int main(CALuint argc, CALchar** argv)
+int main(CALint argc, CALchar** argv)
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/runtime/throughput/throughput.cpp amdcal.new/samples/runtime/throughput/throughput.cpp
--- amdcal/samples/runtime/throughput/throughput.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/runtime/throughput/throughput.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -398,7 +398,7 @@ CALuint run()
 }
 
     int
-main(CALuint argc, CALchar** argv)
+main(CALint argc, CALchar** argv)
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/runtime/throughput_CS/throughput_CS.cpp amdcal.new/samples/runtime/throughput_CS/throughput_CS.cpp
--- amdcal/samples/runtime/throughput_CS/throughput_CS.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/runtime/throughput_CS/throughput_CS.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -423,7 +423,7 @@ CALint run()
 }
 
     int
-main( CALuint argc, CALchar** argv )
+main( CALint argc, CALchar** argv )
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/tutorial/basic/basic.cpp amdcal.new/samples/tutorial/basic/basic.cpp
--- amdcal/samples/tutorial/basic/basic.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/tutorial/basic/basic.cpp	2009-02-03 18:16:50.000000000 +0100
@@ -111,6 +111,8 @@ jurisdiction and venue of these courts.
 
 #include <assert.h>
 #include <string>
+#include <cstdlib>
+#include <cstring>
 
 static const unsigned int Width     = 256;
 static const unsigned int Height    = 256;
@@ -309,7 +311,7 @@ PrintDetails(std::string kernelType)
 }
 
 int
-main(CALuint argc, CALchar** argv)
+main(CALint argc, CALchar** argv)
 {  
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/tutorial/CompileProgram/CompileProgram.cpp amdcal.new/samples/tutorial/CompileProgram/CompileProgram.cpp
--- amdcal/samples/tutorial/CompileProgram/CompileProgram.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/tutorial/CompileProgram/CompileProgram.cpp	2009-02-03 18:17:35.000000000 +0100
@@ -284,7 +284,7 @@ PrintDetails(std::string kernelType)
  * \brief Main function
  */
 int
-main(CALuint argc, CALchar** argv)
+main(CALint argc, CALchar** argv)
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/tutorial/CreateContext/CreateContext.cpp amdcal.new/samples/tutorial/CreateContext/CreateContext.cpp
--- amdcal/samples/tutorial/CreateContext/CreateContext.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/tutorial/CreateContext/CreateContext.cpp	2009-02-03 18:17:26.000000000 +0100
@@ -157,7 +157,7 @@ calAssert(CALresult retval, std::string 
 /**
  * \brief Main function
  */
-int main(CALuint argc, CALchar** argv)
+int main(CALint argc, CALchar** argv)
 {    
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/tutorial/CreateResource/CreateResource.cpp amdcal.new/samples/tutorial/CreateResource/CreateResource.cpp
--- amdcal/samples/tutorial/CreateResource/CreateResource.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/tutorial/CreateResource/CreateResource.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -110,6 +110,7 @@ jurisdiction and venue of these courts.
 #include "Samples.h"
 #include <assert.h>
 #include <string>
+#include <cstdlib>
 
 /**
  * \param retval Value returned from a function
@@ -165,7 +166,7 @@ PrintDetails()
  * \brief Main function
  */
 int 
-main(CALuint argc, CALchar** argv)
+main(CALint argc, CALchar** argv)
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/tutorial/DoublePrecision/DoublePrecision.cpp amdcal.new/samples/tutorial/DoublePrecision/DoublePrecision.cpp
--- amdcal/samples/tutorial/DoublePrecision/DoublePrecision.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/tutorial/DoublePrecision/DoublePrecision.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -110,6 +110,7 @@ jurisdiction and venue of these courts.
 #include <assert.h>
 #include <string>
 #include <cmath>
+#include <cstdlib>
 
 static const CALuint Width     = 8;
 static const CALuint Height    = 8;
@@ -288,7 +289,7 @@ PrintDetails(std::string kernelType)
  * \brief Main function
  */
 int
-main(CALuint argc, CALchar** argv)
+main(CALint argc, CALchar** argv)
 {  
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/tutorial/DownloadReadback/DownloadReadback.cpp amdcal.new/samples/tutorial/DownloadReadback/DownloadReadback.cpp
--- amdcal/samples/tutorial/DownloadReadback/DownloadReadback.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/tutorial/DownloadReadback/DownloadReadback.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -243,7 +243,7 @@ PrintDetails()
  * \brief Main function
  */
 int
-main(CALuint argc, CALchar** argv)
+main(CALint argc, CALchar** argv)
 {
 	//////////////////////////////////////////////////////////////////////////
     //
diff -rup amdcal/samples/tutorial/FindNumDevices/FindNumDevices.cpp amdcal.new/samples/tutorial/FindNumDevices/FindNumDevices.cpp
--- amdcal/samples/tutorial/FindNumDevices/FindNumDevices.cpp	2008-12-03 01:13:38.000000000 +0100
+++ amdcal.new/samples/tutorial/FindNumDevices/FindNumDevices.cpp	2009-02-03 18:06:42.000000000 +0100
@@ -120,7 +120,7 @@ static SampleInfo Info;
  * \brief Main function
  */
 
-int main(CALuint argc, CALchar** argv)
+int main(CALint argc, CALchar** argv)
 {
 	//////////////////////////////////////////////////////////////////////////
     //
--- atical/samples/tutorial/OpenCloseDevice/OpenCloseDevice.cpp_orig	2009-09-07 13:39:55.000000000 +0200
+++ atical/samples/tutorial/OpenCloseDevice/OpenCloseDevice.cpp	2009-09-07 13:40:16.000000000 +0200
@@ -110,6 +110,7 @@
 #include "Samples.h"
 #include <assert.h>
 #include <string>
+#include <cstdio>
 
 /**
  * \var Info
@@ -210,7 +211,7 @@
 }
 
 int 
-main(CALuint argc, CALchar** argv)
+main(CALint argc, CALchar** argv)
 {   
 	//////////////////////////////////////////////////////////////////////////
     //
