commit 51c2d9bd8ec65896adf4a3226b816bbbceb8fd3b
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Fri Apr 15 10:58:23 2011 -0400

    added gcc-4.6.0-bionic.patch from asac

diff --git a/libiberty/configure b/libiberty/configure
index bdabe8d..b2a4723 100755
--- a/libiberty/configure
+++ b/libiberty/configure
@@ -673,6 +673,7 @@ with_target_subdir
 with_build_subdir
 with_cross_host
 with_newlib
+with_bionic
 enable_maintainer_mode
 enable_multilib
 enable_largefile
@@ -1313,6 +1314,7 @@ Optional Packages:
   --with-build-subdir=SUBDIR       Configuring in a subdirectory for build
   --with-cross-host=HOST           Configuring with a cross compiler
   --with-newlib                    Configuring with newlib
+  --with-bionic                    Configuring with Android Bionic
 
 Some influential environment variables:
   CC          C compiler command
@@ -2329,6 +2331,12 @@ if test "${with_newlib+set}" = set; then :
 fi
 
 
+# Check whether --with-bionic was given.
+if test "${with_bionic+set}" = set; then :
+  withval=$with_bionic;
+fi
+
+
 if test "${srcdir}" = "."; then
   if test -n "${with_build_subdir}"; then
     libiberty_topdir="${srcdir}/../.."
@@ -5483,6 +5491,23 @@ _ACEOF
 
   fi
 
+  if test "x${with_bionic}" = "xyes"; then
+
+    for f in $funcs; do
+      case "$f" in
+	*)
+	  n=HAVE_`echo $f | tr 'abcdefghijklmnopqrstuvwxyz' 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'`
+	  cat >>confdefs.h <<_ACEOF
+#define $n 1
+_ACEOF
+
+	  ;;
+      esac
+    done
+
+    setobjs=yes
+  fi
+
   # If we are being configured for Mingw, we know which functions
   # Mingw provides and which ones we will be expected to provide.
 
@@ -6290,6 +6315,7 @@ done
   for v in $vars; do
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $v" >&5
 $as_echo_n "checking for $v... " >&6; }
+    if test x$gcc_no_link != xyes; then
     if { as_var=libiberty_cv_var_$v; eval "test \"\${$as_var+set}\" = set"; }; then :
   $as_echo_n "(cached) " >&6
 else
@@ -6310,12 +6336,15 @@ _ACEOF
 if ac_fn_c_try_link "$LINENO"; then :
   eval "libiberty_cv_var_$v=yes"
 else
-  eval "libiberty_cv_var_$v=no"
+  eval "libiberty_cv_var_$v=$gcc_no_link"
 fi
 rm -f core conftest.err conftest.$ac_objext \
     conftest$ac_exeext conftest.$ac_ext
 fi
 
+    else
+      eval "libiberty_cv_var_$v=yes"
+    fi
     if eval "test \"`echo '$libiberty_cv_var_'$v`\" = yes"; then
       { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
diff --git a/libiberty/configure.ac b/libiberty/configure.ac
index 9f1ff04..f4d15cf 100644
--- a/libiberty/configure.ac
+++ b/libiberty/configure.ac
@@ -19,6 +19,8 @@ AC_ARG_WITH(cross-host,
 [  --with-cross-host=HOST           Configuring with a cross compiler])
 AC_ARG_WITH(newlib,
 [  --with-newlib                    Configuring with newlib])
+AC_ARG_WITH(bionic,
+[  --with-bionic                    Configuring with Android Bionic])
 
 if test "${srcdir}" = "."; then
   if test -n "${with_build_subdir}"; then
@@ -436,6 +438,20 @@ if test -n "${with_target_subdir}"; then
 
   fi
 
+  if test "x${with_bionic}" = "xyes"; then
+
+    for f in $funcs; do
+      case "$f" in
+	*)
+	  n=HAVE_`echo $f | tr 'abcdefghijklmnopqrstuvwxyz' 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'`
+	  AC_DEFINE_UNQUOTED($n)
+	  ;;
+      esac
+    done
+
+    setobjs=yes
+  fi
+
   # If we are being configured for Mingw, we know which functions
   # Mingw provides and which ones we will be expected to provide.
 
@@ -644,10 +660,14 @@ if test -z "${setobjs}"; then
 
   for v in $vars; do
     AC_MSG_CHECKING([for $v])
+    if test x$gcc_no_link != xyes; then
     AC_CACHE_VAL(libiberty_cv_var_$v,
       [AC_LINK_IFELSE([AC_LANG_PROGRAM([[int *p;]],[[extern int $v []; p = $v;]])],
 		      [eval "libiberty_cv_var_$v=yes"],
-		      [eval "libiberty_cv_var_$v=no"])])
+		      [eval "libiberty_cv_var_$v=$gcc_no_link"])])
+    else
+      eval "libiberty_cv_var_$v=yes"
+    fi
     if eval "test \"`echo '$libiberty_cv_var_'$v`\" = yes"; then
       AC_MSG_RESULT(yes)
       n=HAVE_`echo $v | tr 'abcdefghijklmnopqrstuvwxyz' 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'`
diff --git a/libquadmath/configure b/libquadmath/configure
index 8735ac5..7822cfc 100755
--- a/libquadmath/configure
+++ b/libquadmath/configure
@@ -11891,6 +11891,7 @@ ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
 ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
 ac_compiler_gnu=$ac_cv_c_compiler_gnu
 
+fi
 
 # Calculate toolexeclibdir
 # Also toolexecdir, though it's only used in toolexeclibdir
@@ -12335,6 +12336,7 @@ $as_echo "#define HAVE_HIDDEN_VISIBILITY 1" >>confdefs.h
 
 fi
 CFLAGS="$saved_CFLAGS"
+fi
 
 # Check for symbol versioning (copied from libssp).
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether symbol versioning is supported" >&5
diff --git a/libstdc++-v3/configure b/libstdc++-v3/configure
index 4e1661a..755bae2 100755
--- a/libstdc++-v3/configure
+++ b/libstdc++-v3/configure
@@ -5271,7 +5271,7 @@ $as_echo "$as_me: OS config directory is $os_include_dir" >&6;}
 
 
 
-if test "x${with_newlib}" != "xyes"; then
+if test "x${with_newlib}" != "xyes" -a "x${bionic}" != "xyes"; then
   enable_dlopen=yes
 
 
@@ -19542,7 +19542,7 @@ done
 
 
 # Only do link tests if native. Else, hardcode.
-if $GLIBCXX_IS_NATIVE; then
+if test x$GLIBCXX_IS_NATIVE = xtrue; then
 
   # We can do more elaborate tests that assume a working linker.
   CANADIAN=no
@@ -26428,7 +26428,7 @@ else
 
   # First, test for "known" system libraries.  We may be using newlib even
   # on a hosted environment.
-  if test "x${with_newlib}" = "xyes"; then
+  if test "x${with_newlib}" = "xyes" -o "x${bionic}" = "xyes"; then
     os_include_dir="os/newlib"
     $as_echo "#define HAVE_HYPOT 1" >>confdefs.h
 
@@ -63504,7 +63504,8 @@ $as_echo "#define HAVE_GETIPINFO 1" >>confdefs.h
 
 
 
- # Check whether --enable-linux-futex was given.
+if test x$GLIBCXX_IS_NATIVE = xtrue; then
+   # Check whether --enable-linux-futex was given.
 if test "${enable_linux_futex+set}" = set; then :
   enableval=$enable_linux_futex;
       case "$enableval" in
@@ -63621,6 +63622,7 @@ $as_echo "#define HAVE_LINUX_FUTEX 1" >>confdefs.h
 
 fi
 
+fi
 
 
 
diff --git a/libstdc++-v3/configure.ac b/libstdc++-v3/configure.ac
index 128b9c5..454484c 100644
--- a/libstdc++-v3/configure.ac
+++ b/libstdc++-v3/configure.ac
@@ -1,4 +1,4 @@
-# Process this file with autoreconf to produce a configure script.
+ Process this file with autoreconf to produce a configure script.
 
 AC_PREREQ(2.64)
 AC_INIT(package-unused, version-unused,, libstdc++)
@@ -92,7 +92,7 @@ CXXFLAGS="$save_CXXFLAGS"
 # up critical shell variables.
 GLIBCXX_CONFIGURE
 
-if test "x${with_newlib}" != "xyes"; then
+if test "x${with_newlib}" != "xyes" -a "x${bionic}" != "xyes"; then
   AC_LIBTOOL_DLOPEN
 fi
 AM_PROG_LIBTOOL
@@ -178,7 +178,7 @@ sys/resource.h sys/sem.h sys/stat.h sys/time.h sys/types.h unistd.h \
 wchar.h wctype.h])
 
 # Only do link tests if native. Else, hardcode.
-if $GLIBCXX_IS_NATIVE; then
+if test x$GLIBCXX_IS_NATIVE = xtrue; then
 
   # We can do more elaborate tests that assume a working linker.
   CANADIAN=no
@@ -228,7 +228,7 @@ else
 
   # First, test for "known" system libraries.  We may be using newlib even
   # on a hosted environment.
-  if test "x${with_newlib}" = "xyes"; then
+  if test "x${with_newlib}" = "xyes" -o "x${bionic}" = "xyes"; then
     os_include_dir="os/newlib"
     AC_DEFINE(HAVE_HYPOT)
 
@@ -297,7 +297,9 @@ fi
 GCC_CHECK_UNWIND_GETIPINFO
 
 
-GCC_LINUX_FUTEX([AC_DEFINE(HAVE_LINUX_FUTEX, 1, [Define if futex syscall is available.])])
+if test x$GLIBCXX_IS_NATIVE = xtrue; then
+  GCC_LINUX_FUTEX([AC_DEFINE(HAVE_LINUX_FUTEX, 1, [Define if futex syscall is available.])])
+fi
 
 GCC_HEADER_STDINT(include/gstdint.h)
 
