commit af7713f7f93791ff14afbcfe9c62da2336a930d5
Author: root <root@localhost.(none)>
Date:   Sat Apr 16 18:29:57 2011 -0400

    compile against bionic headers

diff --git a/libiberty/getcwd.c b/libiberty/getcwd.c
index 28f26eb..3bb9219 100644
--- a/libiberty/getcwd.c
+++ b/libiberty/getcwd.c
@@ -31,7 +31,8 @@ directory's path doesn't fit in @var{len} characters, the result is
 #endif
 
 extern char *getwd ();
-extern int errno;
+// XXX: in Bionic: #define  errno   (*__errno())
+// extern int errno;
 
 #ifndef MAXPATHLEN
 #define MAXPATHLEN 1024
diff --git a/libiberty/getpagesize.c b/libiberty/getpagesize.c
index 1c3a263..a673163 100644
--- a/libiberty/getpagesize.c
+++ b/libiberty/getpagesize.c
@@ -63,7 +63,8 @@ BUGS
 int
 getpagesize (void)
 {
-  return (GNU_OUR_PAGESIZE);
+// XXX: filthy hack
+  return (4096);
 }
 
 #else /* VMS */
diff --git a/libiberty/gettimeofday.c b/libiberty/gettimeofday.c
index fca1679..1fad5ae 100644
--- a/libiberty/gettimeofday.c
+++ b/libiberty/gettimeofday.c
@@ -19,10 +19,10 @@ that @var{tz} be NULL.  Returns 0 on success, -1 on failure.
 */ 
 
 int
-gettimeofday (struct timeval *tp, void *tz)
+gettimeofday (struct timeval *tp, struct timezone *tz)
 {
-  if (tz)
-    abort ();
+  (void)tz; /* unused */
+
   tp->tv_usec = 0;
   if (time (&tp->tv_sec) == (time_t) -1)
     return -1;
diff --git a/libiberty/setproctitle.c b/libiberty/setproctitle.c
index 3b3f377..42d55c1 100644
--- a/libiberty/setproctitle.c
+++ b/libiberty/setproctitle.c
@@ -43,6 +43,7 @@ setproctitle (const char *name ATTRIBUTE_UNUSED, ...)
 #ifdef PR_SET_NAME
   /* On Linux this sets the top visible "comm", but not necessarily
      the name visible in ps. */
-  prctl (PR_SET_NAME, name);
+  unsigned long arg3=0, arg4=0, arg5=0;
+  prctl (PR_SET_NAME, (unsigned long)name,arg3,arg4,arg5);
 #endif
 }
