commit a0ba19add4a7a76ad008c9c729d0b610e9ef8b78
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Mon May 9 12:53:30 2011 -0400

    c++ without the libstd

diff --git a/configure b/configure
index ba45977..d066eda 100755
--- a/configure
+++ b/configure
@@ -3172,6 +3172,10 @@ case "${target}" in
 	;;
     esac
     ;;
+  *-*-android*|*-*-bionic*)
+    # Android has its own libstdc++, but does not supply the headers
+    noconfigdirs="$noconfigdirs target-libstdc++-v3"
+    ;;
   *-*-kaos*)
     # Remove unsupported stuff on all kaOS configurations.
     skipdirs="target-libiberty ${libgcj} target-libstdc++-v3 target-librx"
diff --git a/configure.ac b/configure.ac
index 6b17b68..4d37d0a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -593,6 +593,10 @@ case "${target}" in
 	;;
     esac
     ;;
+  *-*-android*|*-*-bionic*)
+    # Android has its own libstdc++, but does not supply the headers
+    noconfigdirs="$noconfigdirs target-libstdc++-v3"
+    ;;
   *-*-kaos*)
     # Remove unsupported stuff on all kaOS configurations.
     skipdirs="target-libiberty ${libgcj} target-libstdc++-v3 target-librx"
