commit a28f928505a5ead8cec0f8425f82836db73ca891
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Tue May 10 04:11:37 2011 -0400

    need a working size_t for libstdc++

diff --git a/gcc/config/i386/linux.h b/gcc/config/i386/linux.h
index 6cc0eca..0ceb4d7 100644
--- a/gcc/config/i386/linux.h
+++ b/gcc/config/i386/linux.h
@@ -58,7 +58,7 @@ along with GCC; see the file COPYING3.  If not see
 #define SUBTARGET_FRAME_POINTER_REQUIRED crtl->profile
 
 #undef SIZE_TYPE
-#define SIZE_TYPE "unsigned int"
+#define SIZE_TYPE "long unsigned int"
  
 #undef PTRDIFF_TYPE
 #define PTRDIFF_TYPE "int"
