--- ssdb-1.9.4.orig/ssdb.conf	2017-03-29 17:28:59.000000000 +0800
+++ ssdb-1.9.4/ssdb.conf	2017-06-13 00:01:41.519799034 +0800
@@ -2,8 +2,8 @@
 # MUST indent by TAB!
 
 # relative to path of this file, directory must exists
-work_dir = ./var
-pidfile = ./var/ssdb.pid
+work_dir = /var/lib/ssdb
+pidfile = /run/ssdb/ssdb.pid
 
 server:
 	ip: 127.0.0.1
@@ -34,7 +34,7 @@
 
 logger:
 	level: debug
-	output: log.txt
+	output: /var/log/ssdb/ssdb.log
 	rotate:
 		size: 1000000000
 
