--- sbcl-2.4.3.orig/make-config.sh	2024-03-29 10:02:12.000000000 -0700
+++ sbcl-2.4.3/make-config.sh	2024-03-30 01:03:22.698303351 -0700
@@ -809,7 +809,7 @@
 if [ -n "$SOURCE_DATE_EPOCH" ]; then
   echo '"'hostname-id-"$SOURCE_DATE_EPOCH"'"' > output/build-id.inc
 else
-  echo '"'`hostname`-`id -un`-`date +%Y-%m-%d-%H-%M-%S`'"' > output/build-id.inc
+  echo "\"${PV}\"" > output/build-id.inc
 fi
 
 if [ -n "$SBCL_HOST_LOCATION" ]; then
