--- a/Makefile
+++ b/Makefile
@@ -1,55 +1,77 @@
-JQ = jq
-TREE_SITTER = node_modules/.bin/tree-sitter
+ifeq ($(OS),Windows_NT)
+$(error Windows is not supported)
+endif
+
+LANGUAGE_NAME := tree-sitter-qmljs
+HOMEPAGE_URL := https://github.com/yuja/tree-sitter-qmljs
+VERSION := 0.3.1
+
+# repository
+SRC_DIR := src
+
+TS ?= tree-sitter
+
+# install directory layout
+PREFIX ?= /usr/local
+DATADIR ?= $(PREFIX)/share
+INCLUDEDIR ?= $(PREFIX)/include
+LIBDIR ?= $(PREFIX)/lib
+
+# source/object files
+PARSER := $(SRC_DIR)/parser.c
+EXTRAS := $(filter-out $(PARSER),$(wildcard $(SRC_DIR)/*.c))
+OBJS := $(patsubst %.c,%.o,$(PARSER) $(EXTRAS))
+
+# flags
+override CFLAGS += -I$(SRC_DIR) -std=c11 -fPIC
+
+# ABI versioning
+SONAME_MAJOR = $(shell sed -n 's/\#define LANGUAGE_VERSION //p' $(PARSER))
+SONAME_MINOR = $(word 1,$(subst ., ,$(VERSION)))
+
+# OS-specific bits
+ifeq ($(shell uname),Darwin)
+	SOEXT = dylib
+	SOEXTVER_MAJOR = $(SONAME_MAJOR).$(SOEXT)
+	SOEXTVER = $(SONAME_MAJOR).$(SONAME_MINOR).$(SOEXT)
+	LINKSHARED = -dynamiclib -Wl,-install_name,$(LIBDIR)/lib$(LANGUAGE_NAME).$(SOEXTVER),-rpath,@executable_path/../Frameworks
+else
+	SOEXT = so
+	SOEXTVER_MAJOR = $(SOEXT).$(SONAME_MAJOR)
+	SOEXTVER = $(SOEXT).$(SONAME_MAJOR).$(SONAME_MINOR)
+	LINKSHARED = -shared -Wl,-soname,lib$(LANGUAGE_NAME).$(SOEXTVER)
+endif
+
+all: lib$(LANGUAGE_NAME).$(SOEXT)
+
+lib$(LANGUAGE_NAME).$(SOEXT): $(OBJS)
+	$(CC) $(LDFLAGS) $(LINKSHARED) $^ $(LDLIBS) -o $@
+ifneq ($(STRIP),)
+	$(STRIP) $@
+endif
+
+$(PARSER): $(SRC_DIR)/grammar.json
+	$(TS) generate $^
+
+install: all
+	install -d '$(DESTDIR)$(DATADIR)'/tree-sitter/queries/qmljs '$(DESTDIR)$(INCLUDEDIR)' '$(DESTDIR)$(LIBDIR)'
+	install -m644 bindings/c/$(LANGUAGE_NAME).h '$(DESTDIR)$(INCLUDEDIR)'/$(LANGUAGE_NAME).h
+	install -m755 lib$(LANGUAGE_NAME).$(SOEXT) '$(DESTDIR)$(LIBDIR)'/lib$(LANGUAGE_NAME).$(SOEXTVER)
+	ln -sf lib$(LANGUAGE_NAME).$(SOEXTVER) '$(DESTDIR)$(LIBDIR)'/lib$(LANGUAGE_NAME).$(SOEXTVER_MAJOR)
+	ln -sf lib$(LANGUAGE_NAME).$(SOEXTVER_MAJOR) '$(DESTDIR)$(LIBDIR)'/lib$(LANGUAGE_NAME).$(SOEXT)
+	install -m644 queries/*.scm '$(DESTDIR)$(DATADIR)'/tree-sitter/queries/qmljs
+
+uninstall:
+	$(RM) -r '$(DESTDIR)$(LIBDIR)'/lib$(LANGUAGE_NAME).$(SOEXTVER) \
+		'$(DESTDIR)$(LIBDIR)'/lib$(LANGUAGE_NAME).$(SOEXTVER_MAJOR) \
+		'$(DESTDIR)$(LIBDIR)'/lib$(LANGUAGE_NAME).$(SOEXT) \
+		'$(DESTDIR)$(INCLUDEDIR)'/$(LANGUAGE_NAME).h \
+		'$(DESTDIR)$(DATADIR)'/tree-sitter/queries/qmljs
 
-.PHONY: help
-help:
-	@echo 'Make targets:'
-	@echo '  generate   - generate parser sources'
-	@echo '  clean      - remove build directory and intermediate files'
-	@echo '  tests      - run all tests'
-	@echo '  fetch-examples - fetch example repositories'
-	@echo '  parse-examples - parse example files'
-
-.PHONY: generate
-generate:
-	$(MAKE) src/typescript-scanner.h
-	$(TREE_SITTER) generate
-
-src/typescript-scanner.h: \
- node_modules/tree-sitter-typescript/common/scanner.h \
- node_modules/tree-sitter-typescript/LICENSE \
- node_modules/tree-sitter-typescript/package.json
-	( \
-		echo '/*'; \
-		echo 'Source:'; \
-		$(JQ) -r '.version' node_modules/tree-sitter-typescript/package.json; \
-		echo; \
-		cat node_modules/tree-sitter-typescript/LICENSE; \
-		echo '*/'; \
-		echo; \
-		cat $<; \
-	) > $@
-
-.PHONY: clean
 clean:
-	$(RM) -R build target
-	$(RM) \
-		src/grammar.json \
-		src/node-types.json \
-		src/parser.c \
-		src/tree_sitter/parser.h \
-		src/typescript-scanner.h
-
-.PHONY: tests
-tests:
-	$(TREE_SITTER) test $(TESTFLAGS)
-
-.PHONY: fetch-examples
-fetch-examples:
-	$(MAKE) -C examples all
-
-.PHONY: parse-examples
-parse-examples:
-	cat examples/known-failures.txt \
-		| sed 's|^|!examples/|' \
-		| xargs $(TREE_SITTER) parse -q 'examples/**/*.qml'
+	$(RM) $(OBJS) lib$(LANGUAGE_NAME).$(SOEXT)
+
+test:
+	$(TS) test
+
+.PHONY: all install uninstall clean test
