diff -Nur a/csrc/cpu/CMakeLists.txt b/csrc/cpu/CMakeLists.txt
--- a/csrc/cpu/CMakeLists.txt	2025-01-18 23:36:03.310045123 -0800
+++ b/csrc/cpu/CMakeLists.txt	2025-01-19 14:05:53.468546425 -0800
@@ -159,12 +159,18 @@
 endif()
 
 if(USE_LIBXSMM)
-  if(BUILD_LIBXSMM_VIA_CMAKE)
+  find_package(PkgConfig REQUIRED)
+  pkg_search_module(XSMM REQUIRED libxsmm)
+  if(XSMM_FOUND)
+    add_definitions(-DLIBXSMM_DEFAULT_CONFIG)
+    target_include_directories(${PLUGIN_NAME_CPU} PUBLIC ${XSMM_INCLUDE_DIRS})
+    target_link_libraries(${PLUGIN_NAME_CPU} PRIVATE ${XSMM_LIBRARIES})
+  elseif(BUILD_LIBXSMM_VIA_CMAKE)
     add_subdirectory(${IPEX_CPU_CPP_THIRD_PARTY_ROOT}/libxsmm cpu_third_party/libxsmm EXCLUDE_FROM_ALL)
     add_definitions(-DLIBXSMM_DEFAULT_CONFIG)
     target_include_directories(${PLUGIN_NAME_CPU} PUBLIC ${IPEX_CPU_CPP_THIRD_PARTY_ROOT}/libxsmm/include)
     target_link_libraries(${PLUGIN_NAME_CPU} PRIVATE xsmm)
-  else(BUILD_LIBXSMM_VIA_CMAKE)
+  else()
     include(${CMAKE_ROOT}/Modules/ExternalProject.cmake)
     set(args
     CC=${CMAKE_C_COMPILER}
@@ -183,7 +189,7 @@
       INSTALL_COMMAND ""
       )
     target_link_libraries(${PLUGIN_NAME_CPU} PRIVATE ${IPEX_CPU_CPP_THIRD_PARTY_ROOT}/libxsmm/lib/libxsmm.a)
-  endif(BUILD_LIBXSMM_VIA_CMAKE)
+  endif()
 endif(USE_LIBXSMM)
 
 # setup sleef options:
