--- build/custom.mk.orig        2005-12-12 20:40:40.125949232 +0100
+++ build/custom.mk     2005-12-12 20:42:16.972226360 +0100
@@ -7,15 +7,15 @@
 # Catapult is always installed into a single self-contained directory.
 # But you can change that directory to for example /usr/local/openMSX-Catapult
 # or /usr/games/openMSX-Catapult if you like.
-INSTALL_BASE:=/opt/openMSX-Catapult
+INSTALL_BASE:=/usr/games/openmsx

 # Create a symbolic link to the installed binary?
 # This link is placed in a location that is typically in a user's path:
 # /usr/local/bin for system-wide installs and ~/bin for personal installs.
-SYMLINK_FOR_BINARY:=true
+SYMLINK_FOR_BINARY:=false

 # Locations for openMSX binaries and share directory
 # these are suggested as initial values when you run catapult
 # for the first time
-CATAPULT_OPENMSX_BINARY:=/opt/openMSX/bin/openmsx
-CATAPULT_OPENMSX_SHARE:=/opt/openMSX/share
+CATAPULT_OPENMSX_BINARY:=/usr/games/openmsx/bin
+CATAPULT_OPENMSX_SHARE:=/usr/games/openmsx/share
