diff -Nur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -474,7 +474,7 @@
 add_library(tvm_objs OBJECT ${COMPILER_SRCS})
 add_library(tvm_runtime_objs OBJECT ${RUNTIME_SRCS})
-target_link_libraries(tvm_objs PUBLIC tvm_ffi_header)
-target_link_libraries(tvm_runtime_objs PUBLIC tvm_ffi_header)
+target_link_libraries(tvm_objs PUBLIC tvm_ffi::header)
+target_link_libraries(tvm_runtime_objs PUBLIC tvm_ffi::header)

 include(GNUInstallDirs)

@@ -493,12 +493,12 @@
   string(CONCAT NOTICE ${NOTICE_MULTILINE})
   add_custom_command(TARGET tvm_runtime POST_BUILD
     COMMAND ${CMAKE_COMMAND} -E cmake_echo_color --yellow --bold ${NOTICE})
-  target_link_libraries(tvm_runtime PUBLIC tvm_ffi_static)
+  target_link_libraries(tvm_runtime PUBLIC tvm_ffi::static)
 else()
   add_library(tvm_runtime SHARED
     $<TARGET_OBJECTS:tvm_runtime_objs>
     ${TVM_RUNTIME_EXT_OBJS}
   )
   set_property(TARGET tvm_runtime APPEND PROPERTY LINK_OPTIONS "${TVM_NO_UNDEFINED_SYMBOLS}")
-  target_link_libraries(tvm_runtime PUBLIC tvm_ffi_shared)
+  target_link_libraries(tvm_runtime PUBLIC tvm_ffi::shared)
 endif()
@@ -520,8 +520,8 @@
 # RTLD_LOCAL so compiler-internal symbols don't leak into the global namespace.
 add_library(tvm_compiler SHARED
   $<TARGET_OBJECTS:tvm_objs>
 )
-target_link_libraries(tvm_compiler PUBLIC tvm_runtime tvm_ffi_shared)
+target_link_libraries(tvm_compiler PUBLIC tvm_runtime tvm_ffi::shared)
 target_include_directories(tvm_compiler PUBLIC "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>")
 set_property(TARGET tvm_compiler APPEND PROPERTY LINK_OPTIONS "${TVM_NO_UNDEFINED_SYMBOLS}")
 set_property(TARGET tvm_compiler APPEND PROPERTY LINK_OPTIONS "${TVM_VISIBILITY_FLAG}")
@@ -570,6 +570,7 @@
   add_subdirectory("apps/ios_rpc")
 endif()

-add_subdirectory(3rdparty/tvm-ffi)
+#add_subdirectory(3rdparty/tvm-ffi)
+find_package(tvm_ffi REQUIRED)

 if(TVM_DEBUG_WITH_ABI_CHANGE)
