diff -NuarpbB yenc-ori/setup.py yenc/setup.py
--- yenc-ori/setup.py	2011-10-24 03:42:56.000000000 +0200
+++ yenc/setup.py	2011-10-24 03:48:39.000000000 +0200
@@ -33,7 +33,7 @@ setup(
         platforms        = ["Unix"],
 	package_dir	 = { '': 'lib' },
 	py_modules	 = ["yenc"],
-	ext_modules	 = [Extension("_yenc",["src/_yenc.c"],extra_compile_args=["-O2","-g"])],
+	ext_modules	 = [Extension("_yenc",["src/_yenc.c"])],
         classifiers      = [
             "Programming Language :: Python",
             "Programming Language :: Python :: 2.5",
