diff -Nur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2026-02-22 08:15:39.000000000 -0800
+++ b/CMakeLists.txt	2026-03-22 14:01:24.998645700 -0700
@@ -53,7 +53,7 @@
                            $<INSTALL_INTERFACE:include>
 )
 target_include_directories(
-  tvm_ffi_header INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/dlpack/include>
+  tvm_ffi_header INTERFACE
                            $<INSTALL_INTERFACE:include>
 )
 
@@ -303,31 +303,32 @@
   install(TARGETS tvm_ffi_cython DESTINATION .)
 
   # ######### Installing the source ##########
-  install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/dlpack/include/
-          DESTINATION 3rdparty/dlpack/include/
-  )
-  install(
-    DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/libbacktrace/
-    DESTINATION 3rdparty/libbacktrace/
-    PATTERN ".git" EXCLUDE
-    PATTERN ".git*" EXCLUDE
-    PATTERN "*.tmp" EXCLUDE
-  )
-  install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/src/ffi/ DESTINATION src/ffi/)
-  install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/cmake/Utils/ DESTINATION share/cmake/tvm_ffi/Utils)
-  install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/CMakeLists.txt DESTINATION .)
-  install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/cmake/tvm_ffi-config.cmake
-          DESTINATION share/cmake/tvm_ffi
-  )
-  install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/python/tvm_ffi/cython/tvm_ffi_python_helpers.h
-          DESTINATION include/
-  )
+  #install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/dlpack/include/
+  #        DESTINATION 3rdparty/dlpack/include/
+  #)
+  #install(
+  #  DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/libbacktrace/
+  #  DESTINATION 3rdparty/libbacktrace/
+  #  PATTERN ".git" EXCLUDE
+  #  PATTERN ".git*" EXCLUDE
+  #  PATTERN "*.tmp" EXCLUDE
+  #)
+  #install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/src/ffi/ DESTINATION src/ffi/)
+  #install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/cmake/Utils/ DESTINATION share/cmake/tvm_ffi/Utils)
+  #install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/CMakeLists.txt DESTINATION .)
+  #install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/cmake/tvm_ffi-config.cmake
+  #        DESTINATION share/cmake/tvm_ffi
+  #)
+  #install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/python/tvm_ffi/cython/tvm_ffi_python_helpers.h
+  #        DESTINATION include/
+  #)
 endif ()
 
 # ######### Install the related for normal cmake library ##########
 
+if (NOT TVM_FFI_BUILD_PYTHON_MODULE)
 install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/include/tvm/ffi/ DESTINATION include/tvm/ffi/)
-install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/dlpack/include/ DESTINATION include/)
+#install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/dlpack/include/ DESTINATION include/)
 install(TARGETS tvm_ffi_shared DESTINATION lib)
 # if tvm_ffi_testing is built, we also install it
 if (TARGET tvm_ffi_testing)
@@ -337,6 +338,7 @@
     OPTIONAL
   )
 endif ()
+endif ()
 # ship additional dSYM files for debugging symbols on if available
 if (APPLE)
   install(
diff -Nur a/python/tvm_ffi/libinfo.py b/python/tvm_ffi/libinfo.py
--- a/python/tvm_ffi/libinfo.py	2026-02-22 08:15:39.000000000 -0800
+++ b/python/tvm_ffi/libinfo.py	2026-03-22 13:59:40.921975369 -0700
@@ -102,7 +102,7 @@
     if ret := _resolve_and_validate(
         paths=[
             _rel_top_directory() / "include",
-            _dev_top_directory() / "3rdparty" / "dlpack" / "include",
+            Path("/usr/include"),
         ],
         cond=lambda p: (p / "dlpack").is_dir(),
     ):
@@ -253,6 +253,8 @@
     dll_paths.append(_rel_top_directory() / "lib")
     dll_paths.append(_dev_top_directory() / "build" / "lib")
     dll_paths.append(_dev_top_directory() / "lib")
+    dll_paths.append(Path("/usr/lib64"))
+    dll_paths.append(Path("/usr/lib"))
 
     # Case 2. It is specified in PATH-related environment variables
     if sys.platform.startswith("win32"):
