diff -ur colorize-1.0.orig/clhs-lookup.lisp colorize-1.0/clhs-lookup.lisp
--- colorize-1.0.orig/clhs-lookup.lisp	2004-11-12 09:03:07.000000000 -0600
+++ colorize-1.0/clhs-lookup.lisp	2005-07-19 09:51:51.000000000 -0500
@@ -4,9 +4,7 @@
 (in-package :clhs-lookup)
 
 (defparameter *hyperspec-pathname*
-  (merge-pathnames
-   (make-pathname :directory '(:relative "HyperSpec"))
-   (user-homedir-pathname)))
+  #p"/usr/share/doc/hyperspec-7.0/HyperSpec/")
 
 (defparameter *hyperspec-map-file*
   (merge-pathnames "Data/Map_Sym.txt" *hyperspec-pathname*))
Only in colorize-1.0: clhs-lookup.lisp~
