diff -ur terraform-0.9.0/Makefile.in terraform-0.9.0-patch/Makefile.in
--- terraform-0.9.0/Makefile.in	2002-04-24 18:16:48.000000000 +0200
+++ terraform-0.9.0-patch/Makefile.in	2006-03-07 17:21:27.000000000 +0100
@@ -124,7 +124,7 @@
 cflags_set = @cflags_set@
 cxxflags_set = @cxxflags_set@
 
-SUBDIRS = intl po macros src data desktop-links docs help
+SUBDIRS = intl po macros src data docs help
 
 HOME_INSTALL_DIRS = data src 
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
