diff -Naur como-1.0-orig/base/como.gmk como-1.0/base/como.gmk
--- como-1.0-orig/base/como.gmk	2006-05-10 21:41:43.000000000 +0000
+++ como-1.0/base/como.gmk	2006-10-06 22:40:12.000000000 +0000
@@ -121,11 +121,11 @@
 x=$(EXTRAS:%=$(EXTRADIR)/%)
 SRCS_como += $(x)
 
-capture.o:	build-sniffer-list
+capture.o:	sniffer-list.h
 
 TEMPORARY += sniffer-list.h
 
-build-sniffer-list:
+sniffer-list.h:
 	@( \
 	echo "/* generated file, do not edit */" ;		\
 	for i in $(SNIFFERS); do  				\
