--- como-1.0-orig/como.gmk	2006-05-14 21:29:30.000000000 +0000
+++ como-1.0/como.gmk	2006-10-06 23:23:55.000000000 +0000
@@ -58,12 +58,11 @@
 INSTALL=/usr/bin/install
 
 # Install dirs
-PREFIX?=/usr/local
-INST_BINDIR?=$(DESTDIR)$(PREFIX)/bin
-INST_LIBDIR?=$(DESTDIR)$(PREFIX)/lib/como-$(COMOVER)
-INST_DATADIR?=$(DESTDIR)$(PREFIX)/var/lib/como-$(COMOVER)
-INST_MANDIR?=$(DESTDIR)$(PREFIX)/share/man/man1
-DEFAULT_CFGFILE?=$(PREFIX)/etc/como/como.conf
+INST_BINDIR?=$(DESTDIR)/bin
+INST_LIBDIR?=$(DESTDIR)/lib/como-$(COMOVER)
+INST_DATADIR?=$(DESTDIR)/var/lib/como-$(COMOVER)
+INST_MANDIR?=$(DESTDIR)/share/man/man1
+DEFAULT_CFGFILE?=/etc/como/como.conf
 
 # #define's used to build this version of the code.
 COMOVER = 1.0
