--- a/src/shared/linux_osl.c	2015-09-18 15:47:30.000000000 -0700
+++ b/src/shared/linux_osl.c	2022-06-17 00:55:54.624575147 -0700
@@ -600,7 +600,7 @@
 	if (va)
 		*pap = (ulong)__virt_to_phys(va);
 #else
-	va = pci_alloc_consistent(osh->pdev, size, (dma_addr_t*)pap);
+	va = dma_alloc_coherent(osh->pdev == NULL ? NULL : &((struct pci_dev *)osh->pdev)->dev, size, (dma_addr_t *)pap, GFP_ATOMIC);
 #endif
 	return va;
 }
@@ -613,7 +613,7 @@
 #ifdef __ARM_ARCH_7A__
 	kfree(va);
 #else
-	pci_free_consistent(osh->pdev, size, va, (dma_addr_t)pa);
+        dma_free_coherent(&((struct pci_dev *)osh->pdev)->dev, size, va, (dma_addr_t)pa);
 #endif
 }
 
@@ -623,7 +623,7 @@
 	int dir;
 
 	ASSERT((osh && (osh->magic == OS_HANDLE_MAGIC)));
-	dir = (direction == DMA_TX)? PCI_DMA_TODEVICE: PCI_DMA_FROMDEVICE;
+	dir = (direction == DMA_TX)? DMA_TO_DEVICE: DMA_FROM_DEVICE;
 
 #if defined(__ARM_ARCH_7A__) && defined(BCMDMASGLISTOSL)
 	if (dmah != NULL) {
@@ -641,7 +641,7 @@
 				ASSERT(totsegs + nsegs <= MAX_DMA_SEGS);
 				sg->page_link = 0;
 				sg_set_buf(sg, PKTDATA(osh, skb), PKTLEN(osh, skb));
-				pci_map_single(osh->pdev, PKTDATA(osh, skb), PKTLEN(osh, skb), dir);
+				dma_map_single(&((struct pci_dev *)osh->pdev)->dev, PKTDATA(osh, skb), PKTLEN(osh, skb), (enum dma_data_direction)dir);
 			}
 			totsegs += nsegs;
 			totlen += PKTLEN(osh, skb);
@@ -656,7 +656,7 @@
 	}
 #endif 
 
-	return (pci_map_single(osh->pdev, va, size, dir));
+	return (dma_map_single(&((struct pci_dev *)osh->pdev)->dev, va, size, dir));
 }
 
 void BCMFASTPATH
@@ -665,8 +665,8 @@
 	int dir;
 
 	ASSERT((osh && (osh->magic == OS_HANDLE_MAGIC)));
-	dir = (direction == DMA_TX)? PCI_DMA_TODEVICE: PCI_DMA_FROMDEVICE;
-	pci_unmap_single(osh->pdev, (uint32)pa, size, dir);
+	dir = (direction == DMA_TX)? DMA_TO_DEVICE: DMA_FROM_DEVICE;
+	dma_unmap_single(&((struct pci_dev *)osh->pdev)->dev, (uint32)pa, size, (enum dma_data_direction)dir);
 }
 
 #if defined(BCMDBG_ASSERT)
