diff --git a/src/torrent/log_files.cc b/src/torrent/log_files.cc
index b5c2839..b72c6b6 100644

Index: src/torrent/log_files.cc
===================================================================
--- a/src/torrent/log_files.cc
+++ b/src/torrent/log_files.cc
@@ -40,6 +40,8 @@
 #include <functional>
 #include <cstring>
 #include <fcntl.h>
+#include <unistd.h>
+#include <stdio.h>
 
 #include "rak/timer.h"
 #include "log_files.h"
@@ -57,7 +59,7 @@
 
   m_last_update = rak::timer::current().seconds();
 
-  return (m_fd = open(filename, O_WRONLY | O_CREAT | O_EXCL)) != -1;
+  return (m_fd = open(filename, O_WRONLY | O_CREAT | O_EXCL, S_IRUSR | S_IWUSR)) != -1;
 }
 
 void
