diff -burN /var/tmp/portage/wallpapoz-0.1/work/wallpapoz-0.1/main.cpp wallpapoz-0.1/main.cpp
--- /var/tmp/portage/wallpapoz-0.1/work/wallpapoz-0.1/main.cpp	2004-06-17 08:47:55.000000000 -0500
+++ wallpapoz-0.1/main.cpp	2005-01-15 23:17:09.599192856 -0600
@@ -14,9 +14,9 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
  */
 
+#include <gtkmm/main.h>
 #include "wallpapoz.h"
 #include <iostream>
-#include <gtkmm/main.h>
 using namespace std;
 
 int main(int argc, char* argv[]) {
diff -burN /var/tmp/portage/wallpapoz-0.1/work/wallpapoz-0.1/wallpapoz.h wallpapoz-0.1/wallpapoz.h
--- /var/tmp/portage/wallpapoz-0.1/work/wallpapoz-0.1/wallpapoz.h	2004-06-17 10:04:03.000000000 -0500
+++ wallpapoz-0.1/wallpapoz.h	2005-01-15 23:17:37.648928648 -0600
@@ -16,7 +16,6 @@
 #ifndef WALLPAPOZ_H
 #define WALLPAPOZ_H
 
-#include <libglademm/xml.h>
 #include <gtkmm/treemodelcolumn.h>
 #include <gtkmm/button.h>
 #include <gtkmm/image.h>
@@ -25,6 +24,7 @@
 #include <gtkmm/window.h>
 #include <iostream>
 #include "Process.h"
+#include <libglademm.h>
 
 class Wall : public Gtk::Window {
     public:
