diff -Nur a/src/cpp/CMakeLists.txt b/src/cpp/CMakeLists.txt
--- a/src/cpp/CMakeLists.txt	2024-12-24 18:34:26.431859298 -0800
+++ b/src/cpp/CMakeLists.txt	2024-12-25 01:21:54.942719987 -0800
@@ -5,7 +5,7 @@
 # Dependencies
 
 include(FetchContent)
-
+find_package(nlohmann_json)
 if(NOT TARGET nlohmann_json)
     FetchContent_Declare(nlohmann_json
         URL https://github.com/nlohmann/json/archive/refs/tags/v3.11.3.tar.gz
@@ -47,12 +47,21 @@
     endif()
 endfunction()
 
+find_path(safetensors.h_SOURCE_DIR safetensors.h
+        PATHS
+        /usr/include
+)
+if(NOT safetensors.h_SOURCE_DIR)
 FetchContent_Declare(safetensors.h
     URL https://github.com/hsnyder/safetensors.h/archive/974a85d7dfd6e010558353226638bb26d6b9d756.tar.gz
     URL_HASH SHA256=9aaf5961609601cf9aaa96582a207bce7c6e5fbf57ed2cc669bb7bde6a937d4b)
 FetchContent_MakeAvailable(safetensors.h)
+endif()
 
-ov_genai_build_jinja2cpp()
+find_package(jinja2cpp QUIET)
+if(NOT TARGET jinja2cpp)
+    ov_genai_build_jinja2cpp()
+endif()
 
 # Library
 
diff -Nur a/src/python/CMakeLists.txt b/src/python/CMakeLists.txt
--- a/src/python/CMakeLists.txt	2024-12-24 18:34:26.431859298 -0800
+++ b/src/python/CMakeLists.txt	2024-12-25 01:26:11.096062361 -0800
@@ -2,20 +2,23 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 
-include(FetchContent)
+find_package(pybind11)
+if(NOT pybind11_FOUND)
+    include(FetchContent)
 
-FetchContent_Declare(
-    pybind11
-    URL https://github.com/pybind/pybind11/archive/refs/tags/v2.13.5.tar.gz
-    URL_HASH SHA256=b1e209c42b3a9ed74da3e0b25a4f4cd478d89d5efbb48f04b277df427faf6252
-)
-FetchContent_GetProperties(pybind11)
-# search for FindPython3.cmake instead of legacy modules
-set(PYBIND11_FINDPYTHON ON)
+    FetchContent_Declare(
+        pybind11
+        URL https://github.com/pybind/pybind11/archive/refs/tags/v2.13.5.tar.gz
+        URL_HASH SHA256=b1e209c42b3a9ed74da3e0b25a4f4cd478d89d5efbb48f04b277df427faf6252
+    )
+    FetchContent_GetProperties(pybind11)
+    # search for FindPython3.cmake instead of legacy modules
+    set(PYBIND11_FINDPYTHON ON)
 
-if(NOT pybind11_POPULATED)
-    FetchContent_Populate(pybind11)
-    add_subdirectory(${pybind11_SOURCE_DIR} ${pybind11_BINARY_DIR})
+    if(NOT pybind11_POPULATED)
+        FetchContent_Populate(pybind11)
+        add_subdirectory(${pybind11_SOURCE_DIR} ${pybind11_BINARY_DIR})
+    endif()
 endif()
 
 file(GLOB python_sources "${CMAKE_CURRENT_SOURCE_DIR}/*.cpp")
