diff -Nur a/src/CMakeLists.txt b/src/CMakeLists.txt
--- a/src/CMakeLists.txt	2026-03-24 09:22:29.000000000 -0700
+++ b/src/CMakeLists.txt	2026-04-30 19:47:37.557028261 -0700
@@ -45,7 +45,7 @@
 
 ## Start bitcode library configurations
 
-if(CUDAToolkit_VERSION_MAJOR LESS 13)
+if(CUDAToolkit_VERSION_MAJOR LESS 12)
   set(LIB_CXX_STANDARD 11)
 else()
   set(LIB_CXX_STANDARD 17)
diff -Nur a/src/device/CMakeLists.txt b/src/device/CMakeLists.txt
--- a/src/device/CMakeLists.txt	2026-03-24 09:22:29.000000000 -0700
+++ b/src/device/CMakeLists.txt	2026-04-30 19:47:14.237353793 -0700
@@ -66,7 +66,7 @@
 message(STATUS "NVSHMEM_PREFIX: ${NVSHMEM_PREFIX}")
 message(STATUS "NVSHMEM_DEBUG: ${NVSHMEM_DEBUG}")
 
-if(CUDAToolkit_VERSION_MAJOR LESS 13)
+if(CUDAToolkit_VERSION_MAJOR LESS 12)
   set(LIB_CXX_STANDARD 11)
 else()
   set(LIB_CXX_STANDARD 17)
