--- a/cmake_config/NVSHMEMConfig.cmake.in
+++ b/cmake_config/NVSHMEMConfig.cmake.in
@@ -70,8 +70,8 @@
 
 get_filename_component(NVSHMEM_LIB_DIRS "${CMAKE_CURRENT_LIST_DIR}/../.." ABSOLUTE)
 set(LIB_DIRS
-${PACKAGE_PREFIX_DIR}/lib
 ${NVSHMEM_LIB_DIRS}
+${PACKAGE_PREFIX_DIR}/lib
 )
 
 

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -161,7 +161,7 @@
 configure_file(License.txt License.txt COPYONLY)
 
 set(INCLUDE_INSTALL_DIR include/)
-set(LIB_INSTALL_DIR lib/)
+set(LIB_INSTALL_DIR ${CMAKE_INSTALL_LIBDIR}/)
 set(BIN_INSTALL_DIR bin/)
 
 configure_package_config_file(
