--- a/cmake_config/NVSHMEMConfig.cmake.in
+++ b/cmake_config/NVSHMEMConfig.cmake.in
@@ -61,8 +61,8 @@
 )
 
 set(LIB_DIRS
-${PACKAGE_PREFIX_DIR}/lib
 ${CMAKE_CURRENT_LIST_DIR}/../..
+${PACKAGE_PREFIX_DIR}/lib
 )
 
 
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -177,7 +177,7 @@
 configure_file(License.txt License.txt COPYONLY)
 
 set(INCLUDE_INSTALL_DIR include/)
-set(LIB_INSTALL_DIR lib/)
+set(LIB_INSTALL_DIR ${CMAKE_INSTALL_LIBDIR}/)
 set(BIN_INSTALL_DIR bin/)
 
 configure_package_config_file(
