diff -Nur a/src/CMakeLists.txt b/src/CMakeLists.txt
--- a/src/CMakeLists.txt	2026-03-24 09:22:29.000000000 -0700
+++ b/src/CMakeLists.txt	2026-04-30 23:11:40.967787631 -0700
@@ -25,6 +25,7 @@
                     CMAKE_ARGS  -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
                                 -DNVSHMEM_PREFIX=${NVSHMEM_PREFIX}
                                 -DNVSHMEM_DEBUG=${NVSHMEM_DEBUG}
+				-DCMAKE_INSTALL_PREFIX=${CMAKE_BINARY_DIR}
                                 -DNVSHMEM_DEVICELIB_CUDA_HOME=${NVSHMEM_DEVICELIB_CUDA_HOME}
                                 -DNVSHMEM_ENABLE_ALL_DEVICE_INLINING=${NVSHMEM_ENABLE_ALL_DEVICE_INLINING}
                                 -DNVSHMEM_BUILD_WITH_CUTLASS=${NVSHMEM_BUILD_WITH_CUTLASS}
@@ -1022,16 +1023,16 @@
 
 install(
   FILES ${CMAKE_CURRENT_SOURCE_DIR}/../changelog
-  DESTINATION ./
+  DESTINATION share/doc/nvshmem
 )
 install(
   FILES ${CMAKE_CURRENT_SOURCE_DIR}/../License.txt
-  DESTINATION ./
+  DESTINATION share/doc/nvshmem
 )
 install(
   FILES ${CMAKE_CURRENT_SOURCE_DIR}/../version.txt
-  FILES ${CMAKE_CURRENT_SOURCE_DIR}/../git_commit.txt
-  DESTINATION ./
+        ${CMAKE_CURRENT_SOURCE_DIR}/../git_commit.txt
+  DESTINATION share/doc/nvshmem
 )
 
 include(CMakePackageConfigHelpers)
