--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -29,6 +29,7 @@
                     CMAKE_ARGS  -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
                                 -DNVSHMEM_PREFIX=${NVSHMEM_PREFIX}
                                 -DNVSHMEM_DEBUG=${NVSHMEM_DEBUG}
+				-DCMAKE_INSTALL_PREFIX=${CMAKE_BINARY_DIR}
                                 -DNVSHMEM_VERBOSE=${NVSHMEM_VERBOSE}
                                 -DNVSHMEM_DEVICELIB_CUDA_HOME=${NVSHMEM_DEVICELIB_CUDA_HOME}
                                 -DNVSHMEM_ENABLE_ALL_DEVICE_INLINING=${NVSHMEM_ENABLE_ALL_DEVICE_INLINING}
@@ -1219,17 +1220,17 @@
 install(
   EXPORT NVSHMEMTargets
   NAMESPACE nvshmem::
-  DESTINATION lib/cmake/nvshmem/
+  DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/nvshmem/
   COMPONENT Export
 )
 
 # ${CMAKE_CURRENT_BINARY_DIR}/NVSHMEMConfig.cmake
 install(FILES ${CMAKE_BINARY_DIR}/NVSHMEMVersion.cmake
-        DESTINATION lib/cmake/nvshmem
+	DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/nvshmem
 )
 
 install(FILES ${CMAKE_BINARY_DIR}/NVSHMEMConfig.cmake
-        DESTINATION lib/cmake/nvshmem
+	DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/nvshmem
 	)
 
 export(
@@ -1241,8 +1242,8 @@
 install(
 	TARGETS nvshmem_host nvshmem-info ${ACTIVE_BOOTSTRAPS} ${ACTIVE_TRANSPORTS}
   EXPORT NVSHMEMTargets
-  LIBRARY DESTINATION lib
-  ARCHIVE DESTINATION lib
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
   RUNTIME DESTINATION bin
   INCLUDES
   DESTINATION include
@@ -1251,28 +1252,28 @@
 
 install(
   FILES ${CMAKE_CURRENT_BINARY_DIR}/lib/libnvshmem_device.a
-  DESTINATION lib/
+  DESTINATION ${CMAKE_INSTALL_LIBDIR}/
   PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE
               GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
 )
 
 install(
   FILES ${CMAKE_BINARY_DIR}/lib/cmake/nvshmem/NVSHMEMDeviceTargets.cmake
-  DESTINATION lib/cmake/nvshmem/
+  DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/nvshmem/
   PERMISSIONS OWNER_READ OWNER_WRITE
               GROUP_READ WORLD_READ
 )
 
 install(
   FILES ${CMAKE_BINARY_DIR}/lib/cmake/nvshmem/NVSHMEMDeviceTargets-$<LOWER_CASE:${CMAKE_BUILD_TYPE}>.cmake
-  DESTINATION lib/cmake/nvshmem/
+  DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/nvshmem/
   PERMISSIONS OWNER_READ OWNER_WRITE
               GROUP_READ WORLD_READ
 )
 
 install(
   DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/lib/
-  DESTINATION lib
+  DESTINATION ${CMAKE_INSTALL_LIBDIR}
   FILE_PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE
                    GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
 )
@@ -1320,16 +1321,16 @@
 
 install(
   FILES ${CMAKE_CURRENT_SOURCE_DIR}/../changelog
-  DESTINATION ./
+  DESTINATION share/doc/nvshmem
 )
 install(
   FILES ${CMAKE_CURRENT_SOURCE_DIR}/../License.txt
-  DESTINATION ./
+  DESTINATION share/doc/nvshmem
 )
 install(
   FILES ${CMAKE_CURRENT_SOURCE_DIR}/../version.txt
-  FILES ${CMAKE_CURRENT_SOURCE_DIR}/../git_commit.txt
-  DESTINATION ./
+        ${CMAKE_CURRENT_SOURCE_DIR}/../git_commit.txt
+  DESTINATION share/doc/nvshmem
 )
 
 include(CMakePackageConfigHelpers)
