--- PyLucene-src-2.0.0-1.orig/Makefile	2006-06-26 12:47:33.000000000 +0200
+++ PyLucene-src-2.0.0-1/Makefile	2006-07-05 01:26:52.000000000 +0200
@@ -730,11 +730,11 @@
 
 install:: all
 	mkdir -p $(PYTHON_SITE)
-	install PyLucene.py $(PYTHON_SITE)
+	install -m 644 PyLucene.py $(PYTHON_SITE)
 	install $(PYLUCENE_LIB) $(PYTHON_SITE)
 	mkdir -p $(PYTHON_SITE)/security
-	install $(GCJ_LIBDIR)/security/classpath.security $(PYTHON_SITE)/security
-	install $(GCJ_LIBDIR)/security/libgcj.security $(PYTHON_SITE)/security
+	install -m 644 $(GCJ_LIBDIR)/security/classpath.security $(PYTHON_SITE)/security
+	install -m 644 $(GCJ_LIBDIR)/security/libgcj.security $(PYTHON_SITE)/security
 
 ifneq ($(LIB_INSTALL),)
 install::
