diff -ur local-time-1.0.1_p20100617.orig/src/local-time.lisp local-time-1.0.1_p20100617/src/local-time.lisp
--- local-time-1.0.1_p20100617.orig/src/local-time.lisp	2010-12-27 01:07:31.733466036 +0100
+++ local-time-1.0.1_p20100617/src/local-time.lisp	2011-01-31 12:21:18.795037693 +0100
@@ -206,22 +206,7 @@
 ;;; Variables
 
 (defparameter *default-timezone-repository-path*
-  (flet ((try (project-home-directory)
-           (when project-home-directory
-             (ignore-errors
-               (truename
-                (merge-pathnames "zoneinfo/"
-                                 (make-pathname :directory (pathname-directory project-home-directory))))))))
-    (or (when (find-package "ASDF")
-          (let ((path (eval (read-from-string
-                             "(let ((system (asdf:find-system :local-time nil)))
-                                (when system
-                                  (asdf:component-pathname system)))"))))
-            (try path)))
-        (let ((path (or #.*compile-file-truename*
-                        *load-truename*)))
-          (when path
-            (try (merge-pathnames "../" path)))))))
+  #p"/usr/share/zoneinfo/")
 
 ;;; Month information
 (defparameter +month-names+
