--- pthreads-w32-2-8-0-release/GNUmakefile.orig	2011-08-15 22:27:14.714000092 +0200
+++ pthreads-w32-2-8-0-release/GNUmakefile	2011-08-15 22:29:59.409000094 +0200
@@ -63,7 +63,7 @@
 XOPT	=
 
 RCFLAGS		= --include-dir=.
-LFLAGS		= -lwsock32
+LFLAGS		:= $(LDFLAGS) -lwsock32
 
 # ----------------------------------------------------------------------
 # The library can be built with some alternative behaviour to
@@ -98,7 +98,7 @@
 
 ## Mingw32
 MAKE		?= make
-CFLAGS	= $(OPT) $(XOPT) -I. -DHAVE_CONFIG_H -Wall
+CFLAGS	:= $(CFLAGS) $(OPT) $(XOPT) -I. -DHAVE_CONFIG_H -Wall
 
 DLL_INLINED_OBJS	= \
 		pthread.o \
