commit 3d6cb902c98dccc7432ce2f7e166f5c68056f9ee
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Tue May 10 13:44:47 2011 -0400

    use 64-bits for signals

diff --git a/libc/kernel/arch-x86/asm/signal.h b/libc/kernel/arch-x86/asm/signal.h
index 6d84eab..f14fc75 100644
--- a/libc/kernel/arch-x86/asm/signal.h
+++ b/libc/kernel/arch-x86/asm/signal.h
@@ -16,11 +16,12 @@
 #include <linux/types.h>
 #include <linux/time.h>
 #include <linux/compiler.h>
+#include <stdint.h>
 
 struct siginfo;
 
-#define NSIG 32
-typedef unsigned long sigset_t;
+#define NSIG 64
+typedef uint64_t sigset_t;
 
 #endif
 
