diff -urw sql12.orig/app/src/net/sourceforge/squirrel_sql/client/gui/builders/DefaultFormBuilder.java sql12/app/src/net/sourceforge/squirrel_sql/client/gui/builders/DefaultFormBuilder.java
--- sql12.orig/app/src/net/sourceforge/squirrel_sql/client/gui/builders/DefaultFormBuilder.java	2006-04-18 16:57:39.000000000 +0200
+++ sql12/app/src/net/sourceforge/squirrel_sql/client/gui/builders/DefaultFormBuilder.java	2006-04-18 16:58:11.000000000 +0200
@@ -198,7 +198,7 @@
      */
     public void setLineGapSize(ConstantSize lineGapSize) {
         RowSpec rowSpec = FormFactory.createGapRowSpec(lineGapSize);
-        this.lineGapSpec = rowSpec.asUnmodifyable();
+        this.lineGapSpec = rowSpec;
     }
     
     /**
@@ -219,7 +219,7 @@
      */
     public void setParagraphGapSize(ConstantSize paragraphGapSize) {
         RowSpec rowSpec = FormFactory.createGapRowSpec(paragraphGapSize);
-        this.paragraphGapSpec = rowSpec.asUnmodifyable();
+        this.paragraphGapSpec = rowSpec;
     }
     
     /**
