diff -abBU 1 -rP gmailfs-0.7.2.orig/gmailfs.conf gmailfs-0.7.2/gmailfs.conf
--- gmailfs-0.7.2.orig/gmailfs.conf	2005-11-27 12:22:22.000000000 +0200
+++ gmailfs-0.7.2/gmailfs.conf		2006-02-27 21:48:34.000000000 +0200
@@ -26,2 +26,2 @@
 # equivalent for it
-logfile = ~/gmailfs.log
+logfile = /var/log/gmailfs.log
diff -abBU 1 -rP gmailfs-0.7.2.orig/gmailfs.py gmailfs-0.7.2/gmailfs.py
--- gmailfs-0.7.2.orig/gmailfs.py	2006-02-06 23:40:51.000000000 +0200
+++ gmailfs-0.7.2/gmailfs.py		2006-02-27 21:48:10.000000000 +0200
@@ -1122,3 +1122,3 @@
 
-GmailConfig("/etc/gmailfs.conf")
+GmailConfig("~/.gmailfs")
 try:
