--- a/cmake/For.cmake
+++ b/cmake/For.cmake
@@ -4,11 +4,6 @@
 set(FOR_NAME libfor-${FOR_VERSION})
 set(FOR_TAR_PATH ${DEP_ROOT_DIR}/${FOR_NAME}.tar.gz)
 
-if(NOT EXISTS ${FOR_TAR_PATH})
-    message(STATUS "Downloading libfor...")
-    file(DOWNLOAD https://github.com/cruppstahl/libfor/archive/${FOR_VERSION}.tar.gz ${FOR_TAR_PATH})
-endif()
-
 if(NOT EXISTS ${DEP_ROOT_DIR}/${FOR_NAME})
     message(STATUS "Extracting libfor...")
     execute_process(COMMAND ${CMAKE_COMMAND} -E tar xzf ${FOR_TAR_PATH} WORKING_DIRECTORY ${DEP_ROOT_DIR})
--- a/cmake/H2O.cmake
+++ b/cmake/H2O.cmake
@@ -4,11 +4,6 @@
 set(H2O_NAME h2o-${H2O_VERSION})
 set(H2O_TAR_PATH ${DEP_ROOT_DIR}/${H2O_NAME}.tar.gz)
 
-if(NOT EXISTS ${H2O_TAR_PATH})
-    message(STATUS "Downloading https://github.com/h2o/h2o/archive/${H2O_VERSION}.tar.gz")
-    file(DOWNLOAD https://github.com/h2o/h2o/archive/${H2O_VERSION}.tar.gz ${H2O_TAR_PATH})
-endif()
-
 if(NOT EXISTS ${DEP_ROOT_DIR}/${H2O_NAME})
     message(STATUS "Extracting ${H2O_NAME}...")
     execute_process(COMMAND ${CMAKE_COMMAND} -E tar xzf ${H2O_TAR_PATH} WORKING_DIRECTORY ${DEP_ROOT_DIR}/)
@@ -25,6 +20,7 @@
             WORKING_DIRECTORY ${DEP_ROOT_DIR}/${H2O_NAME})
 
     execute_process(COMMAND ${CMAKE_COMMAND}
+            -DCMAKE_POLICY_VERSION_MINIMUM=3.5
             -DCMAKE_FIND_LIBRARY_SUFFIXES=.a
             -DCMAKE_CXX_FLAGS="-DNDEBUG"
             -DCMAKE_C_FLAGS="-DNDEBUG"
--- a/cmake/RocksDB.cmake
+++ b/cmake/RocksDB.cmake
@@ -20,6 +20,7 @@
     set(ENV{DEBUG_LEVEL} 0)
     set(ENV{USE_RTTI} 1)
     set(ENV{PLATFORM_LDFLAGS} "-Wl,-Bstatic")
+    set(ENV{EXTRA_CXXFLAGS} "-std=c++17")
 
     set(ENV{ROCKSDB_DISABLE_GFLAGS} 1)
     set(ENV{ROCKSDB_DISABLE_BZIP} 1)
