--- robot_postinstall.py.old			2010-09-27 11:41:33.000000000 +0100
+++ robot_postinstall.py			2010-12-10 10:25:51.646782510 +0000
@@ -26,7 +26,7 @@
     Run as part of the generic installation procedure from 'setup.py' after
     running 'python setyp.py install'.
     """
-    _update_scripts(script_names, script_dir, robot_dir)
+    _update_scripts(script_names, script_dir, '/usr/lib/python2.6/site-packages/robot/')
 
 
 def windows_binary_install():
@@ -162,7 +162,7 @@
     elif  sys.platform.count('cygwin'):
         return 'jython.bat', ['/cygdrive/c', '/cygdrive/d']
     else:
-        return 'jython', ['/usr/local','/opt']
+        return 'jython2.5', ['/usr/local','/opt']
 
 def _jython_in_path(jyexe):
     out = os.popen('%s --version 2>&1' % jyexe )
