--- a/CMakeLists.txt	2011-03-08 21:41:25.000000000 +0100
+++ b/CMakeLists.txt	2011-03-08 21:38:35.000000000 +0100
@@ -383,8 +383,8 @@
 			# does not find gdk-pixbuf-2.0.  On the other hand some 2.8.3
 			# users have complained that pkg-config does not find
 			# gdk-pixbuf-2.0
-			if(${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}.${CMAKE_PATCH_VERSION}
-					VERSION_EQUAL 2.8.2)
+			if(${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}.${CMAKE_PATCH_VERSION}	VERSION_EQUAL 2.8.1 
+				OR ${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}.${CMAKE_PATCH_VERSION}	VERSION_EQUAL 2.8.2)
 				check_lib(GTK2 gtk+-2.0 gtk.h REQUIRED)
 			else()
 				include(FindGTK2)
