diff -ur ocrfeeder-0.7.6.orig/src/ocrfeeder/feeder/imageManipulation.py ocrfeeder-0.7.6/src/ocrfeeder/feeder/imageManipulation.py
--- ocrfeeder-0.7.6.orig/src/ocrfeeder/feeder/imageManipulation.py	2011-10-16 19:15:39.111350115 +0200
+++ ocrfeeder-0.7.6/src/ocrfeeder/feeder/imageManipulation.py	2011-10-16 19:28:34.960363628 +0200
@@ -22,7 +22,7 @@
 from ocrfeeder.util.lib import debug
 
 import gettext
-import Image, ImageDraw
+from PIL import Image, ImageDraw
 import os.path
 from ocrfeeder.util import graphics
 import sys
diff -ur ocrfeeder-0.7.6.orig/src/ocrfeeder/feeder/layoutAnalysis.py ocrfeeder-0.7.6/src/ocrfeeder/feeder/layoutAnalysis.py
--- ocrfeeder-0.7.6.orig/src/ocrfeeder/feeder/layoutAnalysis.py	2011-10-16 19:15:39.111350115 +0200
+++ ocrfeeder-0.7.6/src/ocrfeeder/feeder/layoutAnalysis.py	2011-10-16 19:29:45.400364853 +0200
@@ -23,7 +23,7 @@
 from ocrfeeder.util.constants import OCRFEEDER_DEBUG, DTP
 from ocrfeeder.studio.dataHolder import DataBox
 from imageManipulation import ImageProcessor
-import Image
+from PIL import Image
 import re
 import math
 
diff -ur ocrfeeder-0.7.6.orig/src/ocrfeeder/studio/dataHolder.py ocrfeeder-0.7.6/src/ocrfeeder/studio/dataHolder.py
--- ocrfeeder-0.7.6.orig/src/ocrfeeder/studio/dataHolder.py	2011-10-16 19:15:39.107350115 +0200
+++ ocrfeeder-0.7.6/src/ocrfeeder/studio/dataHolder.py	2011-10-16 19:30:44.998365891 +0200
@@ -20,7 +20,7 @@
 
 import os.path
 from ocrfeeder.util import graphics
-import Image
+from PIL import Image
 import gobject
 from ocrfeeder.util.graphics import getImagePrintSize, getImageResolution
 from ocrfeeder.util import TEXT_TYPE, IMAGE_TYPE, ALIGN_LEFT, lib
diff -ur ocrfeeder-0.7.6.orig/src/ocrfeeder/studio/widgetPresenter.py ocrfeeder-0.7.6/src/ocrfeeder/studio/widgetPresenter.py
--- ocrfeeder-0.7.6.orig/src/ocrfeeder/studio/widgetPresenter.py	2011-10-16 19:15:39.107350115 +0200
+++ ocrfeeder-0.7.6/src/ocrfeeder/studio/widgetPresenter.py	2011-10-16 19:30:31.559365658 +0200
@@ -26,7 +26,7 @@
 from ocrfeeder.util.constants import *
 from ocrfeeder.util.graphics import convertPixbufToImage
 from enchant.checker import SpellChecker
-import Image
+from PIL import Image
 import gettext
 import gobject
 import goocanvas
diff -ur ocrfeeder-0.7.6.orig/src/ocrfeeder/util/graphics.py ocrfeeder-0.7.6/src/ocrfeeder/util/graphics.py
--- ocrfeeder-0.7.6.orig/src/ocrfeeder/util/graphics.py	2011-10-16 19:15:39.112350115 +0200
+++ ocrfeeder-0.7.6/src/ocrfeeder/util/graphics.py	2011-10-16 19:29:07.546364194 +0200
@@ -19,7 +19,7 @@
 ###########################################################################
 
 from lib import debug
-import Image
+from PIL import Image
 import gtk
 import math
 
diff -ur ocrfeeder-0.7.6.orig/src/ocrfeeder/util/lib.py ocrfeeder-0.7.6/src/ocrfeeder/util/lib.py
--- ocrfeeder-0.7.6.orig/src/ocrfeeder/util/lib.py	2011-10-16 19:15:39.111350115 +0200
+++ ocrfeeder-0.7.6/src/ocrfeeder/util/lib.py	2011-10-16 19:28:49.417363878 +0200
@@ -20,7 +20,7 @@
 
 import os
 import mimetypes
-import Image
+from PIL import Image
 import tempfile
 import gtk
 import math
