diff -ur ocrfeeder-0.7.6.orig/configure.ac ocrfeeder-0.7.6/configure.ac
--- ocrfeeder-0.7.6.orig/configure.ac	2011-10-16 19:15:39.114350115 +0200
+++ ocrfeeder-0.7.6/configure.ac	2011-10-16 19:16:15.133350741 +0200
@@ -45,7 +45,6 @@
 src/Makefile
 src/ocrfeeder/Makefile
 src/ocrfeeder/feeder/Makefile
-src/ocrfeeder/odf/Makefile
 src/ocrfeeder/studio/Makefile
 src/ocrfeeder/util/Makefile
 src/ocrfeeder/util/constants.py
diff -ur ocrfeeder-0.7.6.orig/src/ocrfeeder/feeder/documentGeneration.py ocrfeeder-0.7.6/src/ocrfeeder/feeder/documentGeneration.py
--- ocrfeeder-0.7.6.orig/src/ocrfeeder/feeder/documentGeneration.py	2011-10-16 19:15:39.111350115 +0200
+++ ocrfeeder-0.7.6/src/ocrfeeder/feeder/documentGeneration.py	2011-10-16 19:19:26.912354081 +0200
@@ -18,11 +18,11 @@
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 ###########################################################################
 
-from ocrfeeder.odf.draw import Frame, TextBox, Image
-from ocrfeeder.odf.opendocument import OpenDocumentText
-from ocrfeeder.odf.style import Style, MasterPage, GraphicProperties, ParagraphProperties, \
+from odf.draw import Frame, TextBox, Image
+from odf.opendocument import OpenDocumentText
+from odf.style import Style, MasterPage, GraphicProperties, ParagraphProperties, \
     TextProperties, PageLayout, PageLayoutProperties
-from ocrfeeder.odf.text import P, Page, PageSequence
+from odf.text import P, Page, PageSequence
 from pango import WEIGHT_BOLD, WEIGHT_NORMAL, STYLE_ITALIC, STYLE_NORMAL, \
     STYLE_OBLIQUE
 from ocrfeeder.util import TEXT_TYPE, IMAGE_TYPE, ALIGN_LEFT, ALIGN_RIGHT, ALIGN_CENTER, \
diff -ur ocrfeeder-0.7.6.orig/src/ocrfeeder/Makefile.am ocrfeeder-0.7.6/src/ocrfeeder/Makefile.am
--- ocrfeeder-0.7.6.orig/src/ocrfeeder/Makefile.am	2011-10-16 19:15:39.107350115 +0200
+++ ocrfeeder-0.7.6/src/ocrfeeder/Makefile.am	2011-10-16 19:16:37.426351129 +0200
@@ -6,8 +6,7 @@
 SUBDIRS = \
 	feeder/ \
 	studio/ \
-	util/ \
-	odf/
+	util/
