--- unix/sys_unix.c.old	2008-03-24 09:19:29.376415475 +0500
+++ unix/sys_unix.c	2008-03-24 09:19:53.732415394 +0500
@@ -240,7 +240,7 @@ void *Sys_GetGameAPI (void *parms)
 		path = FS_NextPath (path);
 		if (!path)
 			return NULL;		// couldn't find one anywhere
-		sprintf (name, "%s/%s/%s", curpath, path, gamename);
+		sprintf (name, "%s/%s", path, gamename);
 		game_library = dlopen (name, RTLD_LAZY );
 		if (game_library)
 		{
