diff -Naur kpager2-0.6.0b/src/config.h kpager2-0.6.0b-new/src/config.h
--- kpager2-0.6.0b/src/config.h	2005-03-30 10:21:09.000000000 +0200
+++ kpager2-0.6.0b-new/src/config.h	2006-09-03 13:38:24.000000000 +0200
@@ -43,6 +43,9 @@
 class KJanusWidget;
 class KKeyChooser;
 class KPagerJanusWidget;
+class KPagerConfigGeneral;
+class KPagerConfigCustom;
+class KPagerConfig3d;
 #include "kpagerdesktop.h"
 //enum KPagerDesktop::WindowTransparentMode;
 
