Gemeinsame Unterverzeichnisse: kpager2-0.5.2.orig/kpager2 und kpager2-0.5.2/kpager2.
diff -u kpager2-0.5.2.orig/kpagermainwin.moc.cpp kpager2-0.5.2/kpagermainwin.moc.cpp
--- kpager2-0.5.2.orig/kpagermainwin.moc.cpp	2005-11-13 13:50:59.000000000 +0100
+++ kpager2-0.5.2/kpagermainwin.moc.cpp	2005-11-13 13:54:49.000000000 +0100
@@ -6,7 +6,8 @@
 **
 ** WARNING! All changes made in this file will be lost!
 *****************************************************************************/
-
+#include "version.h"
+#ifndef KPAGER_IS_APPLET
 #undef QT_NO_COMPAT
 #include "kpagermainwin.h"
 #include <qmetaobject.h>
@@ -101,3 +102,4 @@
 
 bool KPagerMainWindow::qt_static_property( QObject* , int , int , QVariant* ){ return FALSE; }
 #endif // QT_NO_PROPERTIES
+#endif
