diff -Nur a/sycl/cmake/modules/FetchUnifiedRuntime.cmake b/sycl/cmake/modules/FetchUnifiedRuntime.cmake
--- a/sycl/cmake/modules/FetchUnifiedRuntime.cmake	2024-12-09 16:51:36.000000000 -0800
+++ b/sycl/cmake/modules/FetchUnifiedRuntime.cmake	2024-12-13 11:54:36.423812822 -0800
@@ -293,7 +293,3 @@
   endif()
 endif()
 
-install(TARGETS umf
-  LIBRARY DESTINATION "lib${LLVM_LIBDIR_SUFFIX}" COMPONENT unified-memory-framework
-  ARCHIVE DESTINATION "lib${LLVM_LIBDIR_SUFFIX}" COMPONENT unified-memory-framework
-  RUNTIME DESTINATION "bin" COMPONENT unified-memory-framework)
