*** Makefile	Sun Apr 10 22:52:19 2005
--- Makefile.install	Mon Apr 11 23:56:36 2005
***************
*** 4,9 ****
--- 4,10 ----
  
  CFLAGS = -O4 -Wall `sdl-config --cflags` -Wall
  LDFLAGS = -L/usr/X11R6/lib -lGL -lGLU -lSDL_ttf -lSDL_image `sdl-config --libs`
+ INSTALL=/bin/install
  
  DISTDIR = gravit-0.3.0
  TMPBASE = /tmp
***************
*** 11,16 ****
--- 12,21 ----
  
  all: final
  
+ install: final
+ 	$(INSTALL) -d $(DESTDIR)usr/bin
+ 	$(INSTALL) -m 755 gravit $(DESTDIR)usr/bin
+ 	
  dep:
  	@makedepend $(CFLAGS) *.c 2> /dev/null
  
