--- /usr/portage/www-apps/rt/rt-3.6.7.ebuild	2008-07-01 16:48:46.000000000 +0000
+++ /usr/local/portage/www-apps/rt/rt-3.8.1.ebuild	2008-08-27 13:57:23.000000000 +0000
@@ -1,6 +1,6 @@
 # Copyright 1999-2008 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
-# $Header: /var/cvsroot/gentoo-x86/www-apps/rt/rt-3.6.7.ebuild,v 1.1 2008/07/01 16:48:46 wrobel Exp $
+# $Header:  Exp $
 
 inherit webapp eutils depend.apache confutils
 
@@ -15,13 +15,24 @@
 DEPEND="
 	>=dev-lang/perl-5.8.3
 
+	dev-perl/Email-Address
+	dev-perl/MIME-Types
+	dev-perl/PerlIO-eol
+	dev-perl/GnuPG-Interface
+	dev-perl/net-server
+	>=dev-perl/HTTP-Server-Simple-0.34
+	dev-perl/File-ShareDir
+	dev-perl/Data-ICal
+	>=dev-perl/HTML-RewriteAttributes-0.02
+	>=virtual/perl-Encode-2.13
+
 	>=dev-perl/Apache-Session-1.53
 	dev-perl/Cache-Simple-TimedExpiry
 	dev-perl/Calendar-Simple
 	>=dev-perl/class-returnvalue-0.40
 	>=dev-perl/CSS-Squish-0.06
 	>=dev-perl/DBI-1.37
-	>=dev-perl/dbix-searchbuilder-1.53
+	>=dev-perl/dbix-searchbuilder-1.54
 	>=dev-perl/Devel-StackTrace-1.19
 	dev-perl/GD
 	dev-perl/GDGraph
@@ -31,15 +42,14 @@
 	dev-perl/HTML-Parser
 	>=dev-perl/HTML-Scrubber-0.08
 	dev-perl/HTML-Tree
-	>=dev-perl/HTTP-Server-Simple-0.07
 	>=dev-perl/HTTP-Server-Simple-Mason-0.09
 	dev-perl/libwww-perl
 	dev-perl/locale-maketext-fuzzy
 	>=dev-perl/locale-maketext-lexicon-0.32
 	>=dev-perl/log-dispatch-2.0
 	>=dev-perl/MailTools-1.60
-	>=dev-perl/MIME-tools-5.417
-	dev-perl/Module-Versions-Report
+	>=dev-perl/MIME-tools-5.425
+	>=dev-perl/Module-Versions-Report-1.05
 	dev-perl/regexp-common
 	dev-perl/TermReadKey
 	dev-perl/text-autoformat
@@ -52,7 +62,7 @@
 	>=dev-perl/Tree-Simple-1.04
 	dev-perl/UNIVERSAL-require
 	>=dev-perl/XML-RSS-1.05
-	>=virtual/perl-CGI-2.92
+	>=virtual/perl-CGI-3.38
 	virtual/perl-digest-base
 	>=virtual/perl-Digest-MD5-2.27
 	>=virtual/perl-File-Spec-0.8
@@ -150,6 +160,9 @@
 	local web="apache"
 	use lighttpd && web="lighttpd"
 
+	local webhandler='modperl2'
+	use fastcgi && webhandler='fastcgi'
+
 	local dbtype dba
 
 	if use mysql; then
@@ -174,17 +187,18 @@
 		--with-rt-group=rt \
 		--with-web-user=${web} \
 		--with-web-group=${web} \
+		--with-web-handler=${webhandler} \
 		${dbtype} ${dba}
 
 	# check for missing deps and ask to report if something is broken
-	local myconf="--verbose \
-		$(enable_extension_withonly mysql mysql) \
-		$(enable_extension_withonly postgresql postgres) \
-		$(enable_extension_withonly fastcgi fastcgi) \
-		$(enable_extension_withonly fastcgi lighttpd)" \
+	local my_conf="--verbose "
+	enable_extension_withonly  mysql      mysql
+	enable_extension_withonly  postgresql postgres
+	enable_extension_withonly  fastcgi    fastcgi
+	enable_extension_withonly  fastcgi    lighttpd
 
 	if ! useq fastcgi && ! useq lighttpd; then
-		myconf="${myconf} --with-modperl2"
+		my_conf="${my_conf} --with-modperl2"
 	fi
 
 	/usr/bin/perl ./sbin/rt-test-dependencies ${myconf} > "${T}"/t
@@ -201,7 +215,7 @@
 
 src_install() {
 	webapp_src_preinst
-	emake install || die
+	make install || die
 
 	# make sure we don't clobber existing site configuration
 	rm -f "${D}"/${MY_HOSTROOTDIR}/${PF}/etc/RT_SiteConfig.pm
