--- a/src/Transform/ProcessKrnlParallelClause.cpp
+++ b/src/Transform/ProcessKrnlParallelClause.cpp
@@ -75,7 +75,7 @@ struct ProcessKrnlParallelClausePattern
         // Nevertheless, this warning attests that this might be a possibility.
         // In such case, we would get a compiler warning/error of use before
         // def.
         MutableOperandRange mutableNumThreads =
-            parOp.getNumThreadsVarsMutable();
+            parOp.getNumThreadsMutable();
         mutableNumThreads.assign(numThreads);
       }
