--- ./platforms/unix/vm/debug.h.orig	2003-08-07 07:39:34.000000000 +0000
+++ ./platforms/unix/vm/debug.h	2009-05-09 21:38:27.000000000 +0000
@@ -7,6 +7,13 @@
 #endif
 
 
+// we need stdio.h to override glibc's dprintf function
+// since new glibc has this function with different interface
+#include <features.h>
+#ifdef __USE_XOPEN2K8
+# include <stdio.h>
+#endif
+
 #if (DEBUG)
   /* the thing to use here is a variadic macro, but Apple's gcc barfs on
   ** them when running in precomp mode.  did they _really_ have to break
--- olpc/platforms/unix/plugins/MIDIPlugin/sqUnixMIDIALSA.inc.orig	2007-03-11 23:56:36.000000000 +0000
+++ olpc/platforms/unix/plugins/MIDIPlugin/sqUnixMIDIALSA.inc	2009-05-09 22:07:29.000000000 +0000
@@ -116,6 +116,12 @@
 
 #include <alsa/asoundlib.h>
 
+#if (DEBUG)
+# define dprintf(...) printf(__VA_ARGS__)
+#else
+# define dprintf(...)
+#endif
+
 static snd_seq_t *seq      =  0;
 static int	  queue    =  0;
 static int	  in_port  = -1;
