--- squeak-vm-sugar-3.10.3/platforms/unix/config/make.cfg.in.orig	2006-10-18 16:17:11.000000000 +0000
+++ squeak-vm-sugar-3.10.3/platforms/unix/config/make.cfg.in	2008-12-27 02:14:18.000000000 +0000
@@ -65,7 +65,7 @@
 mandir=		@mandir@
 imgdir=		@imgdir@
 plgdir=		@plgdir@
-docdir=		$(prefix)/doc/squeak
+docdir=		@docdir@/squeak
 
 SHELL=		@SHELL@
 MKINSTALLDIRS=	$(SHELL) $(cfgdir)/mkinstalldirs
--- olpc/platforms/unix/config/make.prg.in.orig	2003-08-07 02:44:58.000000000 +0000
+++ olpc/platforms/unix/config/make.prg.in	2009-01-09 09:00:04.000000000 +0000
@@ -11,5 +11,5 @@
 		  $(LDFLAGS) $(XLDFLAGS) -export-dynamic -c -o
 LINK		= $(LIBTOOL) --mode=link \
 		  $(CC) $(CFLAGS) $(XCFLAGS) \
-		  $(LDFLAGS) $(XLDFLAGS) -export-dynamic -R$(libdir) -o
+		  $(LDFLAGS) $(XLDFLAGS) -export-dynamic -o
 RANLIB		= :
