--- a/src/compiler/expr.rs	2026-06-19 02:59:22.000000000 -0700
+++ b/src/compiler/expr.rs	2026-06-21 11:05:53.426753382 -0700
@@ -154,16 +154,6 @@
     pub end: u32,
 }
 
-impl From<std::range::Range<usize>> for Span {
-    #[inline(always)]
-    fn from(value: std::range::Range<usize>) -> Self {
-        Self {
-            start: value.start as u32,
-            end: value.end as u32,
-        }
-    }
-}
-
 impl From<std::ops::Range<usize>> for Span {
     #[inline(always)]
     fn from(value: std::ops::Range<usize>) -> Self {
