diff -urNP edge-1.29-source/SConstruct edge-1.29-source_work/SConstruct
--- edge-1.29-source/SConstruct	2007-06-06 15:07:47.000000000 +0200
+++ edge-1.29-source_work/SConstruct	2007-08-27 23:45:44.000000000 +0200
@@ -24,6 +24,8 @@
 
 base_env = Environment()
 
+base_env.SConsignFile()
+
 if build_info['cross']:
   base_env.Tool('crossmingw', toolpath=['build'])
 
@@ -95,10 +97,10 @@
 env.Append(LIBS = ['png', 'jpeg', 'z'])
 
 # FLTK
-if build_info['platform'] == 'linux' and build_info['release']:
+if build_info['platform'] == 'linux':
     env.Append(CCFLAGS = ['-DUSE_FLTK'])
-    env.ParseConfig('#linux_lib/fltk-1.1.7/fltk-config --cflags')
-    env.Append(LIBPATH = ['#linux_lib/fltk-1.1.7/lib'])
+    env.ParseConfig('fltk-config --cflags')
+    env.ParseConfig('fltk-config --ldflags')
     env.Append(LIBS = ['fltk', 'fltk_images'])
 
 # HawkNL
@@ -161,7 +163,6 @@
 SConscript('src/SConscript')
 SConscript('epi/SConscript')
 SConscript('deh_edge/SConscript.edge')
-SConscript('glbsp/SConscript.edge')
 SConscript('lzo/SConscript')
 # SConscript('humidity/SConscript.edge')

diff -urNP edge-1.29-source/src/l_glbsp.cc edge-1.29-source_work/src/l_glbsp.cc
--- edge-1.29-source/src/l_glbsp.cc	2007-05-28 08:59:34.000000000 +0200
+++ edge-1.29-source_work/src/l_glbsp.cc	2007-08-27 23:47:39.000000000 +0200
@@ -30,7 +30,11 @@
 #include "version.h"
 
 #include "epi/epi.h"
+#ifdef HAVE_GLOBAL_GLBSP_H
+#include "glbsp.h"
+#else
 #include "glbsp/src/glbsp.h"
+#endif
 
 #include <stdio.h>
 #include <stdarg.h>
diff -urNP edge-1.29-source/src/SConscript edge-1.29-source_work/src/SConscript
--- edge-1.29-source/src/SConscript	2007-04-16 14:58:57.000000000 +0200
+++ edge-1.29-source_work/src/SConscript	2007-08-27 23:46:41.000000000 +0200
@@ -10,6 +10,11 @@
 
 env.Append(CCFLAGS = ['-ffast-math'])
 
+conf = Configure(env) 
+if conf.CheckCXXHeader('glbsp.h'): 
+  env.Append(CCFLAGS = ['-DHAVE_GLOBAL_GLBSP_H']) 
+env = conf.Finish()
+
 # allow #include "epi/xxxx.h" to find the correct file
 env.Append(CPPPATH = ['#.'])
 
