--- lux/CMakeLists.txt.orig	2009-07-13 13:33:06.411079974 +0200
+++ lux/CMakeLists.txt	2009-07-13 13:33:50.414430820 +0200
@@ -262,7 +262,7 @@
 #############################################################################
 #############################################################################
 # Dade - default compiler options
-ADD_DEFINITIONS(-O3 -msse2 -mfpmath=sse -ftree-vectorize -funroll-loops -Wall -DLUX_USE_OPENGL -DHAVE_PTHREAD_H)
+ADD_DEFINITIONS(-Wall -DLUX_USE_OPENGL -DHAVE_PTHREAD_H)
 
 
 #############################################################################
