--- bin/i7~	2007-07-11 16:21:37.000000000 +0100
+++ bin/i7	2007-07-11 16:24:15.000000000 +0100
@@ -3,7 +3,7 @@
 use Cwd qw(getcwd realpath);
 use Getopt::Std;
 
-$idesettings{'PREFIX'}="/usr/local";
+$idesettings{'PREFIX'}="/usr";
 $idesettings{'NAME'}="";
 
 $VERSION = "4U65";
@@ -32,9 +32,9 @@
 $idesettings{'BROWSER'} = $ENV{'BROWSER'};
 $idesettings{'BROWSERBG'}=0;
 
-$idesettings{'ZTERP'}="$idesettings{'PREFIX'}/libexec/dumb-frotz";
+$idesettings{'ZTERP'}="/usr/games/bin/frotz";
 $idesettings{'ZTERPBG'}=0;
-$idesettings{'GTERP'}="$idesettings{'PREFIX'}/libexec/dumb-glulxe";
+$idesettings{'GTERP'}="/usr/games/bin/glulxe";
 $idesettings{'GTERPBG'}=0;
 
 $CLEANUP = "";
