diff --git a/Makefile b/Makefile
index 677d59f..a2565be 100644
--- a/Makefile
+++ b/Makefile
@@ -23,7 +23,6 @@ install:
 	$(INSTALL_DATA) $(CONFFILE) $(DESTDIR)$(etcdir)/etckeeper/etckeeper.conf
 	$(INSTALL_EXE) etckeeper $(DESTDIR)$(bindir)/etckeeper
 	$(INSTALL_DATA) etckeeper.8 $(DESTDIR)$(mandir)/man8/etckeeper.8
-	$(INSTALL_DATA) bash_completion $(DESTDIR)$(etcdir)/bash_completion.d/etckeeper
 ifeq ($(HIGHLEVEL_PACKAGE_MANAGER),apt)
 	$(INSTALL_DATA) apt.conf $(DESTDIR)$(etcdir)/apt/apt.conf.d/05etckeeper
 	mkdir -p $(DESTDIR)$(etcdir)/cruft/filters-unex
@@ -36,7 +35,6 @@ ifeq ($(HIGHLEVEL_PACKAGE_MANAGER),yum)
 	$(INSTALL_DATA) yum-etckeeper.py $(DESTDIR)$(prefix)/lib/yum-plugins/etckeeper.py
 	$(INSTALL_DATA) yum-etckeeper.conf $(DESTDIR)$(etcdir)/yum/pluginconf.d/etckeeper.conf
 endif
-	-./etckeeper-bzr/__init__.py install --root=$(DESTDIR) ${PYTHON_INSTALL_OPTS} || echo "** bzr support not installed"
 	echo "** installation successful"
 
 clean: etckeeper.spec
diff --git a/etckeeper.conf b/etckeeper.conf
index c95f377..d346d21 100644
--- a/etckeeper.conf
+++ b/etckeeper.conf
@@ -31,8 +31,12 @@ DARCS_COMMIT_OPTIONS="-a"
 
 # The high-level package manager that's being used.
 # (apt, pacman-g2, yum etc)
-HIGHLEVEL_PACKAGE_MANAGER=apt
+# For gentoo this is emerge
+#HIGHLEVEL_PACKAGE_MANAGER=apt
+HIGHLEVEL_PACKAGE_MANAGER=emerge
 
 # The low-level package manager that's being used.
 # (dpkg, rpm, pacman-g2, etc)
-LOWLEVEL_PACKAGE_MANAGER=dpkg
+# For gentoo this is equery
+#LOWLEVEL_PACKAGE_MANAGER=dpkg
+LOWLEVEL_PACKAGE_MANAGER=equery
diff --git a/list-installed.d/50list-installed b/list-installed.d/50list-installed
index f3f4eed..6c463ec 100755
--- a/list-installed.d/50list-installed
+++ b/list-installed.d/50list-installed
@@ -8,3 +8,10 @@ if [ "$LOWLEVEL_PACKAGE_MANAGER" = dpkg ]; then
 elif [ "$LOWLEVEL_PACKAGE_MANAGER" = rpm ]; then
 	rpm -qa --qf "%|epoch?{%{epoch}}:{0}|:%{name}-%{version}-%{release}.%{arch}\n" | sort
 fi
+
+if [ "$LOWLEVEL_PACKAGE_MANAGER" = equery ]; then
+	equery list '*' | cut -f1
+fi
+if [ "$LOWLEVEL_PACKAGE_MANAGER" = portage ]; then
+	cd /var/db/pkg && ls -d */*
+fi
diff --git a/update-ignore.d/01update-ignore b/update-ignore.d/01update-ignore
index 6d583eb..e110611 100755
--- a/update-ignore.d/01update-ignore
+++ b/update-ignore.d/01update-ignore
@@ -99,6 +99,10 @@ writefile () {
 		ignore "*.pacorig"
 		ignore "*.pacsave"
 		nl
+	elif [ "$LOWLEVEL_PACKAGE_MANAGER" = "equery" ]; then
+		comment "new and old versions of conffiles, stored by emerge"
+		ignore "*.ebuild*"
+		nl
 	fi
 	
 	comment "old versions of files"
