--- xnecview.orig.c	2006-03-26 16:56:21.000000000 +0200
+++ xnecview.c	2006-06-13 22:01:07.000000000 +0200
@@ -14,6 +14,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <math.h>
+#include <locale.h>
 
 #ifndef NO_GETOPT
   #include <getopt.h>
@@ -371,6 +372,7 @@
       usage();
    }
    initX(&argc,argv);
+   setlocale(LC_NUMERIC,"C");
    inputfilename=NULL;
 
    wires=mymalloc(maxwires*sizeof(Wire));
