--- configure.ac
+++ configure.ac
@@ -507,13 +507,14 @@
 	fi
 
 	extra_LIBS_clanSound="$extra_LIBS_clanSound $sound_libs"
-	AM_CONDITIONAL(ALSA, test x$have_alsa = xyes)
 	fi
 
 	if test "$enable_clanSound" = "auto"; then enable_clanSound=yes; fi
 
 fi
 
+AM_CONDITIONAL(ALSA, test x$have_alsa = xyes)
+
 if test "$enable_clanSound" != "no"; then
 	enable_clanSound=yes
 	echo "Checking for clanMikmod stuff"

--- Documentation/Makefile.am
+++ Documentation/Makefile.am
@@ -1,4 +1,4 @@
-pkgdatadir = $(datadir)/doc/@PACKAGE@-@LT_RELEASE@
+pkgdatadir = $(datadir)/doc/@PACKAGE@-@PACKAGE_VERSION@
 
 SUBDIRS = @ClanLib_add_docs@
 
--- Documentation/Overview/Makefile.am
+++ Documentation/Overview/Makefile.am
@@ -1,4 +1,4 @@
-pkgdatadir  = $(datadir)/doc/@PACKAGE@-@LT_RELEASE@/Overview
+pkgdatadir  = $(datadir)/doc/@PACKAGE@-@PACKAGE_VERSION@/Overview
 
 XML_FILES  = $(wildcard $(srcdir)/*.xml)
 HTML_FILES = $(patsubst %.xml,%.html, $(notdir $(XML_FILES)))

--- Documentation/Reference/Makefile.am
+++ Documentation/Reference/Makefile.am
@@ -1,6 +1,6 @@
 EXTRA_DIST = $(wildcart images/*.png)
 
-HTML_PREFIX = $(datadir)/doc/@PACKAGE@-@LT_RELEASE@
+HTML_PREFIX = $(datadir)/doc/@PACKAGE@-@PACKAGE_VERSION@
 
 all-local:
 
@@ -10,7 +10,7 @@
 	mkdir doxyoutput;
 	mkdir reftest
 	doxygen ./clanlib.doxygen
-	make -C ../Utilities/ReferenceDocs
+	$(MAKE) -C ../Utilities/ReferenceDocs
 	../Utilities/ReferenceDocs/ReferenceDocs
 
 install-html:
