diff -pNura luvi-2.14.0.org/cmake/Modules/FindLuaJIT.cmake luvi-2.14.0/cmake/Modules/FindLuaJIT.cmake
--- luvi-2.14.0.org/cmake/Modules/FindLuaJIT.cmake	2022-07-13 17:57:28.000000000 -0700
+++ luvi-2.14.0/cmake/Modules/FindLuaJIT.cmake	2024-04-27 14:17:17.021174464 -0700
@@ -4,8 +4,8 @@
 #  LUAJIT_LIBRARIES
 #  LUAJIT_INCLUDE_DIR, where to find lua.h
 
-FIND_PATH(LUAJIT_INCLUDE_DIR NAMES lua.h PATH_SUFFIXES luajit-2.0)
-FIND_LIBRARY(LUAJIT_LIBRARIES NAMES luajit-5.1 PATH_SUFFIXES luajit-2.0)
+FIND_PATH(LUAJIT_INCLUDE_DIR NAMES lua.h PATH_SUFFIXES luajit-2.0 luajit-2.1)
+FIND_LIBRARY(LUAJIT_LIBRARIES NAMES luajit-5.1 PATH_SUFFIXES luajit-2.0 luajit-2.1)
 
 INCLUDE(FindPackageHandleStandardArgs)
 FIND_PACKAGE_HANDLE_STANDARD_ARGS(LUAJIT DEFAULT_MSG LUAJIT_LIBRARIES LUAJIT_INCLUDE_DIR)
diff -pNura luvi-2.14.0.org/deps/lrexlib.cmake luvi-2.14.0/deps/lrexlib.cmake
--- luvi-2.14.0.org/deps/lrexlib.cmake	2022-07-13 17:57:28.000000000 -0700
+++ luvi-2.14.0/deps/lrexlib.cmake	2024-04-27 14:49:06.400015618 -0700
@@ -4,7 +4,7 @@ include_directories(
   ${LREXLIB_DIR}/src
 )
 
-add_library(lrexlib
+add_library(lrexlib STATIC
   ${LREXLIB_DIR}/src/common.c
   ${LREXLIB_DIR}/src/pcre/lpcre.c
   ${LREXLIB_DIR}/src/pcre/lpcre_f.c
diff -pNura luvi-2.14.0.org/deps/lua-openssl.cmake luvi-2.14.0/deps/lua-openssl.cmake
--- luvi-2.14.0.org/deps/lua-openssl.cmake	2022-07-13 17:57:28.000000000 -0700
+++ luvi-2.14.0/deps/lua-openssl.cmake	2024-04-27 14:49:24.636698312 -0700
@@ -7,6 +7,7 @@ include_directories(
   ${CMAKE_BINARY_DIR}/include
   ${LUA_OPENSSL_DIR}/deps/auxiliar
   ${LUA_OPENSSL_DIR}/deps/lua-compat
+  ${LUA_OPENSSL_DIR}/deps/lua-compat/c-api
   ${LUA_OPENSSL_DIR}/src
 )
 
@@ -24,7 +25,7 @@ else()
   add_definitions(-DPTHREADS)
 endif()
 
-add_library(lua_openssl
+add_library(lua_openssl STATIC
   ${LUA_OPENSSL_DIR}/deps/auxiliar/auxiliar.c
   ${LUA_OPENSSL_DIR}/deps/auxiliar/subsidiar.c
   ${LUA_OPENSSL_DIR}/src/asn1.c
diff -pNura luvi-2.14.0.org/deps/lua-zlib.cmake luvi-2.14.0/deps/lua-zlib.cmake
--- luvi-2.14.0.org/deps/lua-zlib.cmake	2022-07-13 17:57:28.000000000 -0700
+++ luvi-2.14.0/deps/lua-zlib.cmake	2024-04-27 14:47:48.666613955 -0700
@@ -9,7 +9,7 @@ else ()
   set(LUA_ZLIB_LIB ${ZLIB_STATIC})
 endif()
 
-add_library(lua_zlib
+add_library(lua_zlib STATIC
   ${LUA_ZLIB_DIR}/lua_zlib.c
   ${LUA_ZLIB_DIR}/zlib.def
 )
