diff -Nuar classpath-0.97.1_orig/native/jawt/Makefile.in classpath-0.97.1/native/jawt/Makefile.in
--- classpath-0.97.1_orig/native/jawt/Makefile.in	2008-03-11 18:09:29.000000000 -0400
+++ classpath-0.97.1/native/jawt/Makefile.in	2008-03-23 21:16:48.000000000 -0400
@@ -362,6 +362,7 @@
 	test -z "$(nativeexeclibdir)" || $(mkdir_p) "$(DESTDIR)$(nativeexeclibdir)"
 	@list='$(nativeexeclib_LTLIBRARIES)'; for p in $$list; do \
 	  if test -f $$p; then \
+	    cp .libs/$$p .libs/$${p}i; \
 	    f=$(am__strip_dir) \
 	    echo " $(LIBTOOL) --mode=install $(nativeexeclibLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) '$$p' '$(DESTDIR)$(nativeexeclibdir)/$$f'"; \
 	    $(LIBTOOL) --mode=install $(nativeexeclibLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) "$$p" "$(DESTDIR)$(nativeexeclibdir)/$$f"; \
diff -Nuar classpath-0.97.1_orig/native/jni/gconf-peer/Makefile.in classpath-0.97.1/native/jni/gconf-peer/Makefile.in
--- classpath-0.97.1_orig/native/jni/gconf-peer/Makefile.in	2008-03-11 18:09:32.000000000 -0400
+++ classpath-0.97.1/native/jni/gconf-peer/Makefile.in	2008-03-23 21:15:53.000000000 -0400
@@ -357,6 +357,7 @@
 	test -z "$(nativeexeclibdir)" || $(mkdir_p) "$(DESTDIR)$(nativeexeclibdir)"
 	@list='$(nativeexeclib_LTLIBRARIES)'; for p in $$list; do \
 	  if test -f $$p; then \
+	    cp .libs/$$p .libs/$${p}i; \
 	    f=$(am__strip_dir) \
 	    echo " $(LIBTOOL) --mode=install $(nativeexeclibLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) '$$p' '$(DESTDIR)$(nativeexeclibdir)/$$f'"; \
 	    $(LIBTOOL) --mode=install $(nativeexeclibLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) "$$p" "$(DESTDIR)$(nativeexeclibdir)/$$f"; \
diff -Nuar classpath-0.97.1_orig/native/jni/gstreamer-peer/Makefile.in classpath-0.97.1/native/jni/gstreamer-peer/Makefile.in
--- classpath-0.97.1_orig/native/jni/gstreamer-peer/Makefile.in	2008-03-11 18:09:32.000000000 -0400
+++ classpath-0.97.1/native/jni/gstreamer-peer/Makefile.in	2008-03-23 21:15:59.000000000 -0400
@@ -381,6 +381,7 @@
 	test -z "$(nativeexeclibdir)" || $(mkdir_p) "$(DESTDIR)$(nativeexeclibdir)"
 	@list='$(nativeexeclib_LTLIBRARIES)'; for p in $$list; do \
 	  if test -f $$p; then \
+	    cp .libs/$$p .libs/$${p}i; \
 	    f=$(am__strip_dir) \
 	    echo " $(LIBTOOL) --mode=install $(nativeexeclibLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) '$$p' '$(DESTDIR)$(nativeexeclibdir)/$$f'"; \
 	    $(LIBTOOL) --mode=install $(nativeexeclibLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) "$$p" "$(DESTDIR)$(nativeexeclibdir)/$$f"; \
diff -Nuar classpath-0.97.1_orig/native/jni/gtk-peer/Makefile.in classpath-0.97.1/native/jni/gtk-peer/Makefile.in
--- classpath-0.97.1_orig/native/jni/gtk-peer/Makefile.in	2008-03-11 18:09:33.000000000 -0400
+++ classpath-0.97.1/native/jni/gtk-peer/Makefile.in	2008-03-23 21:14:57.000000000 -0400
@@ -450,6 +450,7 @@
 	test -z "$(nativeexeclibdir)" || $(mkdir_p) "$(DESTDIR)$(nativeexeclibdir)"
 	@list='$(nativeexeclib_LTLIBRARIES)'; for p in $$list; do \
 	  if test -f $$p; then \
+	    cp .libs/$$p .libs/$${p}i; \
 	    f=$(am__strip_dir) \
 	    echo " $(LIBTOOL) --mode=install $(nativeexeclibLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) '$$p' '$(DESTDIR)$(nativeexeclibdir)/$$f'"; \
 	    $(LIBTOOL) --mode=install $(nativeexeclibLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) "$$p" "$(DESTDIR)$(nativeexeclibdir)/$$f"; \
diff -Nuar classpath-0.97.1_orig/native/jni/java-io/Makefile.in classpath-0.97.1/native/jni/java-io/Makefile.in
--- classpath-0.97.1_orig/native/jni/java-io/Makefile.in	2008-03-11 18:09:34.000000000 -0400
+++ classpath-0.97.1/native/jni/java-io/Makefile.in	2008-03-23 21:16:25.000000000 -0400
@@ -363,6 +363,7 @@
 	test -z "$(nativeexeclibdir)" || $(mkdir_p) "$(DESTDIR)$(nativeexeclibdir)"
 	@list='$(nativeexeclib_LTLIBRARIES)'; for p in $$list; do \
 	  if test -f $$p; then \
+	    cp .libs/$$p .libs/$${p}i; \
 	    f=$(am__strip_dir) \
 	    echo " $(LIBTOOL) --mode=install $(nativeexeclibLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) '$$p' '$(DESTDIR)$(nativeexeclibdir)/$$f'"; \
 	    $(LIBTOOL) --mode=install $(nativeexeclibLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) "$$p" "$(DESTDIR)$(nativeexeclibdir)/$$f"; \
diff -Nuar classpath-0.97.1_orig/native/jni/java-lang/Makefile.in classpath-0.97.1/native/jni/java-lang/Makefile.in
--- classpath-0.97.1_orig/native/jni/java-lang/Makefile.in	2008-03-11 18:09:34.000000000 -0400
+++ classpath-0.97.1/native/jni/java-lang/Makefile.in	2008-03-23 21:16:30.000000000 -0400
@@ -382,6 +382,7 @@
 	test -z "$(nativeexeclibdir)" || $(mkdir_p) "$(DESTDIR)$(nativeexeclibdir)"
 	@list='$(nativeexeclib_LTLIBRARIES)'; for p in $$list; do \
 	  if test -f $$p; then \
+	    cp .libs/$$p .libs/$${p}i; \
 	    f=$(am__strip_dir) \
 	    echo " $(LIBTOOL) --mode=install $(nativeexeclibLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) '$$p' '$(DESTDIR)$(nativeexeclibdir)/$$f'"; \
 	    $(LIBTOOL) --mode=install $(nativeexeclibLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) "$$p" "$(DESTDIR)$(nativeexeclibdir)/$$f"; \
diff -Nuar classpath-0.97.1_orig/native/jni/java-net/Makefile.in classpath-0.97.1/native/jni/java-net/Makefile.in
--- classpath-0.97.1_orig/native/jni/java-net/Makefile.in	2008-03-11 18:09:34.000000000 -0400
+++ classpath-0.97.1/native/jni/java-net/Makefile.in	2008-03-23 21:15:36.000000000 -0400
@@ -383,6 +383,7 @@
 	test -z "$(nativeexeclibdir)" || $(mkdir_p) "$(DESTDIR)$(nativeexeclibdir)"
 	@list='$(nativeexeclib_LTLIBRARIES)'; for p in $$list; do \
 	  if test -f $$p; then \
+	    cp .libs/$$p .libs/$${p}i; \
 	    f=$(am__strip_dir) \
 	    echo " $(LIBTOOL) --mode=install $(nativeexeclibLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) '$$p' '$(DESTDIR)$(nativeexeclibdir)/$$f'"; \
 	    $(LIBTOOL) --mode=install $(nativeexeclibLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) "$$p" "$(DESTDIR)$(nativeexeclibdir)/$$f"; \
diff -Nuar classpath-0.97.1_orig/native/jni/java-nio/Makefile.in classpath-0.97.1/native/jni/java-nio/Makefile.in
--- classpath-0.97.1_orig/native/jni/java-nio/Makefile.in	2008-03-11 18:09:35.000000000 -0400
+++ classpath-0.97.1/native/jni/java-nio/Makefile.in	2008-03-23 21:16:12.000000000 -0400
@@ -382,6 +382,7 @@
 	test -z "$(nativeexeclibdir)" || $(mkdir_p) "$(DESTDIR)$(nativeexeclibdir)"
 	@list='$(nativeexeclib_LTLIBRARIES)'; for p in $$list; do \
 	  if test -f $$p; then \
+	    cp .libs/$$p .libs/$${p}i; \
 	    f=$(am__strip_dir) \
 	    echo " $(LIBTOOL) --mode=install $(nativeexeclibLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) '$$p' '$(DESTDIR)$(nativeexeclibdir)/$$f'"; \
 	    $(LIBTOOL) --mode=install $(nativeexeclibLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) "$$p" "$(DESTDIR)$(nativeexeclibdir)/$$f"; \
diff -Nuar classpath-0.97.1_orig/native/jni/java-util/Makefile.in classpath-0.97.1/native/jni/java-util/Makefile.in
--- classpath-0.97.1_orig/native/jni/java-util/Makefile.in	2008-03-11 18:09:35.000000000 -0400
+++ classpath-0.97.1/native/jni/java-util/Makefile.in	2008-03-23 21:16:19.000000000 -0400
@@ -354,6 +354,7 @@
 	test -z "$(nativeexeclibdir)" || $(mkdir_p) "$(DESTDIR)$(nativeexeclibdir)"
 	@list='$(nativeexeclib_LTLIBRARIES)'; for p in $$list; do \
 	  if test -f $$p; then \
+	    cp .libs/$$p .libs/$${p}i; \
 	    f=$(am__strip_dir) \
 	    echo " $(LIBTOOL) --mode=install $(nativeexeclibLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) '$$p' '$(DESTDIR)$(nativeexeclibdir)/$$f'"; \
 	    $(LIBTOOL) --mode=install $(nativeexeclibLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) "$$p" "$(DESTDIR)$(nativeexeclibdir)/$$f"; \
diff -Nuar classpath-0.97.1_orig/native/jni/midi-alsa/Makefile.in classpath-0.97.1/native/jni/midi-alsa/Makefile.in
--- classpath-0.97.1_orig/native/jni/midi-alsa/Makefile.in	2008-03-11 18:09:36.000000000 -0400
+++ classpath-0.97.1/native/jni/midi-alsa/Makefile.in	2008-03-23 21:15:12.000000000 -0400
@@ -362,6 +362,7 @@
 	test -z "$(nativeexeclibdir)" || $(mkdir_p) "$(DESTDIR)$(nativeexeclibdir)"
 	@list='$(nativeexeclib_LTLIBRARIES)'; for p in $$list; do \
 	  if test -f $$p; then \
+	    cp .libs/$$p .libs/$${p}i; \
 	    f=$(am__strip_dir) \
 	    echo " $(LIBTOOL) --mode=install $(nativeexeclibLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) '$$p' '$(DESTDIR)$(nativeexeclibdir)/$$f'"; \
 	    $(LIBTOOL) --mode=install $(nativeexeclibLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) "$$p" "$(DESTDIR)$(nativeexeclibdir)/$$f"; \
diff -Nuar classpath-0.97.1_orig/native/jni/midi-dssi/Makefile.in classpath-0.97.1/native/jni/midi-dssi/Makefile.in
--- classpath-0.97.1_orig/native/jni/midi-dssi/Makefile.in	2008-03-11 18:09:36.000000000 -0400
+++ classpath-0.97.1/native/jni/midi-dssi/Makefile.in	2008-03-23 21:16:35.000000000 -0400
@@ -365,6 +365,7 @@
 	test -z "$(nativeexeclibdir)" || $(mkdir_p) "$(DESTDIR)$(nativeexeclibdir)"
 	@list='$(nativeexeclib_LTLIBRARIES)'; for p in $$list; do \
 	  if test -f $$p; then \
+	    cp .libs/$$p .libs/$${p}i; \
 	    f=$(am__strip_dir) \
 	    echo " $(LIBTOOL) --mode=install $(nativeexeclibLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) '$$p' '$(DESTDIR)$(nativeexeclibdir)/$$f'"; \
 	    $(LIBTOOL) --mode=install $(nativeexeclibLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) "$$p" "$(DESTDIR)$(nativeexeclibdir)/$$f"; \
diff -Nuar classpath-0.97.1_orig/native/jni/qt-peer/Makefile.in classpath-0.97.1/native/jni/qt-peer/Makefile.in
--- classpath-0.97.1_orig/native/jni/qt-peer/Makefile.in	2008-03-11 18:09:37.000000000 -0400
+++ classpath-0.97.1/native/jni/qt-peer/Makefile.in	2008-03-23 21:15:28.000000000 -0400
@@ -445,6 +445,7 @@
 	test -z "$(nativeexeclibdir)" || $(mkdir_p) "$(DESTDIR)$(nativeexeclibdir)"
 	@list='$(nativeexeclib_LTLIBRARIES)'; for p in $$list; do \
 	  if test -f $$p; then \
+	    cp .libs/$$p .libs/$${p}i; \
 	    f=$(am__strip_dir) \
 	    echo " $(LIBTOOL) --mode=install $(nativeexeclibLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) '$$p' '$(DESTDIR)$(nativeexeclibdir)/$$f'"; \
 	    $(LIBTOOL) --mode=install $(nativeexeclibLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) "$$p" "$(DESTDIR)$(nativeexeclibdir)/$$f"; \
diff -Nuar classpath-0.97.1_orig/native/jni/xmlj/Makefile.in classpath-0.97.1/native/jni/xmlj/Makefile.in
--- classpath-0.97.1_orig/native/jni/xmlj/Makefile.in	2008-03-11 18:09:38.000000000 -0400
+++ classpath-0.97.1/native/jni/xmlj/Makefile.in	2008-03-23 21:16:06.000000000 -0400
@@ -375,6 +375,7 @@
 	test -z "$(nativeexeclibdir)" || $(mkdir_p) "$(DESTDIR)$(nativeexeclibdir)"
 	@list='$(nativeexeclib_LTLIBRARIES)'; for p in $$list; do \
 	  if test -f $$p; then \
+	    cp .libs/$$p .libs/$${p}i; \
 	    f=$(am__strip_dir) \
 	    echo " $(LIBTOOL) --mode=install $(nativeexeclibLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) '$$p' '$(DESTDIR)$(nativeexeclibdir)/$$f'"; \
 	    $(LIBTOOL) --mode=install $(nativeexeclibLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) "$$p" "$(DESTDIR)$(nativeexeclibdir)/$$f"; \
diff -Nuar classpath-0.97.1_orig/native/plugin/Makefile.in classpath-0.97.1/native/plugin/Makefile.in
--- classpath-0.97.1_orig/native/plugin/Makefile.in	2008-03-11 18:09:38.000000000 -0400
+++ classpath-0.97.1/native/plugin/Makefile.in	2008-03-23 21:16:41.000000000 -0400
@@ -359,6 +359,7 @@
 	test -z "$(nativeexeclibdir)" || $(mkdir_p) "$(DESTDIR)$(nativeexeclibdir)"
 	@list='$(nativeexeclib_LTLIBRARIES)'; for p in $$list; do \
 	  if test -f $$p; then \
+	    cp .libs/$$p .libs/$${p}i; \
 	    f=$(am__strip_dir) \
 	    echo " $(LIBTOOL) --mode=install $(nativeexeclibLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) '$$p' '$(DESTDIR)$(nativeexeclibdir)/$$f'"; \
 	    $(LIBTOOL) --mode=install $(nativeexeclibLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) "$$p" "$(DESTDIR)$(nativeexeclibdir)/$$f"; \
