--- /var/tmp/portage/dev-util/grit-0.8.3/work/grit_src/srcgrit/cli.cpp	2008-05-12 14:39:06.000000000 +0200
+++ /home/lolwut/grit_src/srcgrit/cli.cpp	2009-09-08 19:16:24.000000000 +0200
@@ -9,6 +9,12 @@
 #include "cli.h"
 #include <string.h>
 
+// Date: 08/09/2009
+// User: Kim Nilsson (Gentoo Linux amd64)
+// Issue: Compiler (GCC 4.4.1) needed to have strtoul defined.
+// Fix: Included stdlib.h
+#include <stdlib.h>
+
 //! Find a key in a string array.
 /*!	\return key index or length of array if not found.
 */
