--- /var/tmp/portage/dev-util/grit-0.8.3/work/grit_src/srcgrit/grit_main.cpp	2008-12-07 17:24:08.000000000 +0100
+++ /home/lolwut/grit_src/srcgrit/grit_main.cpp	2009-09-08 19:13:36.000000000 +0200
@@ -49,7 +49,7 @@
 void grit_dump(GritRec *gr, FILE *fp);
 void grit_dump_short(GritRec *gr, FILE *fp, const char *pre);
 
-// typedef std::vector<char*> strvec;
+//typedef std::vector<char*> strvec;
 
 
 // --------------------------------------------------------------------
@@ -786,11 +786,20 @@
 		if(!files.empty())
 		{
 			// find first non-path
-			for(ii=0; ii<argc; ii++)
-				if(args[ii][0]=='-')
+			
+			// Date: 08/09/2009
+			// User: Kim Nilsson ( Gentoo Linux amd64 )
+			// Issue: Would not compile ( x86_64-pc-linux-gnu GCC 4.4.1)
+			// Fix: Changed for-loop to use an iterator so the insertion would not complain.
+			strvec::iterator iter;
+			for( iter = args.begin(); iter != args.end(); iter++ )
+			{
+				if( ( *iter )[0] == '-' )
+				{
+					args.insert( iter, files.begin(), files.end() );
 					break;
-
-			args.insert(&args[ii], files.begin(), files.end());
+				}
+			}
 		}
 
 	}
