diff -rN -u old-kaya/libs/Makefile.in new-kaya/libs/Makefile.in
--- old-kaya/libs/Makefile.in   2008-05-29 21:50:16.000000000 +0100
+++ new-kaya/libs/Makefile.in   2008-05-29 21:50:18.000000000 +0100
@@ -41,6 +41,8 @@
        rm -f *~ Makefile
        rm -rf autom4te.cache
 
+Multicore.o: Threads.ki
+Threads.o: thread_glue.h
 MyDB.o: my_inter.h ../stdlib/DB.ki
 PostgresDB.o: pg_inter.h ../stdlib/DB.ki
 SQLiteDB.o: sqlite_inter.h ../stdlib/DB.ki
