diff -up kaya-0.5.0/configure.ac.conf kaya-0.5.0/configure.ac
--- kaya-0.5.0/configure.ac.conf	2008-05-26 12:53:59.000000000 +0200
+++ kaya-0.5.0/configure.ac	2008-06-05 20:21:32.000000000 +0200
@@ -220,7 +220,7 @@ PGINC=""
 AC_ARG_DISABLE([postgres],
 	AS_HELP_STRING([--disable-postgres],
 	[Disable Postgres database support]),
-	[AC_CHECK_HEADER([postgresql/libpq-fe.h],[PGINC="-I/usr/include/postgresql"])
+	[AC_CHECK_HEADER([libpq-fe.h],[PGINC="-I/usr/include"])
 	AC_CHECK_LIB(pq, PQconnectdb, 
 		[PGSTUB=""
 	         PGMAN="PostgresDB.libs"
@@ -247,6 +247,7 @@ AC_ARG_DISABLE([mysql],
 	AS_HELP_STRING([--disable-mysql],
 	[Disable MySQL database support]),
 [AC_CHECK_HEADER([mysql/mysql.h])
+LIBS="$LIBS -L${libdir}/mysql"
 AC_CHECK_LIB(mysqlclient, mysql_init, 
 	[AC_MSG_NOTICE([Found libmysqlclient, is it recent enough?])
 	 AC_CHECK_LIB(mysqlclient, mysql_stmt_init, 
