--- Makefile	2011-08-25 04:25:02.000000000 +1000
+++ Makefile	2011-08-25 04:26:13.000000000 +1000
@@ -3,14 +3,7 @@
 RPM_BUILDROOT=rpm.buildroot
 
 waitmax: waitmax.c
-	@if type diet >/dev/null ; then \
-	    echo "Compiling with diet (please ignore diet-warnings)..." ; \
-	    diet gcc -s -o waitmax waitmax.c ; \
-	else \
-	    echo "Compiling with normal gcc..." ; \
-	    gcc -s -o waitmax waitmax.c ; \
-	fi
-	echo "Fine. Typing 'make install' as root now will install into /usr/bin"
+	$(CC) $(CFLAGS) -o waitmax waitmax.c $(LDFLAGS)
 
 install:
 	mkdir -p $(DESTDIR)/usr/bin
