diff -Nur a/cmake/SeastarDependencies.cmake b/cmake/SeastarDependencies.cmake
--- a/cmake/SeastarDependencies.cmake	2025-05-26 08:29:20.000000000 -0700
+++ b/cmake/SeastarDependencies.cmake	2025-06-05 11:11:20.985475734 -0700
@@ -87,7 +87,7 @@
     seastar_find_dep (dpdk)
   endif()
   seastar_find_dep (fmt 8.1.1 REQUIRED)
-  seastar_find_dep (lz4 1.7.3 REQUIRED)
+  seastar_find_dep (lz4 REQUIRED)
   seastar_find_dep (GnuTLS 3.3.26 REQUIRED)
   if (Seastar_IO_URING)
     seastar_find_dep (LibUring 2.0 REQUIRED)
