--- DH1080.c
+++ DH1080.c
@@ -30,7 +30,7 @@
 {
 	unsigned char raw_buf[160];
 	unsigned long seed;
-	int len;
+	size_t len;
 
 	mpz_t mpz_privkey, mpz_prime, mpz_pubkey, mpz_base;
 	gmp_randstate_t randstate;
@@ -81,7 +81,8 @@
 //         HisPubKey = the secret key
 int DH1080_comp(char *MyPrivKey, char *HisPubKey)
 {
-	int i=0, len;
+	int i=0;
+	size_t len;
 	unsigned char SHA256digest[35], base64_tmp[160];
 	mpz_t mpz_myPrivkey, mpz_hisPubkey, mpz_prime, mpz_theKey;
 
