--- SHA-256.c
+++ SHA-256.c
@@ -12,6 +12,7 @@
  * */
 
 #include <stdio.h>
+#include <string.h>
 
 typedef struct {
     unsigned int state[8], length, curlen;
--- DH1080.c
+++ DH1080.c
@@ -12,6 +12,7 @@
 
 
 #include <stdio.h>
+#include <string.h>
 #include <time.h>
 #include "DH1080.h"
 
--- blowfish.c
+++ blowfish.c
@@ -1,3 +1,5 @@
+#include <stdlib.h>
+#include <string.h>
 #include "blowfish.h"
 
 /* #define S(x,i) (bf_S[i][x.w.byte##i]) */
--- base64.c
+++ base64.c
@@ -1,3 +1,5 @@
+#include <string.h>
+
 /*
     Public Base64 conversion tables
 */
