--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -164,6 +164,7 @@
         "${_sp_src}/third_party/protobuf-lite" # for "google/protobuf/port_def.inc"
         "${_sp_src}/third_party/" # for "absl/strings/string_view.h" and "darts_clone/darts.h"
         "${_sp_src}" # for "third_party/absl/strings/string_view.h"
+        "${sentencepiece_BINARY_DIR}" # for "config.h"
         )
     else()
       target_include_directories(${TARGET_NAME} SYSTEM PRIVATE
@@ -498,12 +498,6 @@
   install(FILES ${extra_libs} DESTINATION ${extra_libs_location} COMPONENT openvino_tokenizers)
 endif()
 
-install(FILES "${openvino_tokenizers_SOURCE_DIR}/LICENSE"
-              "${openvino_tokenizers_SOURCE_DIR}/third-party-programs.txt"
-              "${openvino_tokenizers_SOURCE_DIR}/README.md"
-        DESTINATION "docs/openvino_tokenizers"
-        COMPONENT openvino_tokenizers_docs)
-
 #
 # Cpack configuration
 #
