diff -Nur a/pyproject.toml b/pyproject.toml
--- a/pyproject.toml	2024-12-07 04:06:16.000000000 -0800
+++ b/pyproject.toml	2024-12-24 18:24:25.088504866 -0800
@@ -91,13 +91,12 @@
 build_type = "Release"
 config = ["Release"]
 find_python3 = true
-build_args = ["--parallel", "--target", "openvino_tokenizers"]
+build_args = ["--parallel"]
 install_args = ["--strip"]
 install_components = ["openvino_tokenizers", "openvino_tokenizers_python", "openvino_tokenizers_licenses"]
 
 [tool.py-build-cmake.cmake.options]
-OPENVINO_TOKENIZERS_INSTALL_BINDIR = "openvino_tokenizers/lib"
-OPENVINO_TOKENIZERS_INSTALL_LIBDIR = "openvino_tokenizers/lib"
+BUILD_CPP_EXTENSION = "OFF"
 
 [tool.py-build-cmake.wheel]
 python_abi = "none"
