--- midisport-firmware-1.2.orig/configure.ac	2006-03-05 15:01:45.000000000 -0600
+++ midisport-firmware-1.2/configure.ac	2010-09-13 14:15:36.000000000 -0500
@@ -23,12 +23,17 @@
 fi
 
 AC_MSG_CHECKING([for udev version])
-udev_ver=`udevinfo -V | awk '{ print $3 }'`
+udev_ver=`udevinfo -V 2>/dev/null | awk '{ print $3 }'`
 if test -n "$udev_ver"; then
   AC_MSG_RESULT([$udev_ver])
 else
-  AC_MSG_RESULT([none])
-  udev_ver="0"
+  udev_ver=`udevadm -V 2>/dev/null`
+  if test -n "$udev_ver"; then
+    AC_MSG_RESULT([$udev_ver])
+  else
+    AC_MSG_RESULT([none])
+    udev_ver="0"
+  fi
 fi
 if test "$udev_ver" -lt 057; then
   AC_MSG_ERROR([udev version 057 or later is required])
