diff -Naur elmerfem-5.4.1.orig/front/src/Makefile.am elmerfem-5.4.1/front/src/Makefile.am
--- elmerfem-5.4.1.orig/front/src/Makefile.am	2008-02-29 08:11:35.000000000 +0000
+++ elmerfem-5.4.1/front/src/Makefile.am	2008-04-27 01:24:29.000000000 +0100
@@ -85,16 +85,16 @@
 endif
 
 install-data-local:
-	@echo "Installing tcl scripts to ${prefix}/share/elmerfront/tcl"
+	@echo "Installing tcl scripts to ${DESTDIR}${prefix}/share/elmerfront/tcl"
 
-	mkdir -p "${prefix}/share/elmerfront/tcl"
-	cp -r ./tcl/* "${prefix}/share/elmerfront/tcl"
-	rm -rf `find $(prefix)/share/elmerfront/tcl -name .svn`
-
-	@echo "Installing *.edf *.rgb to ${prefix}/share/elmerfront/lib"
-	mkdir -p "${prefix}/share/elmerfront/lib"
-	cp -r ./lib/* "${prefix}/share/elmerfront/lib"
-	rm -rf `find $(prefix)/share/elmerfront/lib -name .svn`
+	mkdir -p "${DESTDIR}${prefix}/share/elmerfront/tcl"
+	cp -r ./tcl/* "${DESTDIR}${prefix}/share/elmerfront/tcl"
+	rm -rf `find ${DESTDIR}$(prefix)/share/elmerfront/tcl -name .svn`
+
+	@echo "Installing *.edf *.rgb to ${DESTDIR}${prefix}/share/elmerfront/lib"
+	mkdir -p "${DESTDIR}${prefix}/share/elmerfront/lib"
+	cp -r ./lib/* "${DESTDIR}${prefix}/share/elmerfront/lib"
+	rm -rf `find ${DESTDIR}$(prefix)/share/elmerfront/lib -name .svn`
 
 
 
