--- tos/system/AMStandard.nc-old	2006-01-10 22:32:15.000000000 +0100
+++ tos/system/AMStandard.nc	2006-01-10 22:32:43.000000000 +0100
@@ -84,7 +84,7 @@
   uint16_t counter;
   
   // Initialization of this component
-  command bool Control.init() {
+  command result_t Control.init() {
     result_t ok1, ok2;
 
     call TimerControl.init();
@@ -100,7 +100,7 @@
   }
 
   // Command to be used for power managment
-  command bool Control.start() {
+  command result_t Control.start() {
     result_t ok0 = call TimerControl.start();
     result_t ok1 = call UARTControl.start();
     result_t ok2 = call RadioControl.start();
@@ -116,7 +116,7 @@
   }
 
   
-  command bool Control.stop() {
+  command result_t Control.stop() {
     result_t ok1 = call UARTControl.stop();
     result_t ok2 = call RadioControl.stop();
     result_t ok3 = call ActivityTimer.stop();
