Modifications to fit the new libpng interface.

--- bjfilter-2.50/bjfilter/src/bjfimage.c	2011-12-18 02:02:33.000000000 +0100
+++ bjfilter-2.50/bjfilter/src/bjfimage.c	2011-12-18 02:11:14.000000000 +0100
@@ -1278,8 +1278,8 @@
 	short			tmpformat;
 	short			retbyte = 0;
 	short			bpp = 3;
-	long			width = 0;
-	long			length = 0;
+	png_uint_32		width = 0;
+	png_uint_32		length = 0;
 	long			rstep = 0;
 	long			RasterLength = 0;
 	long			i;
@@ -1332,7 +1332,7 @@
 		goto onErr;
 	}
 
-	if (setjmp (png_ptr->jmpbuf))
+	if (setjmp(png_jmpbuf(png_ptr)))
 	{
 		png_destroy_read_struct(&png_ptr, &info_ptr, (png_infopp)NULL);
 		goto onErr;
