diff -Nur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2024-12-18 08:12:40.000000000 -0800
+++ b/CMakeLists.txt	2025-02-02 17:17:48.971951917 -0800
@@ -119,8 +119,8 @@
 #
 # build the library
 #
-
-if (NOT TARGET ggml)
+find_package(ggml REQUIRED)
+if (NOT TARGET ggml::ggml)
     add_subdirectory(ggml)
     # ... otherwise assume ggml is added by a parent CMakeLists.txt
 endif()
diff -Nur a/examples/CMakeLists.txt b/examples/CMakeLists.txt
--- a/examples/CMakeLists.txt	2024-12-18 08:12:40.000000000 -0800
+++ b/examples/CMakeLists.txt	2025-02-02 17:27:48.528639817 -0800
@@ -50,6 +50,7 @@
     set(COMMON_SOURCES_FFMPEG ffmpeg-transcode.cpp)
 endif()
 
+list(APPEND COMMON_EXTRA_LIBS ggml::ggml-cpu)
 
 add_library(${TARGET} STATIC
     common.h
