diff -Nur a/examples/cli/cli.cpp b/examples/cli/cli.cpp
--- a/examples/cli/cli.cpp
+++ b/examples/cli/cli.cpp
@@ -83,7 +83,7 @@
     std::string language  = "en";
     std::string prompt;
     std::string font_path = "/System/Library/Fonts/Supplemental/Courier New Bold.ttf";
-    std::string model     = "models/ggml-base.en.bin";
+    std::string model     = "/usr/share/whisper/whisper-default.bin";
     std::string grammar;
     std::string grammar_rule;
 
