diff -ur mgeups-psp-3.0.1/src/MgeSystemTray.h mgeups-psp-3.0.1-gcc4/src/MgeSystemTray.h
--- mgeups-psp-3.0.1/src/MgeSystemTray.h	2005-06-30 14:09:27.000000000 +0200
+++ mgeups-psp-3.0.1-gcc4/src/MgeSystemTray.h	2006-03-31 22:12:21.000000000 +0200
@@ -38,7 +38,7 @@
   void setTooltip(const Glib::ustring& tip, const Glib::ustring& private_tip = Glib::ustring());
 
   // Callbacks
-  void MgeSystemTray::onUpsStatusActivate();
+  void onUpsStatusActivate();
 
   // Should the systray be visible?
   void setVisibility(bool visibility);
Seulement dans mgeups-psp-3.0.1-gcc4/src: MgeSystemTray.h~
