diff -Nur a/src/ggml-sycl/CMakeLists.txt b/src/ggml-sycl/CMakeLists.txt
--- a/src/ggml-sycl/CMakeLists.txt	2025-02-03 15:17:15.000000000 -0800
+++ b/src/ggml-sycl/CMakeLists.txt	2025-02-04 00:29:57.502576213 -0800
@@ -64,6 +64,10 @@
     find_package(MKL REQUIRED)
     target_link_libraries(ggml-sycl PRIVATE IntelSYCL::SYCL_CXX MKL::MKL MKL::MKL_SYCL)
 else()
+    find_package(PkgConfig REQUIRED)
+    pkg_search_module(mkl REQUIRED mkl-sdl)
+    target_include_directories(ggml-sycl PRIVATE ${mkl_INCLUDE_DIRS})
+
     if (GGML_SYCL_TARGET STREQUAL "INTEL")
         target_link_libraries(ggml-sycl PRIVATE sycl OpenCL mkl_core pthread m dl mkl_sycl_blas mkl_intel_ilp64 mkl_tbb_thread)
     elseif (GGML_SYCL_TARGET STREQUAL "NVIDIA")
