--- CompressedMagic.cpp	2007-12-25 05:40:49.000000000 -0500
+++ CompressedMagic.cpp.patched	2008-05-03 20:51:56.000000000 -0400
@@ -67,7 +67,8 @@
 {
 	const char *mime;
 
-	Lock lock(m_Mutex);
+	//Lock lock(m_Mutex);
+	m_Mutex.Lock();
 
 	mime = magic_buffer(m_magic, buf, len);
 
@@ -77,9 +78,11 @@
 
 		if (m_table.find(mime) != m_table.end())
 		{
+			m_Mutex.Unlock();
 			return true;
 		}
 	}
+	m_Mutex.Unlock();
 	return false;
 }
 
