diff -Nur a/csrc/cpu/CMakeLists.txt b/csrc/cpu/CMakeLists.txt
--- a/csrc/cpu/CMakeLists.txt	2024-12-19 19:33:09.000000000 -0800
+++ b/csrc/cpu/CMakeLists.txt	2024-12-28 23:06:55.427128514 -0800
@@ -193,12 +193,7 @@
 endif(USE_LIBXSMM)
 
 # setup sleef options:
-set(SLEEF_BUILD_SHARED_LIBS OFF CACHE BOOL "Build sleef as static library" FORCE)
-set(SLEEF_BUILD_DFT OFF CACHE BOOL "Don't build sleef DFT lib" FORCE)
-set(SLEEF_BUILD_GNUABI_LIBS OFF CACHE BOOL "Don't build sleef gnuabi libs" FORCE)
-set(SLEEF_BUILD_TESTS OFF CACHE BOOL "Don't build sleef tests" FORCE)
-set(SLEEF_BUILD_SCALAR_LIB OFF CACHE BOOL "libsleefscalar will be built." FORCE)
-add_subdirectory(${IPEX_CPU_CPP_THIRD_PARTY_ROOT}/sleef ${THIRD_PARTY_BUILD_PATH_NAME}/sleef EXCLUDE_FROM_ALL)
+find_package(sleef REQUIRED)
 target_link_libraries(${PLUGIN_NAME_CPU} PRIVATE sleef)
 
 add_dependencies(${PLUGIN_NAME_CPU} dnnl)
