--- src/archivers/zip/unzip.h	2011-11-20 20:30:15.555322552 +0100
+++ src/archivers/zip/unzip.h	2011-11-20 20:30:15.555322552 +0100
@@ -45,19 +45,19 @@
 extern "C" {
 #endif
 
+/* Work-around for NetBSD. Its zconf.h doesn't define OF(x) */
+#ifndef OF
+#ifdef __STDC__
+#define OF(x) x
+#else
+#define OF(x) ()
+#endif
+#endif
+
 #ifndef _ZLIB_H
 #include "zlib.h"
 #endif
 
-typedef int (ZEXPORT *INFLATEINIT2)(z_streamp strm, int windowBits, const char *version, int stream_size);
-typedef int (ZEXPORT *INFLATEINIT)(z_streamp strm, const char *version, int stream_size);
-typedef int (ZEXPORT *INFLATEEND)(z_streamp strm);
-typedef int (ZEXPORT *INFLATE)(z_streamp strm, int flush);
-typedef int (ZEXPORT *DEFLATEINIT)(z_streamp strm, int level, const char *version, int stream_size);
-typedef int (ZEXPORT *DEFLATEEND)(z_streamp strm);
-typedef int (ZEXPORT *DEFLATE)(z_streamp strm, int flush);
-typedef uLong (ZEXPORT *CRC32)(uLong crc, const Bytef *buf, uInt len);
-
 #if defined(STRICTUNZIP) || defined(STRICTZIPUNZIP)
 /* like the STRICT of WIN32, we define a pointer that cannot be converted
     from (void*) without cast */
@@ -68,8 +68,6 @@
 #endif
 
 
-#define ZIP_BZIP2 12
-
 #define UNZ_OK                                  (0)
 #define UNZ_END_OF_LIST_OF_FILE (-100)
 #define UNZ_ERRNO               (Z_ERRNO)
@@ -135,7 +133,7 @@
 */
 
 
-//extern unzFile ZEXPORT unzOpen OF((struct zfile *path));
+extern unzFile ZEXPORT unzOpen OF((struct zfile *path));
 /*
   Open a Zip file. path contain the full pathname (by example,
      on a Windows NT computer "c:\\zlib\\zlib111.zip" or on an Unix computer
