--- usr/lib64/OpenCL/vendors/intel/iocgui64.sh.orig	2012-03-21 16:59:56.000000000 -0400
+++ usr/lib64/OpenCL/vendors/intel/iocgui64.sh	2012-06-18 14:12:19.536422807 -0400
@@ -53,13 +53,13 @@
 then
     # Get architecture supported on this machine
 	grep -q -e "^flags.*avx" /proc/cpuinfo
-	retval = $?
+	retval=$?
 	if [ $retval -eq 0 ];
 	then
 	    architecture="avx"
 	else
 	    grep -q -e "^flags.*sse4_2" /proc/cpuinfo
-		retval = $?
+		retval=$?
 	    if [ $retval -eq 0 ];
 		then
 		    architecture="sse4_2"
@@ -68,7 +68,7 @@

     # Add 32 bit support when available
     java -jar "$jarpath" x64 $architecture
-	retval = $?
+	retval=$?
     if [ $retval -ne 0 ];
     then
         reporterror "could not find java runtime environment"
