commit 56c8666eb7eb6df7dcd2f2a7cd2906dc5b75d4c2
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Fri Jan 13 12:48:09 2012 -0500

    defined HAVE_MKFIFO

diff --git a/lib/sh/oslib.c b/lib/sh/oslib.c
index b3470d1..94b412c 100644
--- a/lib/sh/oslib.c
+++ b/lib/sh/oslib.c
@@ -233,6 +233,8 @@ killpg (pgrp, sig)
 }
 #endif /* !HAVE_KILLPG */
 
+#define HAVE_MKFIFO
+
 #if !defined (HAVE_MKFIFO) && defined (PROCESS_SUBSTITUTION)
 int
 mkfifo (path, mode)
