--- vmc/common/oal.py.orig	2007-07-07 21:47:58.000000000 +0200
+++ vmc/common/oal.py		2007-07-07 21:51:40.000000000 +0200
@@ -171,6 +171,13 @@
 class DebianBasedDistribution(BaseDistribution):
     __required__ = '/etc/debian_version'
 
+class GentooBasedDistribution(BaseDistribution):
+    __required__ = '/etc/gentoo-release'
+    customization = {
+            'WVDIAL_CONN_SWITCH' : '--config',
+            'gksudo_name' : 'gksu'
+    }
+
 class FedoraBasedDistribution(BaseDistribution):
     __required__ = '/etc/fedora-release'
     customization = {
@@ -195,4 +202,5 @@
 
 OSNAME_COLLECTION = [DebianBasedDistribution,
                      FedoraBasedDistribution,
-                     SuseBasedDistribution]
+                     SuseBasedDistribution,
+		     GentooBasedDistribution]
