--- eina-0.8.0.ebuild.orig	2010-01-07 19:15:47.000000000 +0100
+++ eina-0.8.0.ebuild	2010-01-07 20:16:05.000000000 +0100
@@ -1,4 +1,4 @@
-# Copyright 1999-2009 Gentoo Foundation
+# Copyright 1999-2010 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 # $Header: $
 
@@ -11,31 +11,32 @@
 LICENSE="GPL-3"
 SLOT="0"
 KEYWORDS="~x86"
-IUSE=""
+IUSE="nls"
 
-RDEPEND="
-	>=dev-db/sqlite-3.2
+RDEPEND="dev-db/sqlite:3
 	>=dev-libs/dbus-glib-0.80
-	>=dev-libs/glib-2.16
-	>=dev-libs/libunique-1.0
-	>=media-libs/gstreamer-0.10
+	dev-libs/glib:2
+	dev-libs/libunique
+	media-libs/gstreamer
 	net-misc/curl
 	sys-devel/gettext
-	>=x11-libs/libnotify-0.4
+	virtual/libintl
+	x11-libs/libnotify
 	>=x11-libs/gtk+-2.14"
-DEPEND="
-	>=dev-util/pkgconfig-0.9
+DEPEND="dev-util/pkgconfig
+	nls? ( sys-devel/gettext )
 	${RDEPEND}"
 
 src_configure() {
-	econf --enable-release-build || die "configure failed"
+	use nls || my_conf="--disable-nls"
+	econf --enable-release-build ${my_conf} || die
 }
 
 src_compile() {
-	emake CFLAGS="${CFLAGS}" LDFLAGS="${LDFLAGS}" || die "compile failed"
+	emake CFLAGS="${CFLAGS}" LDFLAGS="${LDFLAGS}" || die
 }
 
 src_install() {
-	emake DESTDIR="${D}" install || die "install failed"
-	dodoc AUTHORS ChangeLog README NEWS || die "docs install failed" 
+	emake DESTDIR="${D}" install || die
+	dodoc AUTHORS ChangeLog README NEWS || die
 }
