--- Makefile.flags_orig	2011-05-01 13:23:38.386269115 -0400
+++ Makefile.flags	2011-04-28 04:04:53.195033415 -0400
@@ -27,6 +27,7 @@
 CFLAGS += $(call cc-option,-Wmissing-prototypes -Wmissing-declarations,)
 # warn about C99 declaration after statement
 CFLAGS += $(call cc-option,-Wdeclaration-after-statement,)
+CFLAGS += $(call cc-option,-Werror-implicit-function-declaration,)
 # If you want to add more -Wsomething above, make sure that it is
 # still possible to build bbox without warnings.
 
