--- coreutils/date.c_orig	2011-05-01 13:28:38.262264136 -0400
+++ coreutils/date.c	2011-05-01 13:29:09.182263429 -0400
@@ -249,7 +249,7 @@
 		xstat(filename, &statbuf);
 		ts.tv_sec = statbuf.st_mtime;
 #if ENABLE_FEATURE_DATE_NANO
-		ts.tv_nsec = statbuf.st_mtim.tv_nsec;
+		ts.tv_nsec = statbuf.st_mtimensec;
 		/* some toolchains use .st_mtimensec instead of st_mtim.tv_nsec */
 #endif
 	} else {
