--- miscutils/ionice.c_orig	2011-05-01 16:59:05.211279170 -0400
+++ miscutils/ionice.c	2011-05-01 16:59:53.025255531 -0400
@@ -13,12 +13,12 @@
 
 static int ioprio_set(int which, int who, int ioprio)
 {
-	return syscall(SYS_ioprio_set, which, who, ioprio);
+	return syscall(__NR_ioprio_set, which, who, ioprio);
 }
 
 static int ioprio_get(int which, int who)
 {
-	return syscall(SYS_ioprio_get, which, who);
+	return syscall(__NR_ioprio_get, which, who);
 }
 
 enum {
