--- networking/brctl.c_orig	2011-02-04 08:12:02.000000000 -0500
+++ networking/brctl.c	2011-04-28 15:26:31.828059855 -0400
@@ -43,7 +43,7 @@
 # include <linux/if_bridge.h>
 
 /* FIXME: These 4 funcs are not really clean and could be improved */
-static ALWAYS_INLINE void strtotimeval(struct timeval *tv,
+static ALWAYS_INLINE void _strtotimeval(struct timeval *tv,
 		const char *time_str)
 {
 	double secs;
@@ -80,7 +80,7 @@
 static unsigned long str_to_jiffies(const char *time_str)
 {
 	struct timeval tv;
-	strtotimeval(&tv, time_str);
+	_strtotimeval(&tv, time_str);
 	return tv_to_jiffies(&tv);
 }
 
