commit 33b27bdc2568742d6977ae676f5132e97e6a3051
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Sun May 1 18:29:41 2011 -0400

    fixed ifplugd

diff --git a/networking/ifplugd.c b/networking/ifplugd.c
index 0567b7f..844a959 100644
--- a/networking/ifplugd.c
+++ b/networking/ifplugd.c
@@ -12,6 +12,7 @@
 #include <linux/if.h>
 #include <linux/mii.h>
 #include <linux/ethtool.h>
+#include <netinet/if_ether.h>
 #include <net/if_ether.h>
 #include <linux/netlink.h>
 #include <linux/rtnetlink.h>
commit ba14719eb13a2633a406aadcd43632194cf19a63
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Sun May 1 18:35:58 2011 -0400

    added default dfinition of _PATH_VARRUN

diff --git a/networking/ifplugd.c b/networking/ifplugd.c
index 844a959..fae33a3 100644
--- a/networking/ifplugd.c
+++ b/networking/ifplugd.c
@@ -519,6 +519,9 @@ int ifplugd_main(int argc UNUSED_PARAM, char **argv)
 	applet_name = xasprintf("ifplugd(%s)", G.iface);
 
 #if ENABLE_FEATURE_PIDFILE
+#ifndef _PATH_VARRUN
+#define _PATH_VARRUN "/var/run/"
+#endif
 	pidfile_name = xasprintf(_PATH_VARRUN"ifplugd.%s.pid", G.iface);
 	pid_from_pidfile = read_pid(pidfile_name);
 
