commit b8ffe3a26399d268fc8ac3431da0bfd872978425
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Sun May 1 19:19:44 2011 -0400

    nslookup

diff --git a/networking/nslookup.c b/networking/nslookup.c
index dcac737..72ec42a 100644
--- a/networking/nslookup.c
+++ b/networking/nslookup.c
@@ -11,6 +11,7 @@
  * Licensed under GPLv2 or later, see file LICENSE in this source tree.
  */
 
+#include <sys/socket.h>
 #include <resolv.h>
 #include "libbb.h"
 
@@ -21,6 +22,7 @@
  * This applet is the only user of res_init(). Without it,
  * you may avoid pulling in _res global from libc.
  */
+extern int res_init(void);
 
 /* Examples of 'standard' nslookup output
  * $ nslookup yahoo.com
