commit f84783791c9fde24e036c1afe3bb35ee6787a694
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Thu May 5 20:04:03 2011 -0400

    for some reason, an enum was not sufficient here

diff --git a/util-linux/minix.h b/util-linux/minix.h
index e0fbcf7..ff778b4 100644
--- a/util-linux/minix.h
+++ b/util-linux/minix.h
@@ -54,25 +54,25 @@ struct minix_dir_entry {
 /* Believe it or not, but mount.h has this one #defined */
 #undef BLOCK_SIZE
 
-enum {
-	BLOCK_SIZE              = 1024,
-	BITS_PER_BLOCK          = BLOCK_SIZE << 3,
+#define	BLOCK_SIZE         1024
+#define	BITS_PER_BLOCK     (BLOCK_SIZE << 3)
 
-	MINIX_ROOT_INO          = 1,
-	MINIX_BAD_INO           = 2,
+#define	MINIX_ROOT_INO 1
+#define	MINIX_BAD_INO  2
 
-	MINIX1_SUPER_MAGIC      = 0x137F,       /* original minix fs */
-	MINIX1_SUPER_MAGIC2     = 0x138F,       /* minix fs, 30 char names */
-	MINIX2_SUPER_MAGIC      = 0x2468,       /* minix V2 fs */
-	MINIX2_SUPER_MAGIC2     = 0x2478,       /* minix V2 fs, 30 char names */
-	MINIX_VALID_FS          = 0x0001,       /* clean fs */
-	MINIX_ERROR_FS          = 0x0002,       /* fs has errors */
+#define MINIX1_SUPER_MAGIC  0x137F       /* original minix fs */
+#define MINIX1_SUPER_MAGIC2 0x138F       /* minix fs, 30 char names */
+#define MINIX2_SUPER_MAGIC  0x2468       /* minix V2 fs */
+#define MINIX2_SUPER_MAGIC2 0x2478       /* minix V2 fs, 30 char names */
 
-	INODE_SIZE1             = sizeof(struct minix1_inode),
-	INODE_SIZE2             = sizeof(struct minix2_inode),
-	MINIX1_INODES_PER_BLOCK = BLOCK_SIZE / sizeof(struct minix1_inode),
-	MINIX2_INODES_PER_BLOCK = BLOCK_SIZE / sizeof(struct minix2_inode),
-};
+#define MINIX_VALID_FS 0x0001       /* clean fs */
+#define MINIX_ERROR_FS 0x0002       /* fs has errors */
+
+#define INODE_SIZE1 sizeof(struct minix1_inode)
+#define INODE_SIZE2 sizeof(struct minix2_inode)
+
+#define MINIX1_INODES_PER_BLOCK (BLOCK_SIZE / sizeof(struct minix1_inode))
+#define MINIX2_INODES_PER_BLOCK (BLOCK_SIZE / sizeof(struct minix2_inode))
 
 /*
 Basic test script for regressions in mkfs/fsck.
