commit a75a04c12e87fe159477e6968e830ab7116c502a
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Thu May 5 20:10:49 2011 -0400

    ushort -> unsigned short

diff --git a/util-linux/ipcs.c b/util-linux/ipcs.c
index 2626d59..da3b13d 100644
--- a/util-linux/ipcs.c
+++ b/util-linux/ipcs.c
@@ -251,7 +251,7 @@ static NOINLINE void do_sem(void)
 	struct passwd *pw;
 	union semun arg;
 
-	arg.array = (ushort *) (void *) &seminfo;
+	arg.array = (unsigned short *) (void *) &seminfo;
 	maxid = semctl(0, 0, SEM_INFO, arg);
 	if (maxid < 0) {
 		printf("kernel not configured for %s\n", "semaphores");
@@ -261,7 +261,7 @@ static NOINLINE void do_sem(void)
 	switch (format) {
 	case LIMITS:
 		printf("------ Semaphore %s --------\n", "Limits");
-		arg.array = (ushort *) (void *) &seminfo;	/* damn union */
+		arg.array = (unsigned short *) (void *) &seminfo;	/* damn union */
 		if ((semctl(0, 0, IPC_INFO, arg)) < 0)
 			return;
 		printf("max number of arrays = %d\n"
