commit 5e2598f6ecad54f93dd292032a1453e842b20ce6
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Thu May 5 20:14:17 2011 -0400

    removed dupl struct sysinfo and included sys/sysinfo.h

diff --git a/include/libbb.h b/include/libbb.h
index 7c1db3f..4d84253 100644
--- a/include/libbb.h
+++ b/include/libbb.h
@@ -26,6 +26,7 @@
 #include <stdarg.h>
 #include <stddef.h>
 #include <string.h>
+#include <sys/sysinfo.h>
 #include <sys/poll.h>
 #include <sys/ioctl.h>
 #include <sys/mman.h>
@@ -100,25 +101,6 @@ int klogctl(int type, char *b, int len);
 #if !defined __FreeBSD__
 char *dirname(char *path);
 #endif
-/* Include our own copy of struct sysinfo to avoid binary compatibility
- * problems with Linux 2.4, which changed things.  Grumble, grumble. */
-struct sysinfo {
-	long uptime;			/* Seconds since boot */
-	unsigned long loads[3];		/* 1, 5, and 15 minute load averages */
-	unsigned long totalram;		/* Total usable main memory size */
-	unsigned long freeram;		/* Available memory size */
-	unsigned long sharedram;	/* Amount of shared memory */
-	unsigned long bufferram;	/* Memory used by buffers */
-	unsigned long totalswap;	/* Total swap space size */
-	unsigned long freeswap;		/* swap space still available */
-	unsigned short procs;		/* Number of current processes */
-	unsigned short pad;			/* Padding needed for m68k */
-	unsigned long totalhigh;	/* Total high memory size */
-	unsigned long freehigh;		/* Available high memory size */
-	unsigned int mem_unit;		/* Memory unit size in bytes */
-	char _f[20 - 2 * sizeof(long) - sizeof(int)]; /* Padding: libc5 uses this.. */
-};
-int sysinfo(struct sysinfo* info);
 #ifndef PATH_MAX
 # define PATH_MAX 256
 #endif
