commit a6ca6a797de4ec9087e3ee9880d4856f98d739e9
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Fri May 6 09:04:00 2011 -0400

    temporarily remove the link with libcrypt

diff --git a/Makefile.flags b/Makefile.flags
index a122cee..c3f3cdc 100644
--- a/Makefile.flags
+++ b/Makefile.flags
@@ -98,7 +98,7 @@ CFLAGS += $(strip $(subst ",,$(CONFIG_EXTRA_CFLAGS)))
 #"))
 endif
 
-LDLIBS += m crypt
+LDLIBS += m
 
 ifeq ($(CONFIG_PAM),y)
 LDLIBS += pam pam_misc
