commit 3cd6b0b07fee5c10c7d14d922528703cb0ae3cfc
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Fri Jan 13 21:55:34 2012 -0500

    do not call freeaddrinfo on null result

diff --git a/libbb/xconnect.c b/libbb/xconnect.c
index 3a6585c..b757ba0 100644
--- a/libbb/xconnect.c
+++ b/libbb/xconnect.c
@@ -285,6 +285,7 @@ IF_NOT_FEATURE_IPV6(sa_family_t af = AF_INET;)
  set_port:
 	set_nport(r, htons(port));
  ret:
+	if ( result )
 	freeaddrinfo(result);
 	return r;
 }
