diff -Nur a/source/CMakeLists.txt b/source/CMakeLists.txt
--- a/source/CMakeLists.txt
+++ b/source/CMakeLists.txt
@@ -70,7 +70,7 @@
     NAMELINK_COMPONENT ${SDK_COMPONENT_STRING}
 )
 
-if(!BUILD_INSTALLER)
+#if(!BUILD_INSTALLER)
 file(RELATIVE_PATH pkgconfig_prefix "${CMAKE_INSTALL_FULL_LIBDIR}/pkgconfig" "${CMAKE_INSTALL_PREFIX}")
 file(RELATIVE_PATH pkgconfig_include_dir "${CMAKE_INSTALL_PREFIX}" "${CMAKE_INSTALL_FULL_INCLUDEDIR}")
 file(RELATIVE_PATH pkgconfig_lib_dir "${CMAKE_INSTALL_PREFIX}" "${CMAKE_INSTALL_FULL_LIBDIR}")
@@ -81,4 +81,4 @@
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/libze_loader.pc"
               "${CMAKE_CURRENT_BINARY_DIR}/level-zero.pc"
         DESTINATION "${CMAKE_INSTALL_FULL_LIBDIR}/pkgconfig" COMPONENT ${SDK_COMPONENT_STRING})
-endif()
+#endif()
