diff --git a/Rakefile b/Rakefile
index a9346c5..fe3353b 100644
--- a/Rakefile
+++ b/Rakefile
@@ -1,8 +1,3 @@
-begin
-  require 'bundler'
-  Bundler::GemHelper.install_tasks
-rescue Exception
-end
 
 require 'rake/testtask'
 
@@ -47,25 +42,6 @@ namespace 'test' do
     t.verbose = true
   end
 
-  begin
-    require 'sinatra'
-    spec = Gem::Specification.find_by_name('sinatra')
-    Rake::TestTask.new('sinatra') do |t|
-      # FIXME: Rename deprecated attribute
-      file = "#{spec.gem_dir}/test/slim_test.rb"
-      code = File.read(file)
-      code.gsub!('attr_wrapper', 'attr_quote')
-      File.open(file, 'w') {|out| out.write(code) }
-
-      # Run Slim integration test in Sinatra
-      t.test_files = FileList[file]
-      t.verbose = true
-    end
-  rescue LoadError
-    task :sinatra do
-      abort 'Sinatra is not available'
-    end
-  end
 end
 
 begin
