--- /usr/portage/dev-python/gnome-python-extras/gnome-python-extras-2.12.0.ebuild	2006-02-02 04:39:03.000000000 +0100
+++ /usr/local/portage/dev-python/gnome-python-extras/gnome-python-extras-2.12.0.ebuild	2006-02-12 13:46:58.000000000 +0100
@@ -12,7 +12,7 @@
 KEYWORDS="alpha amd64 ia64 ppc sparc x86"
 IUSE="doc firefox mozilla"
 
-RDEPEND=">=dev-lang/python-2.3
+RDEPEND=">=dev-lang/python-2.4
 	>=x11-libs/gtk+-2.4
 	>=dev-libs/glib-2.6
 	>=dev-python/gnome-python-2.10
@@ -48,7 +48,7 @@
 	# necessary to disable gda module because they are using
 	# functions only available in python 2.4. when 2.4 goes
 	# stable, this can be removed.
-	epatch ${FILESDIR}/${PN}-nogda.patch
+#	epatch ${FILESDIR}/${PN}-nogda.patch
 
 	automake
 	libtoolize --copy --force
