--- /usr/local/portage/net-misc/dhcp-helper/files/dhcp-helper.rc	2011-08-15 12:56:10.989569151 +0200
+++ /etc/init.d/dhcp-helper	2011-08-15 13:59:54.199567677 +0200
@@ -8,7 +8,7 @@
 }
 
 start() {
-        if [ ! ${DHCPHELPER_OPTS} ]; then
+        if [ ! "${DHCPHELPER_OPTS}" ]; then
 		einfo "You must specify some options to dhcp-helper."
 		einfo "Edit the /etc/conf.d/dhcp-helper file."
 		einfo "See the man dhcp-helper for options."
