--- /usr/local/portage/app-text/ocropus/ocropus-0.4-r1.ebuild.kraal	2011-11-22 08:28:42.000000000 -0500
+++ /usr/local/portage/app-text/ocropus/ocropus-0.4-r1.ebuild	2011-11-22 08:48:32.000000000 -0500
@@ -19,15 +19,15 @@
 KEYWORDS="~amd64 ~x86"
 IUSE="sdl lua openfst spell interactive"
 
-RDEPEND=">=app-text/tesseract-2
+RDEPEND="=app-text/tesseract-2*
 	media-libs/tiff
 	media-libs/libpng
 	media-libs/jpeg
 	sci-libs/gsl
 	>=media-libs/iulib-0.4
 	spell? ( app-text/aspell )
-	openfst? ( sci-misc/openfst )
-	lua? ( media-libs/leptonica )
+	openfst? ( || ( media-libs/OpenFst  sci-misc/openfst ) )
+	lua? ( || ( media-libs/leptonica  dev-libs/leptonlib ) )
 	interactive? ( dev-libs/libedit )"
 # depedencies in ubuntu: build-essential imagemagick \ tesseract-ocr tesseract-ocr-dev \ libpng12-dev libjpeg62-dev libtiff4-dev libgif-dev \ libsdl-gfx1.2-dev libsdl-image1.2-dev libsdl1.2-dev \ libavcodec-dev libavdevice-dev libavformat-dev libavutil-dev \ libavfilter-dev \ tesseract-ocr-dev libleptonica-dev \ libgsl0-dev libblas-dev \ python-dev swig1.3
 
@@ -45,10 +45,9 @@
 S=${WORKDIR}/${P%.1}/ocropus
 
 pkg_setup() {
-    append-ldflags $(no-as-needed)
+	append-ldflags $(no-as-needed)
 }
 
-
 src_unpack() {
 	unpack ${A}
 	#cd "${S}/.."
