diff -cr ./ocr-utils/xml-entities.cc /var/tmp/portage/app-text/ocropus-0.4-r1/work/ocropus-0.4/ocropus/ocr-utils/xml-entities.cc
*** ocr-utils/xml-entities.cc	Sun May 31 20:18:41 2009
--- ocr-utils/xml-entities.cc	Wed Oct 28 11:47:56 2009
***************
*** 111,120 ****
      // This function probably logically belongs to Lua
      // but due to the lack of good string manipulation functions there
      // it's actually simpler in C.
!     void xml_unescape(nustring &dest, const char *src) {
          CHECK_ARG(src);
          dest.clear();
!         const char *begin = src;
          char *p;
          while((p = strchr(begin, '&'))) {
              append_to_nustring(dest, begin, p - begin);
--- 111,120 ----
      // This function probably logically belongs to Lua
      // but due to the lack of good string manipulation functions there
      // it's actually simpler in C.
!     void xml_unescape(nustring &dest, char *src) {
          CHECK_ARG(src);
          dest.clear();
!         char *begin = src;
          char *p;
          while((p = strchr(begin, '&'))) {
              append_to_nustring(dest, begin, p - begin);
