--- lush-1.1/include/header.h.orig	2006-06-20 00:18:29.000000000 +0200
+++ lush-1.1/include/header.h	2006-06-20 00:18:41.000000000 +0200
@@ -805,7 +805,6 @@
 #define DATE_MINUTE     4
 #define DATE_SECOND     5
 
-extern char *ansidatenames[];
 extern class date_class;
 
 TLAPI char *str_date( at *p, int *pfrom, int *pto );
--- lush-1.1/src/date.c.orig	2006-06-20 00:16:58.000000000 +0200
+++ lush-1.1/src/date.c	2006-06-20 00:24:17.000000000 +0200
@@ -311,10 +311,11 @@
   return string_buffer;
 }
 
+static at *make_date(struct date *d);
+
 static void
 date_serialize(at **pp, int code)
 {
-  static at *make_date(struct date *d);
   void *p;
 
   if (code == SRZ_READ)
--- lush-1.1/src/index.c.orig	2006-06-20 00:25:41.000000000 +0200
+++ lush-1.1/src/index.c	2006-06-20 00:26:17.000000000 +0200
@@ -141,6 +141,8 @@
   return string_buffer;
 }
 
+static at *index_set(struct index*,at**,at*,int);
+static at *index_ref(struct index*,at**);
 
 static at *
 index_listeval(at *p, at *q)
@@ -150,9 +152,6 @@
   register at *qsav;
   at *myp[MAXDIMS];
 
-  static at *index_set(struct index*,at**,at*,int);
-  static at *index_ref(struct index*,at**);
-
   ind = p->Object;
 
   if (ind->flags & IDF_UNSIZED)
--- lush-1.1/src/string.c.orig	2006-06-20 00:30:02.000000000 +0200
+++ lush-1.1/src/string.c	2006-06-20 00:30:07.000000000 +0200
@@ -1041,13 +1041,13 @@
     return NIL;
 #if HAVE_WCHAR_T
   {
-    int n = strlen(s);
+    int n = strlen((char*)s);
     mbstate_t ps;
     memset(&ps, 0, sizeof(mbstate_t));
     while(n > 0)
       {
 	wchar_t wc = 0;
-	int m = (int)mbrtowc(&wc, s, n, &ps);
+	int m = (int)mbrtowc(&wc, (char*)s, n, &ps);
 	if (m == 0)
 	  break;
 	if (m < 0)
--- lush-1.1/src/unix.c.orig	2006-06-20 00:31:25.000000000 +0200
+++ lush-1.1/src/unix.c	2006-06-20 00:31:32.000000000 +0200
@@ -1487,7 +1487,7 @@
       for (i=0; i<kidpidsize; i++)
         if (kidpid[i])
           close(i);
-      execl("/bin/sh", "sh", "-c", cmd, 0);
+      execl("/bin/sh", "sh", "-c", cmd, NULL);
       _exit(127);
     }
   /* Parent process */
@@ -1594,7 +1594,7 @@
         if (kidpid[i])
           close(i);
 #endif
-      execl("/bin/sh", "sh", "-c", string_buffer, 0);
+      execl("/bin/sh", "sh", "-c", string_buffer, NULL);
       _exit(127);
     }
   /* Parent process */ 
@@ -1688,7 +1688,7 @@
         if (kidpid[i])
           close(i);
 # endif
-      execl("/bin/sh", "sh", "-c", string_buffer, 0);
+      execl("/bin/sh", "sh", "-c", string_buffer, NULL);
       _exit(127);
     }
   /* Parent process */ 
