diff -Naur rekall-2.4.6/script/python/kb_basic.cpp rekall-2.4.6-1/script/python/kb_basic.cpp
--- rekall-2.4.6/script/python/kb_basic.cpp	2007-02-14 07:35:41.000000000 +1100
+++ rekall-2.4.6-1/script/python/kb_basic.cpp	2007-11-25 18:41:09.000000000 +1100
@@ -401,7 +401,7 @@
 	QStringList	pairs	;
 	PyObject	*key	;
 	PyObject	*value	;
-	int		pos	= 0 ;
+	long		pos	= 0 ;
 
 	//xxx use pickle or marshal to create the data representation instead
 	//xxx of a simple string representation?
diff -Naur rekall-2.4.6/script/python/kb_pycontainer.cpp rekall-2.4.6-1/script/python/kb_pycontainer.cpp
--- rekall-2.4.6/script/python/kb_pycontainer.cpp	2006-10-11 16:30:04.000000000 +1000
+++ rekall-2.4.6-1/script/python/kb_pycontainer.cpp	2007-11-25 18:58:06.000000000 +1100
@@ -60,7 +60,7 @@
 	}
 
 	char	*buffer	;
-	int	len	;
+	long	len	;
 	PyString_AsStringAndSize (pyImage, &buffer, &len) ;
 
 	QByteArray ba	;
diff -Naur rekall-2.4.6/script/python/kb_pysupport.cpp rekall-2.4.6-1/script/python/kb_pysupport.cpp
--- rekall-2.4.6/script/python/kb_pysupport.cpp	2006-08-11 01:29:42.000000000 +1000
+++ rekall-2.4.6-1/script/python/kb_pysupport.cpp	2007-11-25 19:57:48.000000000 +1100
@@ -37,7 +37,7 @@
 {
 	PyObject	*key	;
 	PyObject	*value	;
-	int		pos	= 0 ;
+	long		pos	= 0 ;
 
 	kbDict.setAutoDelete (true) ;
 
diff -Naur rekall-2.4.6/script/python/tkc_pydebugbase.cpp rekall-2.4.6-1/script/python/tkc_pydebugbase.cpp
--- rekall-2.4.6/script/python/tkc_pydebugbase.cpp	2006-08-11 01:29:42.000000000 +1000
+++ rekall-2.4.6-1/script/python/tkc_pydebugbase.cpp	2007-11-25 19:17:07.000000000 +1100
@@ -1082,7 +1082,7 @@
 	{
 		PyObject *pkey	 ;
 		PyObject *pvalue ;
-		int	 dictPos = 0 ;
+		long	 dictPos = 0 ;
 
 		while (PyDict_Next (pyDict, &dictPos, &pkey, &pvalue))
 		{
@@ -1135,7 +1135,7 @@
 	PyObject  *pyKey   ;
 	PyObject  *pyValue ;
 	QString	  name	   ;
-	int	  dictPos  = 0 ;
+	long	  dictPos  = 0 ;
 
 	/* See if we have been here before, in which case there is no	*/
 	/* point in going there again. This also prevents infinite	*/
