diff -ur src.0/audiere.h src/audiere.h
--- src.0/audiere.h	2006-02-14 07:57:01.000000000 +0300
+++ src/audiere.h	2008-11-19 14:09:03.000000000 +0300
@@ -28,6 +28,7 @@
 
 #include <vector>
 #include <string>
+#include <cstring>
 
 #ifdef _MSC_VER
 #pragma warning(disable : 4786)
diff -ur src.0/debug.h src/debug.h
--- src.0/debug.h	2006-02-14 07:57:01.000000000 +0300
+++ src/debug.h	2008-11-19 14:12:08.000000000 +0300
@@ -5,7 +5,7 @@
 #include <assert.h>
 #include <stdio.h>
 #include <string>
-
+#include <cstdlib>
 
 namespace audiere {
 
