diff -r b70cf37e0eb6 -r f32ba03fa5a2 src/grlida.cpp
--- a/src/grlida.cpp	Tue Oct 28 20:58:48 2008 +0100
+++ b/src/grlida.cpp	Fri Nov 07 17:25:00 2008 +0100
@@ -88,7 +88,7 @@
 	ui.twJuegos->setColumnWidth(2,90);
 	ui.twJuegos->setContextMenuPolicy(Qt::CustomContextMenu);
 	ui.twJuegos->installEventFilter(this);
-	ui.twJuegos->setItemDelegate( new StarDelegate( ui.twJuegos ) );
+	ui.twJuegos->setItemDelegate( new StarDelegate );
 
 	ui.txtInfo_Comentario->installEventFilter(this);
 
diff -r b70cf37e0eb6 -r f32ba03fa5a2 src/stardelegate.cpp
--- a/src/stardelegate.cpp	Tue Oct 28 20:58:48 2008 +0100
+++ b/src/stardelegate.cpp	Fri Nov 07 17:25:00 2008 +0100
@@ -1,6 +1,4 @@
 #include "stardelegate.h"
-
-#include <QtGui>
 
 StarDelegate::StarDelegate(QWidget *parent)
     : QItemDelegate(parent)
@@ -10,6 +8,8 @@
 	star_on.load(":/images/star_on.png");
 	star_off.load(":/images/star_off.png");
 }
+
+StarDelegate::~StarDelegate(){}
 
 void StarDelegate::paint(QPainter *painter, const QStyleOptionViewItem &option,
 		const QModelIndex &index) const
diff -r b70cf37e0eb6 -r f32ba03fa5a2 src/stardelegate.h
--- a/src/stardelegate.h	Tue Oct 28 20:58:48 2008 +0100
+++ b/src/stardelegate.h	Fri Nov 07 17:25:00 2008 +0100
@@ -1,18 +1,19 @@
 #ifndef STARDELEGATE_H
 #define STARDELEGATE_H
 
+#include <QtCore>
+#include <QtGui>
 #include <QItemDelegate>
 #include <QSize>
 
 #include "dbsql.h"
-
-QT_FORWARD_DECLARE_CLASS(QPainter)
 
 class StarDelegate : public QItemDelegate
 {
 	Q_OBJECT
 public:
 	StarDelegate(QWidget *parent = 0);
+	~StarDelegate();
 
 	QSize sizeHint(const QStyleOptionViewItem &option, const QModelIndex &index) const;
 	void paint(QPainter *painter, const QStyleOptionViewItem &option, const QModelIndex &index) const;
@@ -26,4 +27,3 @@
 };
 
 #endif // STARDELEGATE_H
-
