--- a/setup.py	2026-10-04 20:31:10.453035339 -0700
+++ b/setup.py	2026-10-04 20:31:10.454988701 -0700
@@ -65,6 +65,9 @@
         if "FETCHCONTENT_SOURCE_DIR_DLPACK" in os.environ:
             cmake_args.append(f"-DFETCHCONTENT_SOURCE_DIR_DLPACK={os.environ['FETCHCONTENT_SOURCE_DIR_DLPACK']}")
 
+        if "CUDNN_FRONTEND_USE_SYSTEM_DLPACK" in os.environ:
+            cmake_args.append(f"-DCUDNN_FRONTEND_USE_SYSTEM_DLPACK={os.environ['CUDNN_FRONTEND_USE_SYSTEM_DLPACK']}")
+
         # Using Ninja-build since it a) is available as a wheel and b)
         # multithreads automatically. MSVC would require all variables be
         # exported for Ninja to pick it up, which is a little tricky to do.
