diff -Nauir mysdl_1.0-p/src/mysdl.cpp mysdl_1.0-p2/src/mysdl.cpp
--- mysdl_1.0-p/src/mysdl.cpp	2009-09-28 14:49:29.000000000 +0200
+++ mysdl_1.0-p2/src/mysdl.cpp	2009-09-28 16:16:18.000000000 +0200
@@ -771,23 +771,23 @@
 			throw std::string("Could not open file \"")+std::string(filename)+std::string("\" for writing.");
 		}
 
-		png_ptr=png_create_write_struct(PNG_LIBPNG_VER_STRING,0,0,0);
+//		png_ptr=png_create_write_struct(PNG_LIBPNG_VER_STRING,0,0,0);
 
 		if(png_ptr==0)
 		{
 			fclose(file);
-			png_destroy_write_struct(&png_ptr,(png_infopp)0);
+//			png_destroy_write_struct(&png_ptr,(png_infopp)0);
 
 			throw std::string("Could not create libpng data structures.");
 		}
 		else
 		{
-			info_ptr=png_create_info_struct(png_ptr);
+//			info_ptr=png_create_info_struct(png_ptr);
 
 			if(info_ptr==0)
 			{
 				fclose(file);
-				png_destroy_write_struct(&png_ptr,(png_infopp)0);
+//				png_destroy_write_struct(&png_ptr,(png_infopp)0);
 
 				throw std::string("Could not create libpng data structures.");
 			}
@@ -796,13 +796,13 @@
 				if(setjmp(png_jmpbuf(png_ptr)))
 				{
 					fclose(file);
-					png_destroy_write_struct(&png_ptr,(png_infopp)NULL);
+//					png_destroy_write_struct(&png_ptr,(png_infopp)NULL);
 
 					throw std::string("Could not set libpng error callback.");
 				}
 				else
 				{
-					png_init_io(png_ptr,file);
+//					png_init_io(png_ptr,file);
 
 					info_ptr->width=surf->w;
 					info_ptr->height=surf->h;
@@ -811,7 +811,7 @@
 					info_ptr->interlace_type=1;
 					info_ptr->valid=0;
 
-					png_write_info(png_ptr,info_ptr);
+//					png_write_info(png_ptr,info_ptr);
 
 					// --- Save the picture... ---
 					unsigned char **png_rows;
@@ -832,7 +832,7 @@
 						}
 					}
 
-					png_write_image(png_ptr,png_rows);
+//					png_write_image(png_ptr,png_rows);
 
 					for(y=0; y<surf->h; y++)
 					{
@@ -840,8 +840,8 @@
 					}
 
 					free(png_rows);
-					png_write_end(png_ptr, NULL);
-					png_destroy_write_struct(&png_ptr, &info_ptr);
+//					png_write_end(png_ptr, NULL);
+//					png_destroy_write_struct(&png_ptr, &info_ptr);
 
 					fclose(file);
 				}
