--- setup.py	2004-10-04 20:05:01.000000000 +0800
+++ setup.py.new	2004-10-05 12:43:10.283230448 +0800
@@ -10,13 +10,13 @@
 superlu_defs = [('USE_VENDOR_BLAS',1)]
 f77_defs = []
 
-umfpack_defs = [('DINT', 1), ('NBLAS', 1)] # most basic configuration, no BLAS
-umfpack_libraries = ['m']
+umfpack_defs = [('DINT', 1)] # most basic configuration, no BLAS
+umfpack_libraries = ['blas', 'atlas', 'frtbegin', 'g2c', 'm']
 umfpack_include_dirs = ['amd', 'umfpack']
 umfpack_library_dirs = []
 
 #umfpack_defs = [('DINT', 1), ('CBLAS', 1)] # with atlas c-blas (http://math-atlas.sourceforge.net)
-#umfpack_libraries = ['atlas', 'cblas', 'm']
+#umfpack_libraries = ['lapack', 'cblas', 'm']
 #umfpack_include_dirs = ['amd', 'umfpack'] # you may need to set this to find the atlas
 #umfpack_library_dirs = []
 
