=== modified file 'sdas/as6808/Makefile.in'
--- sdas/as6808/Makefile.in	2010-11-17 18:14:19 +0000
+++ sdas/as6808/Makefile.in	2010-11-17 18:16:52 +0000
@@ -36,7 +36,7 @@
 
 CPPFLAGS        = @CPPFLAGS@ -DSDCDB -DNOICE -DINDEXLIB -I. -I$(srcdir) -I$(srcdir)/../asxxsrc -I$(srcdir)/../../support/Util
 CFLAGS          = @CFLAGS@ @WALL_FLAG@ $(CPPFLAGS)
-LDFLAGS         = @LDFLAGS@ -lm
+LDFLAGS         = @LDFLAGS@
 
 OBJDIR = obj
 
@@ -65,7 +65,7 @@
 all: checkconf $(ASX)
 
 $(ASX): $(ASOBJECTS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(ASOBJECTS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(ASOBJECTS) -lm
 
 # Compiling and installing everything and running test
 # ----------------------------------------------------

=== modified file 'sdas/as8051/Makefile.in'
--- sdas/as8051/Makefile.in	2010-11-17 18:14:19 +0000
+++ sdas/as8051/Makefile.in	2010-11-17 18:17:35 +0000
@@ -36,7 +36,7 @@
 
 CPPFLAGS        = @CPPFLAGS@ -DSDCDB -DNOICE -DINDEXLIB -I. -I$(srcdir) -I$(srcdir)/../asxxsrc -I$(srcdir)/../../support/Util
 CFLAGS          = @CFLAGS@ @WALL_FLAG@ $(CPPFLAGS)
-LDFLAGS         = @LDFLAGS@ -lm
+LDFLAGS         = @LDFLAGS@
 
 OBJDIR = obj
 
@@ -65,7 +65,7 @@
 all: checkconf $(ASX)
 
 $(ASX): $(ASOBJECTS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(ASOBJECTS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(ASOBJECTS) -lm
 
 # Compiling and installing everything and running test
 # ----------------------------------------------------

=== modified file 'sdas/asgb/Makefile.in'
--- sdas/asgb/Makefile.in	2010-11-17 18:14:19 +0000
+++ sdas/asgb/Makefile.in	2010-11-17 18:18:13 +0000
@@ -36,7 +36,7 @@
 
 CPPFLAGS        = @CPPFLAGS@ -DSDCDB -DNOICE -DINDEXLIB -I. -I$(srcdir) -I$(srcdir)/../asxxsrc -I$(srcdir)/../../support/Util
 CFLAGS          = @CFLAGS@ @WALL_FLAG@ $(CPPFLAGS)
-LDFLAGS         = @LDFLAGS@ -lm
+LDFLAGS         = @LDFLAGS@
 
 OBJDIR = obj
 
@@ -65,7 +65,7 @@
 all: checkconf $(ASX)
 
 $(ASX): $(ASOBJECTS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(ASOBJECTS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(ASOBJECTS) -lm
 
 # Compiling and installing everything and running test
 # ----------------------------------------------------

=== modified file 'sdas/asz80/Makefile.in'
--- sdas/asz80/Makefile.in	2010-11-17 18:14:19 +0000
+++ sdas/asz80/Makefile.in	2010-11-17 18:17:55 +0000
@@ -36,7 +36,7 @@
 
 CPPFLAGS        = @CPPFLAGS@ -DSDCDB -DNOICE -DINDEXLIB -I. -I$(srcdir) -I$(srcdir)/../asxxsrc -I$(srcdir)/../../support/Util
 CFLAGS          = @CFLAGS@ @WALL_FLAG@ $(CPPFLAGS)
-LDFLAGS         = @LDFLAGS@ -lm
+LDFLAGS         = @LDFLAGS@
 
 OBJDIR = obj
 
@@ -65,7 +65,7 @@
 all: checkconf $(ASX)
 
 $(ASX): $(ASOBJECTS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(ASOBJECTS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(ASOBJECTS) -lm
 
 # Compiling and installing everything and running test
 # ----------------------------------------------------

