--- nibbles-0.11.orig/nibbles.asd	2013-01-15 11:54:47.000000000 +0800
+++ nibbles-0.11/nibbles.asd	2013-03-16 09:09:00.416681058 +0800
@@ -5,27 +5,16 @@
 
 (cl:in-package :nibbles-system)
 
-(defclass txt-file (asdf:doc-file) ((type :initform "txt")))
-(defclass css-file (asdf:doc-file) ((type :initform "css")))
-
 (asdf:defsystem :nibbles
   :version "0.11"
   :author "Nathan Froyd <froydnj@gmail.com>"
   :maintainer "Nathan Froyd <froydnj@gmail.com>"
   :description "A library for accessing octet-addressed blocks of data"
-  :components ((:static-file "README")
-               (:static-file "LICENSE")
-               (:static-file "NEWS")
-               (:file "package")
+  :components ((:file "package")
                (:file "types" :depends-on ("package"))
                (:file "macro-utils" :depends-on ("package"))
                (:file "vectors" :depends-on ("types" "macro-utils"))
                (:file "streams" :depends-on ("vectors"))
-	       (:module "doc"
-			:components
-			((:html-file "index")
-			 (:txt-file "nibbles-doc")
-			 (:css-file "style")))
                (:module "sbcl-opt"
                         :depends-on ("package" "macro-utils")
                         :components ((:file "fndb")
