diff -ur nibbles_0.9.orig/nibbles.asd nibbles_0.9/nibbles.asd
--- nibbles_0.9.orig/nibbles.asd	2011-02-10 23:12:49.000000000 +0100
+++ nibbles_0.9/nibbles.asd	2011-02-12 23:07:07.252409497 +0100
@@ -5,30 +5,17 @@
 
 (cl:in-package :nibbles-system)
 
-(defclass txt-file (asdf:doc-file) ())
-(defclass css-file (asdf:doc-file) ())
-
-(defmethod asdf:source-file-type ((c txt-file) (s asdf:module)) "txt")
-(defmethod asdf:source-file-type ((c css-file) (s asdf:module)) "css")
-
 (asdf:defsystem :nibbles
   :version "0.9"
   :author "Nathan Froyd <froydnj@gmail.com>"
   :maintainer "Nathan Froyd <froydnj@gmail.com>"
   :description "A library for accessing octet-addressed blocks of data"
-  :components ((:static-file "README")
-               (:static-file "LICENSE")
-               (:file "package")
+  :components ((:file "package")
                (:file "types" :depends-on ("package"))
                (:file "macro-utils" :depends-on ("package"))
                (:file "vectors" :depends-on ("types")
                       :in-order-to ((asdf:compile-op (asdf:load-op "macro-utils"))))
                (:file "streams" :depends-on ("vectors"))
-	       (:module "doc"
-			:components
-			((:html-file "index")
-			 (:txt-file "nibbles-doc")
-			 (:css-file "style")))
                (:module "sbcl-opt"
                         :depends-on ("package" "macro-utils")
                         :if-component-dep-fails :ignore
