--- rbutilqt.pro.orig	2011-01-05 10:42:19.299331569 +0100
+++ rbutilqt.pro	2011-01-05 10:44:24.933347928 +0100
@@ -45,6 +45,12 @@
 
 message("Rockbox Base dir: "$$RBBASE_DIR)
 
+unix {
+	# Some of the tool's Makefiles do not create their build dir with "mkdir -p", so do it here:
+	LIST = rbspeex ucl mkamsboot mktccboot mkmpioboot
+	for(dir, LIST):!exists($$MYBUILDDIR/$$dir):system(mkdir -p $$MYBUILDDIR/$$dir)
+}
+
 # custom rules for rockbox-specific libs
 !mac {
     RBLIBPOSTFIX = .a
@@ -67,20 +67,20 @@
 }
 
 rbspeex.commands = @$(MAKE) \
-        TARGET_DIR=$$MYBUILDDIR -C $$RBBASE_DIR/tools/rbspeex \
+        TARGET_DIR=$$MYBUILDDIR/rbspeex/ -C $$RBBASE_DIR/tools/rbspeex \
         librbspeex$$RBLIBPOSTFIX CC=\"$$QMAKE_CC\" \
         SYS_SPEEX=\"$$LIBSPEEX\"
 libucl.commands = @$(MAKE) \
-        TARGET_DIR=$$MYBUILDDIR -C $$RBBASE_DIR/tools/ucl/src \
+        TARGET_DIR=$$MYBUILDDIR/ucl/ -C $$RBBASE_DIR/tools/ucl/src \
         libucl$$RBLIBPOSTFIX CC=\"$$QMAKE_CC\"
 libmkamsboot.commands = @$(MAKE) \
-        TARGET_DIR=$$MYBUILDDIR -C $$RBBASE_DIR/rbutil/mkamsboot \
+        TARGET_DIR=$$MYBUILDDIR/mkamsboot/ -C $$RBBASE_DIR/rbutil/mkamsboot \
         libmkamsboot$$RBLIBPOSTFIX CC=\"$$QMAKE_CC\"
 libmktccboot.commands = @$(MAKE) \
-        TARGET_DIR=$$MYBUILDDIR -C $$RBBASE_DIR/rbutil/mktccboot \
+        TARGET_DIR=$$MYBUILDDIR/mktccboot/ -C $$RBBASE_DIR/rbutil/mktccboot \
         libmktccboot$$RBLIBPOSTFIX CC=\"$$QMAKE_CC\"
 libmkmpioboot.commands = @$(MAKE) \
-        TARGET_DIR=$$MYBUILDDIR -C $$RBBASE_DIR/rbutil/mkmpioboot \
+        TARGET_DIR=$$MYBUILDDIR/mkmpioboot/ -C $$RBBASE_DIR/rbutil/mkmpioboot \
         libmkmpioboot$$RBLIBPOSTFIX CC=\"$$QMAKE_CC\"
 QMAKE_EXTRA_TARGETS += rbspeex libucl libmkamsboot libmktccboot libmkmpioboot
 PRE_TARGETDEPS += rbspeex libucl libmkamsboot libmktccboot libmkmpioboot
@@ -105,7 +105,7 @@
 
 DEPENDPATH = $$INCLUDEPATH
 
-LIBS += -L$$OUT_PWD -L$$MYBUILDDIR -lrbspeex -lmkamsboot -lmktccboot -lmkmpioboot -lucl
+LIBS += -L$$OUT_PWD -L$$MYBUILDDIR -L$$MYBUILDDIR/rbspeex/ -lrbspeex -L$$MYBUILDDIR/mkamsboot/ -lmkamsboot -L$$MYBUILDDIR/mktccboot/ -lmktccboot -L$$MYBUILDDIR/mkmpioboot/ -lmkmpioboot -L$$MYBUILDDIR/ucl/ -lucl
 
 # Add a (possibly found) libspeex now, don't do this before -lrbspeex!
 !static:!isEmpty(LIBSPEEX) {
