--- configure_orig	2008-09-17 19:04:19.000000000 +0200
+++ configure	2008-09-17 19:04:30.000000000 +0200
@@ -56,7 +56,6 @@
 	echo "EXT=${ext}" >> Makefile
 	echo "OUTPUT_DIR=lib" >> Makefile
 	echo "WXCFG=${wxcfg}" >> Makefile
-	echo "DESTDIR=" >> Makefile
 	echo "" >> Makefile
 	echo "CCFLAGS= -Wall \$(shell wx-config --cxxflags \$(WXCFG)) -fno-strict-aliasing -DASTYLE_LIB -DYY_NEVER_INTERACTIVE=1 -DGTK -DSCI_LEXER -DLINK_LEXERS -D__WX__ " >> Makefile
 	echo "LINK_FLAGS =  -L${base_dir}/\$(OUTPUT_DIR) -lplugin\$(EXT) -lcodelite\$(EXT) -lwxsqlite3\$(EXT) \\" >> Makefile
