diff -Nur a/setup.py b/setup.py
--- a/setup.py	2025-01-28 16:09:34.000000000 -0800
+++ b/setup.py	2025-02-01 11:44:01.312092822 -0800
@@ -256,6 +256,7 @@
 import setuptools.command.sdist
 from setuptools import Extension, find_packages, setup
 from setuptools.dist import Distribution
+sys.path[:0] = os.getcwd()
 from tools.build_pytorch_libs import build_pytorch
 from tools.generate_torch_version import get_torch_version
 from tools.setup_helpers.cmake import CMake
