diff -Nur a/cmake/pypi-z3/FindZ3.cmake b/cmake/pypi-z3/FindZ3.cmake
--- a/cmake/pypi-z3/FindZ3.cmake
+++ b/cmake/pypi-z3/FindZ3.cmake
@@ -12,7 +12,7 @@
     message(FATAL_ERROR "Failed to locate z3 Python package. Ensure z3-solver>=4.13.0 is installed.")
 endif()
 message("-- Find Z3 in path: ${Z3_PATH}")
-find_path(Z3_INCLUDE_DIR NO_DEFAULT_PATH NAMES z3++.h PATHS ${Z3_PATH}/include)
+find_path(Z3_INCLUDE_DIR NAMES z3++.h PATHS ${Z3_PATH}/include)
 if(WIN32)
     find_file(Z3_RUNTIME_LIBRARY NO_DEFAULT_PATH NAMES libz3.dll z3.dll PATHS ${Z3_PATH}/bin ${Z3_PATH}/lib ${Z3_PATH}/lib64)
     find_library(Z3_IMPORT_LIBRARY NO_DEFAULT_PATH NAMES libz3 z3 PATHS ${Z3_PATH}/bin ${Z3_PATH}/lib ${Z3_PATH}/lib64)
@@ -31,7 +31,7 @@
     set(Z3_LIBRARY "${Z3_IMPORT_LIBRARY}")
 else()
-    find_library(Z3_LIBRARY NO_DEFAULT_PATH NAMES z3 libz3 PATHS ${Z3_PATH}/bin ${Z3_PATH}/lib ${Z3_PATH}/lib64)
+    find_library(Z3_LIBRARY NAMES z3 libz3 PATHS ${Z3_PATH}/bin ${Z3_PATH}/lib ${Z3_PATH}/lib64)
     if(APPLE AND Z3_LIBRARY)
         set(Z3_ORIGINAL_LIBRARY "${Z3_LIBRARY}")
         set(Z3_LIBRARY "${CMAKE_BINARY_DIR}/lib/libz3${CMAKE_SHARED_LIBRARY_SUFFIX}")
