diff -Nur poco-1.7.4-all.orig/Data/ODBC/ODBC.make poco-1.7.4-all/Data/ODBC/ODBC.make
--- poco-1.7.4-all.orig/Data/ODBC/ODBC.make	2016-07-21 01:52:57.000000000 +0800
+++ poco-1.7.4-all/Data/ODBC/ODBC.make	2016-08-14 12:52:39.310627216 +0800
@@ -14,9 +14,9 @@
 ifeq (0, $(shell test -d /usr/lib/$(OSARCH)-linux-gnu; echo $$?))
 POCO_ODBC_LIB = /usr/lib/$(OSARCH)-linux-gnu
 else ifeq (0, $(shell test -d /usr/lib64; echo $$?))
-POCO_ODBC_LIB = /usr/lib64
+POCO_ODBC_LIB = /usr/$(LIBDIR)
 else 
-POCO_ODBC_LIB = /usr/lib
+POCO_ODBC_LIB = /usr/$(LIBDIR)
 endif
 endif
 
@@ -46,7 +46,7 @@
 ##
 else ifeq (0, $(shell test -e $(POCO_ODBC_LIB)/libodbc$(LIBLINKEXT); echo $$?))
 SYSLIBS += -lodbc
-ifeq (0, $(shell test -e $(POCO_ODBC_LIB)/libodbcinst$(LIBLINKEXT); echo $$?))
+else ifeq (unixodbc, $(GENTOO_ODBC))
 SYSLIBS += -lodbcinst
 endif
 COMMONFLAGS += -DPOCO_UNIXODBC
@@ -54,7 +54,7 @@
 ##
 ## iODBC
 ##
-else ifeq (0, $(shell test -e $(POCO_ODBC_LIB)/libiodbc$(LIBLINKEXT); echo $$?))
+else ifeq (unixodbc, $(GENTOO_ODBC))
 SYSLIBS += -liodbc -liodbcinst
 COMMONFLAGS += -DPOCO_IODBC -I/usr/include/iodbc
 
diff -Nur poco-1.7.4-all.orig/Makefile poco-1.7.4-all/Makefile
--- poco-1.7.4-all.orig/Makefile	2016-07-21 01:53:00.000000000 +0800
+++ poco-1.7.4-all/Makefile	2016-08-14 14:18:01.235706943 +0800
@@ -95,7 +95,7 @@
 
 install: libexecs
 	mkdir -p $(INSTALLDIR)/include/Poco
-	mkdir -p $(INSTALLDIR)/lib
+	mkdir -p $(INSTALLDIR)/$(LIBDIR)
 	mkdir -p $(INSTALLDIR)/bin
 	for comp in $(filter-out $(foreach f,$(OMIT),$f%),$(COMPONENTS)) ; do \
 		if [ -d "$(POCO_BASE)/$$comp/include" ] ; then \
@@ -109,11 +109,11 @@
 	find $(POCO_BUILD)/lib/$(OSNAME)/$(OSARCH) -name "cygPoco*" -type f -exec cp -f  {} $(INSTALLDIR)/bin \;
 	find $(POCO_BUILD)/lib/$(OSNAME)/$(OSARCH) -name "cygPoco*" -type l -exec cp -Rf {} $(INSTALLDIR)/bin \;
 endif
-	find $(POCO_BUILD)/lib/$(OSNAME)/$(OSARCH) -name "libPoco*" -type f -exec cp -f  {} $(INSTALLDIR)/lib \;
-	find $(POCO_BUILD)/lib/$(OSNAME)/$(OSARCH) -name "libPoco*" -type l -exec cp -Rf {} $(INSTALLDIR)/lib \;
+	find $(POCO_BUILD)/$(LIBDIR) -name "libPoco*" -type f -exec cp -f  {} $(INSTALLDIR)/$(LIBDIR) \;
+	find $(POCO_BUILD)/$(LIBDIR) -name "libPoco*" -type l -exec cp -Rf {} $(INSTALLDIR)/$(LIBDIR) \;
 
-libexecs =  Foundation-libexec XML-libexec JSON-libexec Util-libexec Net-libexec Crypto-libexec NetSSL_OpenSSL-libexec Data-libexec Data/SQLite-libexec Data/ODBC-libexec Data/MySQL-libexec MongoDB-libexec Zip-libexec PageCompiler-libexec PageCompiler/File2Page-libexec
-tests    =  Foundation-tests XML-tests JSON-tests Util-tests Net-tests Crypto-tests NetSSL_OpenSSL-tests Data-tests Data/SQLite-tests Data/ODBC-tests Data/MySQL-tests MongoDB-tests Zip-tests
+libexecs =  Foundation-libexec XML-libexec JSON-libexec Util-libexec Net-libexec Data-libexec Zip-libexec PageCompiler-libexec PageCompiler/File2Page-libexec
+tests    =  Foundation-tests XML-tests JSON-tests Util-tests Net-tests Data-tests Zip-tests
 samples  =  Foundation-samples XML-samples JSON-samples Util-samples Net-samples Crypto-samples NetSSL_OpenSSL-samples Data-samples MongoDB-samples Zip-samples PageCompiler-samples
 cleans   =  Foundation-clean XML-clean JSON-clean Util-clean Net-clean Crypto-clean NetSSL_OpenSSL-clean Data-clean Data/SQLite-clean Data/ODBC-clean Data/MySQL-clean MongoDB-clean Zip-clean PageCompiler-clean PageCompiler/File2Page-clean
 
diff -Nur poco-1.7.4-all.orig/components poco-1.7.4-all/components
--- poco-1.7.4-all.orig/components	2016-07-21 01:52:59.000000000 +0800
+++ poco-1.7.4-all/components	2016-08-14 13:04:39.957638433 +0800
@@ -1,17 +1,9 @@
-CppUnit
-CppUnit/WinTestRunner
 Foundation
 XML
 JSON
 Util
 Net
-Crypto
-NetSSL_OpenSSL
 Data
-Data/SQLite
-Data/ODBC
-Data/MySQL
-MongoDB
 Zip
 PageCompiler
 PageCompiler/File2Page
