diff -ru guidance-0.8.0.o/powermanager/guidance-power-manager.py guidance-0.8.0/powermanager/guidance-power-manager.py
--- guidance-0.8.0.o/powermanager/guidance-power-manager.py	2007-03-26 19:14:50.000000000 +0400
+++ guidance-0.8.0/powermanager/guidance-power-manager.py	2008-06-12 10:09:42.000000000 +0400
@@ -660,7 +660,8 @@
             self.checkIdletime()
 
         # CPU stuff
-        self._updateCpuWidgets()
+	if self.powermanager.hasCpuFreq:
+	    self._updateCpuWidgets()
 
         if self.powermanager.hasBattery:
             on_battery = self.powermanager.onBattery()
