diff -Nur a/opencl/CMakeLists.txt b/opencl/CMakeLists.txt
--- a/opencl/CMakeLists.txt	2026-09-18 00:00:00.000000000 -0700
+++ b/opencl/CMakeLists.txt	2026-09-18 00:00:00.000000000 -0700
@@ -17,6 +17,11 @@
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/../unified-runtime/cmake")
 
-include(FetchOpenCL)
+#include(FetchOpenCL)
+find_package(OpenCLHeaders REQUIRED)
+add_library(OpenCL-Headers ALIAS OpenCL::Headers)
+
+find_package(OpenCL 3.0)
+add_library(OpenCL-ICD ALIAS OpenCL::OpenCL)
 
 add_subdirectory(opencl-aot)
diff -Nur a/sycl/CMakeLists.txt b/sycl/CMakeLists.txt
--- a/sycl/CMakeLists.txt	2026-09-18 00:00:00.000000000 -0700
+++ b/sycl/CMakeLists.txt	2026-09-18 00:00:00.000000000 -0700
@@ -170,7 +170,7 @@
     file(COPY ${OpenCL_INCLUDE_DIR}/CL DESTINATION ${SYCL_INCLUDE_BUILD_DIR})
 
-    # Include OpenCL Headers into final bundle.
-    install(DIRECTORY ${OpenCL_INCLUDE_DIR}/CL
-            DESTINATION ${SYCL_INCLUDE_DIR}
-            COMPONENT OpenCL-Headers)
+    ## Include OpenCL Headers into final bundle.
+    #install(DIRECTORY ${OpenCL_INCLUDE_DIR}/CL
+    #        DESTINATION ${SYCL_INCLUDE_DIR}
+    #        COMPONENT OpenCL-Headers)
   endif()
 endif()
