diff -Nur a/sycl/tools/sycl-trace/CMakeLists.txt b/sycl/tools/sycl-trace/CMakeLists.txt
--- a/sycl/tools/sycl-trace/CMakeLists.txt	2025-10-09 13:35:26.000000000 -0700
+++ b/sycl/tools/sycl-trace/CMakeLists.txt	2025-12-13 01:56:05.092205172 -0800
@@ -109,6 +109,8 @@
   target_include_directories(cuda_trace_collector
     SYSTEM PRIVATE
     ${CUDAToolkit_CUPTI_INCLUDE_DIR}
+    ${CUDAToolkit_INCLUDE_DIRS}
+    ${CUDA_INCLUDE_DIRS}
   )
 
   # FindCUDATookit fails to find CUPTI in CUDA installations from runfile.
@@ -134,7 +136,7 @@
   endif()
   target_link_libraries(cuda_trace_collector
     PRIVATE
-      cudadrv
+      cuda
       ${CUDA_cupti_LIBRARY}
   )
 
