diff -Nur a/opencl/CMakeLists.txt b/opencl/CMakeLists.txt
--- a/opencl/CMakeLists.txt	2025-12-11 13:30:09.000000000 -0800
+++ b/opencl/CMakeLists.txt	2025-12-12 13:51:07.694005634 -0800
@@ -17,6 +17,11 @@
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/../unified-runtime/cmake")
 
-include(FetchOpenCL)
+#include(FetchOpenCL)
+find_package(OpenCLHeaders REQUIRED)
+add_library(OpenCL-Headers ALIAS OpenCL::Headers)
+
+find_package(OpenCL 3.0)
+add_library(OpenCL-ICD ALIAS OpenCL::OpenCL)
 
 add_subdirectory(opencl-aot)
diff -Nur a/sycl/CMakeLists.txt b/sycl/CMakeLists.txt
--- a/sycl/CMakeLists.txt	2025-12-11 13:30:09.000000000 -0800
+++ b/sycl/CMakeLists.txt	2025-12-12 13:51:55.654007322 -0800
@@ -156,10 +156,10 @@
   # e.g. when run LIT tests
   file(COPY ${OpenCL_INCLUDE_DIR}/CL DESTINATION ${SYCL_INCLUDE_BUILD_DIR})
 
-  # Include OpenCL Headers into final bundle.
-  install(DIRECTORY ${OpenCL_INCLUDE_DIR}/CL
-          DESTINATION ${SYCL_INCLUDE_DIR}
-          COMPONENT OpenCL-Headers)
+  ## Include OpenCL Headers into final bundle.
+  #install(DIRECTORY ${OpenCL_INCLUDE_DIR}/CL
+  #        DESTINATION ${SYCL_INCLUDE_DIR}
+  #        COMPONENT OpenCL-Headers)
 endif()
 
 # Option for enabling building the SYCL major release preview library.
