diff -Nur a/sycl/cmake/modules/BuildUnifiedRuntime.cmake b/sycl/cmake/modules/BuildUnifiedRuntime.cmake
--- a/sycl/cmake/modules/BuildUnifiedRuntime.cmake	2026-05-13 16:53:18.000000000 -0700
+++ b/sycl/cmake/modules/BuildUnifiedRuntime.cmake	2026-05-14 21:05:32.990154697 -0700
@@ -69,6 +69,20 @@
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-error")
 endif()
 
+find_package(unified-runtime REQUIRED)
+if(unified-runtime_FOUND)
+add_library(UnifiedRuntimeLoader ALIAS unified-runtime::ur_loader)
+add_library(UnifiedRuntimeCommon ALIAS unified-runtime::ur_common)
+add_library(UnifiedRuntime-Headers ALIAS unified-runtime::ur_headers)
+
+find_package(umf REQUIRED)
+add_library(ur_umf INTERFACE)
+target_link_libraries(ur_umf INTERFACE
+	umf::umf
+	umf::headers
+)
+add_library(UnifiedMemoryFramework ALIAS ur_umf)
+else()
 set(UR_INTREE_SOURCE_DIR "${LLVM_SOURCE_DIR}/../unified-runtime")
 cmake_path(NORMAL_PATH UR_INTREE_SOURCE_DIR OUTPUT_VARIABLE UR_INTREE_SOURCE_DIR)
 
@@ -299,3 +313,4 @@
     ARCHIVE DESTINATION "lib${LLVM_LIBDIR_SUFFIX}" COMPONENT unified-memory-framework
     RUNTIME DESTINATION "bin" COMPONENT unified-memory-framework)
 endif()
+endif()
diff -Nur a/sycl/CMakeLists.txt b/sycl/CMakeLists.txt
--- a/sycl/CMakeLists.txt	2026-05-13 16:53:18.000000000 -0700
+++ b/sycl/CMakeLists.txt	2026-05-14 21:05:32.994372688 -0700
@@ -271,7 +271,7 @@
           ${OUT_HEADERS_IN_CL_DIR}
           ${OUT_HEADERS_IN_STD_DIR}
           ${OUT_HEADERS_IN_SYCLCOMPAT_DIR}
-          ${OUT_HEADERS_IN_UR_DIR}
+	  #${OUT_HEADERS_IN_UR_DIR}
           sycl-device-aspect-macros-header
           )
 
@@ -280,23 +280,23 @@
           ${OUT_HEADERS_IN_CL_DIR}
           ${OUT_HEADERS_IN_STD_DIR}
           ${OUT_HEADERS_IN_SYCLCOMPAT_DIR}
-          ${OUT_HEADERS_IN_UR_DIR}
+	  #${OUT_HEADERS_IN_UR_DIR}
   DEPENDS ${HEADERS_IN_SYCL_DIR}
           ${HEADERS_IN_CL_DIR}
           ${HEADERS_IN_STD_DIR}
           ${HEADERS_IN_SYCLCOMPAT_DIR}
-          ${HEADERS_IN_UR_DIR}
+	  #${HEADERS_IN_UR_DIR}
   COMMAND ${CMAKE_COMMAND} -E copy_directory ${sycl_inc_dir}/sycl ${SYCL_INCLUDE_BUILD_DIR}/sycl
   COMMAND ${CMAKE_COMMAND} -E copy_directory ${sycl_inc_dir}/CL ${SYCL_INCLUDE_BUILD_DIR}/CL
   COMMAND ${CMAKE_COMMAND} -E copy_directory ${sycl_inc_dir}/std ${SYCL_INCLUDE_BUILD_DIR}/std
-  COMMAND ${CMAKE_COMMAND} -E copy_directory ${UNIFIED_RUNTIME_INCLUDE_DIR}/unified-runtime ${SYCL_INCLUDE_BUILD_DIR}/unified-runtime
+  #COMMAND ${CMAKE_COMMAND} -E copy_directory ${UNIFIED_RUNTIME_INCLUDE_DIR}/unified-runtime ${SYCL_INCLUDE_BUILD_DIR}/unified-runtime
   COMMENT "Copying SYCL headers ...")
 
 # Copy SYCL headers from source to install directory
 install(DIRECTORY "${sycl_inc_dir}/sycl" DESTINATION ${SYCL_INCLUDE_DIR} COMPONENT sycl-headers)
 install(DIRECTORY "${sycl_inc_dir}/CL" DESTINATION ${SYCL_INCLUDE_DIR}/ COMPONENT sycl-headers)
 install(DIRECTORY "${sycl_inc_dir}/std" DESTINATION ${SYCL_INCLUDE_DIR} COMPONENT sycl-headers)
-install(DIRECTORY "${UNIFIED_RUNTIME_INCLUDE_DIR}/unified-runtime" DESTINATION ${SYCL_INCLUDE_DIR} COMPONENT sycl-headers)
+#install(DIRECTORY "${UNIFIED_RUNTIME_INCLUDE_DIR}/unified-runtime" DESTINATION ${SYCL_INCLUDE_DIR} COMPONENT sycl-headers)
 
 if (WIN32)
   set(SYCL_RT_LIBS sycl${SYCL_MAJOR_VERSION})
@@ -495,8 +495,8 @@
      sycl-headers-extras
      sycl
      libsycldevice
-     unified-memory-framework
-     unified-runtime-loader
+     #unified-memory-framework
+     #unified-runtime-loader
      ${XPTIFW_LIBS}
      ${SYCL_TOOLCHAIN_DEPS}
 )
@@ -549,8 +549,8 @@
       "CUDA support requires adding \"libclc\" to the CMake argument \"LLVM_ENABLE_PROJECTS\"")
   endif()
 
-  add_dependencies(sycl-toolchain ur_adapter_cuda)
-  list(APPEND SYCL_TOOLCHAIN_DEPLOY_COMPONENTS ur_adapter_cuda)
+  add_dependencies(sycl-toolchain unified-runtime::ur_adapter_cuda)
+  list(APPEND SYCL_TOOLCHAIN_DEPLOY_COMPONENTS unified-runtime::ur_adapter_cuda)
 endif()
 
 if("hip" IN_LIST SYCL_ENABLE_BACKENDS)
@@ -565,8 +565,8 @@
       "HIP support requires adding \"lld\" to the CMake argument \"LLVM_ENABLE_PROJECTS\"")
   endif()
 
-  add_dependencies(sycl-toolchain ur_adapter_hip)
-  list(APPEND SYCL_TOOLCHAIN_DEPLOY_COMPONENTS ur_adapter_hip)
+  add_dependencies(sycl-toolchain unified-runtime::ur_adapter_hip)
+  list(APPEND SYCL_TOOLCHAIN_DEPLOY_COMPONENTS unified-runtime::ur_adapter_hip)
 endif()
 
 if("offload" IN_LIST SYCL_ENABLE_BACKENDS)
