diff -Nur a/sycl/tools/sycl-trace/CMakeLists.txt b/sycl/tools/sycl-trace/CMakeLists.txt
--- a/sycl/tools/sycl-trace/CMakeLists.txt	2025-01-31 11:09:57.000000000 -0800
+++ b/sycl/tools/sycl-trace/CMakeLists.txt	2025-02-03 21:56:41.475584927 -0800
@@ -11,13 +11,21 @@
 )
 
 if ("level_zero" IN_LIST SYCL_ENABLE_BACKENDS)
+  find_package(PkgConfig REQUIRED)
+  pkg_search_module(LEVEL_ZERO REQUIRED level-zero)
   add_library(ze_trace_collector SHARED
     ze_trace_collector.cpp
   )
+  target_include_directories(ze_trace_collector PRIVATE
+    ${LEVEL_ZERO_INCLUDE_DIRS}
+  )
   set(EXTRA_TARGETS_TO_INSTALL
     ze_trace_collector
   )
   add_dependencies(sycl-trace ze_trace_collector)
+  target_include_directories(sycl-trace PRIVATE
+    ${LEVEL_ZERO_INCLUDE_DIRS}
+  )
 endif()
 
 if ("cuda" IN_LIST SYCL_ENABLE_BACKENDS)
@@ -42,7 +50,10 @@
 
 # To get L0 loader
 if ("level_zero" IN_LIST SYCL_ENABLE_BACKENDS)
-  target_link_libraries(ze_trace_collector PRIVATE LevelZeroLoader-Headers)
+  find_package(PkgConfig REQUIRED)
+  pkg_search_module(LEVEL_ZERO_LOADER REQUIRED libze_loader)
+
+  target_link_libraries(ze_trace_collector PRIVATE ${LEVEL_ZERO_LOADER_LIBRARIES})
   target_compile_definitions(ze_trace_collector PRIVATE SYCL_HAS_LEVEL_ZERO)
   target_link_libraries(ze_trace_collector PRIVATE xptifw)
 
