diff -Nur a/sycl/source/CMakeLists.txt b/sycl/source/CMakeLists.txt
--- a/sycl/source/CMakeLists.txt	2025-12-11 13:30:09.000000000 -0800
+++ b/sycl/source/CMakeLists.txt	2025-12-12 14:44:56.037452621 -0800
@@ -160,7 +160,7 @@
 
   if (LLVM_ENABLE_ZSTD)
     target_compile_definitions(${LIB_OBJ_NAME} PRIVATE SYCL_RT_ZSTD_AVAILABLE)
-    target_link_libraries(${LIB_NAME} PRIVATE ${zstd_STATIC_LIBRARY})
+    target_link_libraries(${LIB_NAME} PRIVATE ${zstd_LIBRARY})
     target_include_directories(${LIB_OBJ_NAME} PRIVATE ${zstd_INCLUDE_DIR})
   endif()
 
