diff -Nur a/sycl/source/CMakeLists.txt b/sycl/source/CMakeLists.txt
--- a/sycl/source/CMakeLists.txt	2025-01-31 11:09:57.000000000 -0800
+++ b/sycl/source/CMakeLists.txt	2025-02-03 22:54:50.602374403 -0800
@@ -72,7 +72,7 @@
   if (NOT LLVM_ENABLE_ZSTD)
     target_compile_definitions(${LIB_OBJ_NAME} PRIVATE SYCL_RT_ZSTD_NOT_AVAIABLE)
   else()
-    target_link_libraries(${LIB_NAME} PRIVATE ${zstd_STATIC_LIBRARY})
+    target_link_libraries(${LIB_NAME} PRIVATE ${zstd_LIBRARY})
     target_include_directories(${LIB_OBJ_NAME} PRIVATE ${zstd_INCLUDE_DIR})
   endif()
 
