diff -Nur a/cmake/FetchLevelZero.cmake b/cmake/FetchLevelZero.cmake
--- a/cmake/FetchLevelZero.cmake	2025-05-05 01:51:11.000000000 -0700
+++ b/cmake/FetchLevelZero.cmake	2025-05-14 01:35:52.070544773 -0700
@@ -131,11 +131,11 @@
 else()
 
     # Check if UR_COMPUTE_RUNTIME_REPO is set. Throw if not.
-    if (UR_COMPUTE_RUNTIME_REPO STREQUAL "")
-        message(FATAL_ERROR "UR_COMPUTE_RUNTIME_FETCH_REPO is set to OFF but UR_COMPUTE_RUNTIME_REPO is not set. Please set it to the compute runtime repo.")
-    endif()
+    #if (UR_COMPUTE_RUNTIME_REPO STREQUAL "")
+    #    message(FATAL_ERROR "UR_COMPUTE_RUNTIME_FETCH_REPO is set to OFF but UR_COMPUTE_RUNTIME_REPO is not set. Please set it to the compute runtime repo.")
+    #endif()
 
-    set(COMPUTE_RUNTIME_LEVEL_ZERO_INCLUDE "${UR_COMPUTE_RUNTIME_REPO}/level_zero/include")
+    set(COMPUTE_RUNTIME_LEVEL_ZERO_INCLUDE "${UR_COMPUTE_RUNTIME_DIR}/level_zero/include")
     set(COMPUTE_RUNTIME_REPO_PATH "${UR_COMPUTE_RUNTIME_REPO}")
 endif()
 
