diff -Nur a/cmake/FetchLevelZero.cmake b/cmake/FetchLevelZero.cmake
--- a/cmake/FetchLevelZero.cmake
+++ b/cmake/FetchLevelZero.cmake
@@ -97,6 +97,7 @@
               "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>"
 )
 
+if (NOT DEFINED UR_COMPUTE_RUNTIME_DIR)
 if (UR_COMPUTE_RUNTIME_REPO STREQUAL "")
 set(UR_COMPUTE_RUNTIME_REPO "https://github.com/intel/compute-runtime.git")
 endif()
@@ -112,6 +113,10 @@
 endif()
 add_library(ComputeRuntimeLevelZero-Headers INTERFACE)
 set(COMPUTE_RUNTIME_LEVEL_ZERO_INCLUDE "${exp-headers_SOURCE_DIR}/../..")
+else()
+add_library(ComputeRuntimeLevelZero-Headers INTERFACE)
+set(COMPUTE_RUNTIME_LEVEL_ZERO_INCLUDE "${UR_COMPUTE_RUNTIME_DIR}")
+endif()
 message(STATUS "Level Zero Adapter: Using Level Zero headers from ${COMPUTE_RUNTIME_LEVEL_ZERO_INCLUDE}")
 target_include_directories(ComputeRuntimeLevelZero-Headers
     INTERFACE "$<BUILD_INTERFACE:${COMPUTE_RUNTIME_LEVEL_ZERO_INCLUDE}>"
diff -Nur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -133,6 +133,8 @@
     endif()
 endif()
 
+add_compile_definitions(CL_ENABLE_BETA_EXTENSIONS)
+
 if(UR_ENABLE_TRACING)
     add_compile_definitions(UR_ENABLE_TRACING)
 
