diff -Nur a/cmake/FetchLevelZero.cmake b/cmake/FetchLevelZero.cmake
--- a/cmake/FetchLevelZero.cmake	2025-02-13 03:43:34.000000000 -0800
+++ b/cmake/FetchLevelZero.cmake	2025-03-17 21:36:52.402009454 -0700
@@ -97,21 +97,25 @@
               "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>"
 )
 
-if (UR_COMPUTE_RUNTIME_REPO STREQUAL "")
-set(UR_COMPUTE_RUNTIME_REPO "https://github.com/intel/compute-runtime.git")
-endif()
-if (UR_COMPUTE_RUNTIME_TAG STREQUAL "")
-set(UR_COMPUTE_RUNTIME_TAG 24.39.31294.12)
-endif()
-include(FetchContent)
-# Sparse fetch only the dir with level zero headers for experimental features to avoid pulling in the entire compute-runtime.
-FetchContentSparse_Declare(exp-headers ${UR_COMPUTE_RUNTIME_REPO} "${UR_COMPUTE_RUNTIME_TAG}" "level_zero/include")
-FetchContent_GetProperties(exp-headers)
-if(NOT exp-headers_POPULATED)
-  FetchContent_Populate(exp-headers)
+if(NOT DEFINED UR_COMPUTE_RUNTIME_DIR)
+    if (UR_COMPUTE_RUNTIME_REPO STREQUAL "")
+        set(UR_COMPUTE_RUNTIME_REPO "https://github.com/intel/compute-runtime.git")
+    endif()
+    if (UR_COMPUTE_RUNTIME_TAG STREQUAL "")
+        set(UR_COMPUTE_RUNTIME_TAG 24.39.31294.12)
+    endif()
+    include(FetchContent)
+    # Sparse fetch only the dir with level zero headers for experimental features to avoid pulling in the entire compute-runtime.
+    FetchContentSparse_Declare(exp-headers ${UR_COMPUTE_RUNTIME_REPO} "${UR_COMPUTE_RUNTIME_TAG}" "level_zero/include")
+    FetchContent_GetProperties(exp-headers)
+    if(NOT exp-headers_POPULATED)
+        FetchContent_Populate(exp-headers)
+    endif()
+    set(COMPUTE_RUNTIME_LEVEL_ZERO_INCLUDE "${exp-headers_SOURCE_DIR}/../..")
+else()
+    set(COMPUTE_RUNTIME_LEVEL_ZERO_INCLUDE "${UR_COMPUTE_RUNTIME_DIR}")
 endif()
 add_library(ComputeRuntimeLevelZero-Headers INTERFACE)
-set(COMPUTE_RUNTIME_LEVEL_ZERO_INCLUDE "${exp-headers_SOURCE_DIR}/../..")
 message(STATUS "Level Zero Adapter: Using Level Zero headers from ${COMPUTE_RUNTIME_LEVEL_ZERO_INCLUDE}")
 target_include_directories(ComputeRuntimeLevelZero-Headers
     INTERFACE "$<BUILD_INTERFACE:${COMPUTE_RUNTIME_LEVEL_ZERO_INCLUDE}>"
diff -Nur a/source/adapters/level_zero/common.cpp b/source/adapters/level_zero/common.cpp
--- a/source/adapters/level_zero/common.cpp	2025-02-13 03:43:34.000000000 -0800
+++ b/source/adapters/level_zero/common.cpp	2025-03-17 22:01:41.695395690 -0700
@@ -11,7 +11,7 @@
 #include "common.hpp"
 #include "logger/ur_logger.hpp"
 #include "usm.hpp"
-#include <level_zero/include/ze_intel_gpu.h>
+#include <level_zero/ze_intel_gpu.h>
 
 ur_result_t ze2urResult(ze_result_t ZeResult) {
   if (ZeResult == ZE_RESULT_SUCCESS)
diff -Nur a/source/adapters/level_zero/device.hpp b/source/adapters/level_zero/device.hpp
--- a/source/adapters/level_zero/device.hpp	2025-02-13 03:43:34.000000000 -0800
+++ b/source/adapters/level_zero/device.hpp	2025-03-17 21:37:27.468677372 -0700
@@ -20,7 +20,7 @@
 
 #include "adapters/level_zero/platform.hpp"
 #include "common.hpp"
-#include <level_zero/include/ze_intel_gpu.h>
+#include <level_zero/ze_intel_gpu.h>
 #include <ur/ur.hpp>
 #include <ur_ddi.h>
 #include <ze_api.h>
