commit 9b46c1a794e48b3971c0ffae5993cdd393a10117
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Mon May 9 06:49:25 2011 -0400

    added crypt stub

diff --git a/libc/bionic/stubs.c b/libc/bionic/stubs.c
index c3ed1b1..ff32f0e 100644
--- a/libc/bionic/stubs.c
+++ b/libc/bionic/stubs.c
@@ -513,3 +513,10 @@ int ether_hostton(const char *hostname, struct ether_addr *addr)
     errno = ENOSYS;
     return -1;
 }
+
+char *crypt(const char *key, const char *salt)
+{
+    fprintf(stderr, "FIX ME! implement %s() %s:%d\n", __FUNCTION__, __FILE__, __LINE__);
+    errno = ENOSYS;
+    return NULL;
+}
diff --git a/libc/include/unistd.h b/libc/include/unistd.h
index 83df392..8ae938b 100644
--- a/libc/include/unistd.h
+++ b/libc/include/unistd.h
@@ -202,6 +202,7 @@ extern int sethostname(const char *, size_t);
 // FIXME: stubs
 extern pid_t  getsid(pid_t);
 extern int getlogin_r(char* name, size_t namesize);
+extern char *crypt(const char *key, const char *salt);
 
 /* Used to retry syscalls that can return EINTR. */
 #define TEMP_FAILURE_RETRY(exp) ({         \
