commit e9c1d3ebb289752d39a1eb24c123b59ca454d726
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Thu Apr 28 08:15:43 2011 -0400

    added getlogin_r, tcdrain, ether_aton_r, mempcpy stubs

diff --git a/libc/bionic/stubs.c b/libc/bionic/stubs.c
index bb9a697..07096e0 100644
--- a/libc/bionic/stubs.c
+++ b/libc/bionic/stubs.c
@@ -307,6 +307,13 @@ getlogin(void)
     }
 }
 
+int getlogin_r(char *buf, size_t bufsize)
+{
+    fprintf(stderr, "FIX ME! implement %s() %s:%d\n", __FUNCTION__, __FILE__, __LINE__);
+    errno = ENOSYS;
+    return -1;
+}
+
 struct group*
 getgrgid(gid_t gid)
 {
@@ -472,3 +479,24 @@ struct utmp *getutline(struct utmp *ut)
     return NULL;
 }
 
+int tcdrain(int fd)
+{
+    fprintf(stderr, "FIX ME! implement %s() %s:%d\n", __FUNCTION__, __FILE__, __LINE__);
+    errno = ENOSYS;
+    return -1;
+}
+
+void *mempcpy(void *dest, const void *src, size_t n)
+{
+    fprintf(stderr, "FIX ME! implement %s() %s:%d\n", __FUNCTION__, __FILE__, __LINE__);
+    errno = ENOSYS;
+    return NULL;
+}
+
+struct ether_addr *ether_aton_r(const char *asc,
+                                       struct ether_addr *addr)
+{
+    fprintf(stderr, "FIX ME! implement %s() %s:%d\n", __FUNCTION__, __FILE__, __LINE__);
+    errno = ENOSYS;
+    return NULL;
+}
diff --git a/libc/include/net/if_ether.h b/libc/include/net/if_ether.h
index 121f9ac..0cd6d8d 100644
--- a/libc/include/net/if_ether.h
+++ b/libc/include/net/if_ether.h
@@ -206,6 +206,8 @@ __BEGIN_DECLS
 char *	ether_ntoa __P((const struct ether_addr *));
 struct ether_addr *
 	ether_aton __P((const char *));
+struct ether_addr *
+	ether_aton_r(const char *asc, struct ether_addr *addr);
 int	ether_ntohost __P((char *, const struct ether_addr *));
 int	ether_hostton __P((const char *, struct ether_addr *));
 int	ether_line __P((const char *, struct ether_addr *, char *));
diff --git a/libc/include/string.h b/libc/include/string.h
index 011f41f..97de4fc 100644
--- a/libc/include/string.h
+++ b/libc/include/string.h
@@ -39,6 +39,7 @@ extern void*  memchr(const void *, int, size_t);
 extern void*  memrchr(const void *, int, size_t);
 extern int    memcmp(const void *, const void *, size_t);
 extern void*  memcpy(void *, const void *, size_t);
+extern void*  mempcpy(void *dest, const void *src, size_t n);
 extern void*  memmove(void *, const void *, size_t);
 extern void*  memset(void *, int, size_t);
 extern void*  memmem(const void *, size_t, const void *, size_t);
diff --git a/libc/include/termios.h b/libc/include/termios.h
index 99276dd..803a01f 100644
--- a/libc/include/termios.h
+++ b/libc/include/termios.h
@@ -113,6 +113,8 @@ static __inline__ void cfmakeraw(struct termios *s)
     s->c_cflag |= CS8;
 }
 
+extern int tcdrain(int fd);
+
 __END_DECLS
 
 #endif /* _TERMIOS_H_ */
