commit 1a8785f270d890ab46129c24a556b1cbf40f5d90
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Thu Apr 28 07:51:36 2011 -0400

    added hasmntopt and addmntent stubs

diff --git a/libc/bionic/stubs.c b/libc/bionic/stubs.c
index e6bd3ef..0c5ac40 100644
--- a/libc/bionic/stubs.c
+++ b/libc/bionic/stubs.c
@@ -444,3 +444,17 @@ ssize_t getline(char **lineptr, size_t *n, FILE *stream)
     errno = ENOSYS;
     return -1;
 }
+
+int addmntent(FILE *fp, const struct mntent *mnt)
+{
+    fprintf(stderr, "FIX ME! implement %s() %s:%d\n", __FUNCTION__, __FILE__, __LINE__);
+    errno = ENOSYS;
+    return -1;
+}
+
+char *hasmntopt(const struct mntent *mnt, const char *opt)
+{
+    fprintf(stderr, "FIX ME! implement %s() %s:%d\n", __FUNCTION__, __FILE__, __LINE__);
+    errno = ENOSYS;
+    return NULL;
+}
diff --git a/libc/include/mntent.h b/libc/include/mntent.h
index 5d3a66b..598a932 100644
--- a/libc/include/mntent.h
+++ b/libc/include/mntent.h
@@ -51,6 +51,8 @@ extern FILE *setmntent(const char *filename, const char *type);
 extern int endmntent(FILE *fp);
 extern struct mntent *getmntent_r(FILE *fp, struct mntent *mntbuf,
                                   char *buf, int buflen);
+extern int addmntent(FILE *fp, const struct mntent *mnt);
+extern char *hasmntopt(const struct mntent *mnt, const char *opt);
 
 __END_DECLS
 
