commit 57bbaaa5f2cfdf4ce69670fb324420bb76517d9c
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Thu May 5 19:21:39 2011 -0400

    more syscalls

diff --git a/libc/SYSCALLS.TXT b/libc/SYSCALLS.TXT
index 09c3983..3465943 100644
--- a/libc/SYSCALLS.TXT
+++ b/libc/SYSCALLS.TXT
@@ -83,6 +83,7 @@ int     capset(cap_user_header_t header, const cap_user_data_t data) 185
 int     sigaltstack(const stack_t*, stack_t*) 186
 int     acct(const char*  filepath)  51
 int     sethostname(const char *, size_t) 74
+int     pivot_root(const char *, const char *) 117,118,117
 
 # file descriptors
 ssize_t     read (int, void*, size_t)        3
diff --git a/libc/bionic/stubs.c b/libc/bionic/stubs.c
index 07096e0..acf724c 100644
--- a/libc/bionic/stubs.c
+++ b/libc/bionic/stubs.c
@@ -383,7 +383,7 @@ FILE *setmntent(const char *filename, const char *type) {
 int endmntent(FILE *fp) {
     fprintf(stderr, "FIX ME! implement %s() %s:%d\n", __FUNCTION__, __FILE__, __LINE__);
     errno = ENOSYS;
-    return -ENOSYS;
+    return -1;
 }
 
 char* ttyname(int fd)
@@ -397,7 +397,7 @@ int ttyname_r(int fd, char *buf, size_t buflen)
 {
     fprintf(stderr, "FIX ME! implement %s() %s:%d\n", __FUNCTION__, __FILE__, __LINE__);
     errno = ENOSYS;
-    return -ENOSYS;
+    return -1;
 }
 
 struct netent *getnetbyaddr(uint32_t net, int type)
@@ -500,3 +500,10 @@ struct ether_addr *ether_aton_r(const char *asc,
     errno = ENOSYS;
     return NULL;
 }
+
+int ether_hostton(const char *hostname, struct ether_addr *addr)
+{
+    fprintf(stderr, "FIX ME! implement %s() %s:%d\n", __FUNCTION__, __FILE__, __LINE__);
+    errno = ENOSYS;
+    return -1;
+}
