commit 96547bc27accbe796c08c9983aafacd22cc514a7
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Sun May 1 12:02:38 2011 -0400

    added sys/personality.h

diff --git a/libc/include/sys/personality.h b/libc/include/sys/personality.h
new file mode 100644
index 0000000..2534414
--- /dev/null
+++ b/libc/include/sys/personality.h
@@ -0,0 +1,6 @@
+#ifndef _SYS_PERSONALITY_H_
+#define _SYS_PERSONALITY_H_
+
+#include <linux/personality.h>
+
+#endif /* _SYS_PERSONALITY_H_ */
