commit 056b57e1ce79c9fe05a11884629e13b68d71353c
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Thu May 12 23:37:26 2011 -0400

    added S_BLKSIZE in sys/stat.h

diff --git a/libc/include/sys/stat.h b/libc/include/sys/stat.h
index 2e5488a..98072f7 100644
--- a/libc/include/sys/stat.h
+++ b/libc/include/sys/stat.h
@@ -37,6 +37,10 @@
 
 __BEGIN_DECLS
 
+#ifndef S_BLKSIZE
+#define S_BLKSIZE 512
+#endif
+
 /* POSIX.1b objects.  Note that these macros always evaluate to zero.  But
    they do it by enforcing the correct use of the macros.  */
 #define S_TYPEISMQ(buf)  ((buf)->st_mode - (buf)->st_mode)
