commit 0b74d398569a115e767ab20253886383ee4434cd
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Thu Apr 28 03:25:26 2011 -0400

    added pw_gecos (real name) field to struct passwd

diff --git a/libc/include/pwd.h b/libc/include/pwd.h
index 6f3fad5..c33f407 100644
--- a/libc/include/pwd.h
+++ b/libc/include/pwd.h
@@ -104,6 +104,7 @@ struct passwd
     char* pw_passwd;
     uid_t pw_uid;
     gid_t pw_gid;
+    char* pw_gecos;
     char* pw_dir;
     char* pw_shell;
 };
