commit 4c321226918fa7a8252194ec653130ee8b48bc03
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Sun May 8 16:43:38 2011 -0400

    /system/etc/resolv.conf -> /etc/resolv.conf

diff --git a/libc/include/netdb.h b/libc/include/netdb.h
index c2e08ea..690b276 100644
--- a/libc/include/netdb.h
+++ b/libc/include/netdb.h
@@ -66,12 +66,12 @@
 #include <sys/socket.h>
 
 #ifndef _PATH_HEQUIV
-# define	_PATH_HEQUIV	"/system/etc/hosts.equiv"
+# define	_PATH_HEQUIV	"/etc/hosts.equiv"
 #endif
-#define	_PATH_HOSTS	"/system/etc/hosts"
-#define	_PATH_NETWORKS	"/system/etc/networks"
-#define	_PATH_PROTOCOLS	"/system/etc/protocols"
-#define	_PATH_SERVICES	"/system/etc/services"
+#define	_PATH_HOSTS	"/etc/hosts"
+#define	_PATH_NETWORKS	"/etc/networks"
+#define	_PATH_PROTOCOLS	"/etc/protocols"
+#define	_PATH_SERVICES	"/etc/services"
 
 #define  MAXHOSTNAMELEN  256
 
