commit 127a65daf63343b704bacc0f538b02d03e5af4ea
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Thu Apr 28 00:59:06 2011 -0400

    added setmntent as a stub

diff --git a/libc/bionic/stubs.c b/libc/bionic/stubs.c
index d495674..710f2bd 100644
--- a/libc/bionic/stubs.c
+++ b/libc/bionic/stubs.c
@@ -355,6 +355,11 @@ struct mntent* getmntent(FILE* f)
     return NULL;
 }
 
+FILE *setmntent(const char *filename, const char *type) {
+    fprintf(stderr, "FIX ME! implement setmntent() %s:%d\n", __FILE__, __LINE__);
+    return NULL;
+}
+
 char* ttyname(int fd)
 {
     fprintf(stderr, "FIX ME! implement ttyname() %s:%d\n", __FILE__, __LINE__);
diff --git a/libc/include/mntent.h b/libc/include/mntent.h
index b83da1f..292cfc2 100644
--- a/libc/include/mntent.h
+++ b/libc/include/mntent.h
@@ -47,6 +47,7 @@ __BEGIN_DECLS
 
 
 struct mntent* getmntent(FILE*);
+FILE *setmntent(const char *filename, const char *type);
 
 __END_DECLS
 

