commit b0af3f09289955b7f35debc3daf5402b356aabbe
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Wed Apr 27 23:31:02 2011 -0400

    added definition of mkdtemp to unistd.h

diff --git a/libc/include/unistd.h b/libc/include/unistd.h
index 29154a2..75d0b5e 100644
--- a/libc/include/unistd.h
+++ b/libc/include/unistd.h
@@ -186,6 +186,8 @@ extern int cacheflush(long start, long end, long flags);
 extern pid_t tcgetpgrp(int fd);
 extern int   tcsetpgrp(int fd, pid_t _pid);
 
+extern char * mkdtemp(char *template);
+
 #if 0 /* MISSING FROM BIONIC */
 extern pid_t  getsid(pid_t);
 extern int execvpe(const char *, char * const *, char * const *);
