commit a701246fe45c4684ca8c3a8767541c74fda420f6
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Thu Apr 28 07:45:33 2011 -0400

    added stubs for getdelim, getline

diff --git a/libc/bionic/stubs.c b/libc/bionic/stubs.c
index dd863ad..e6bd3ef 100644
--- a/libc/bionic/stubs.c
+++ b/libc/bionic/stubs.c
@@ -341,18 +341,19 @@ getgrnam(const char *name)
 
 struct netent* getnetbyname(const char *name)
 {
-    fprintf(stderr, "FIX ME! implement getgrnam() %s:%d\n", __FILE__, __LINE__);
+    fprintf(stderr, "FIX ME! implement %s() %s:%d\n", __FUNCTION__, __FILE__, __LINE__);
     errno = ENOSYS;
     return NULL;
 }
 
 void endpwent(void)
 {
+    fprintf(stderr, "FIX ME! implement %s() %s:%d\n", __FUNCTION__, __FILE__, __LINE__);
 }
 
 struct mntent* getmntent(FILE* f)
 {
-    fprintf(stderr, "FIX ME! implement getmntent() %s:%d\n", __FILE__, __LINE__);
+    fprintf(stderr, "FIX ME! implement %s() %s:%d\n", __FUNCTION__, __FILE__, __LINE__);
     errno = ENOSYS;
     return NULL;
 }
@@ -361,33 +362,33 @@ struct mntent* getmntent(FILE* f)
 struct mntent *getmntent_r(FILE *fp, struct mntent *mntbuf,
                                   char *buf, int buflen)
 {
-    fprintf(stderr, "FIX ME! implement getmntent_r() %s:%d\n", __FILE__, __LINE__);
+    fprintf(stderr, "FIX ME! implement %s() %s:%d\n", __FUNCTION__, __FILE__, __LINE__);
     errno = ENOSYS;
     return NULL;
 }
 
 FILE *setmntent(const char *filename, const char *type) {
-    fprintf(stderr, "FIX ME! implement setmntent() %s:%d\n", __FILE__, __LINE__);
+    fprintf(stderr, "FIX ME! implement %s() %s:%d\n", __FUNCTION__, __FILE__, __LINE__);
     errno = ENOSYS;
     return NULL;
 }
 
 int endmntent(FILE *fp) {
-    fprintf(stderr, "FIX ME! implement endmntent() %s:%d\n", __FILE__, __LINE__);
+    fprintf(stderr, "FIX ME! implement %s() %s:%d\n", __FUNCTION__, __FILE__, __LINE__);
     errno = ENOSYS;
     return -ENOSYS;
 }
 
 char* ttyname(int fd)
 {
-    fprintf(stderr, "FIX ME! implement ttyname() %s:%d\n", __FILE__, __LINE__);
+    fprintf(stderr, "FIX ME! implement %s() %s:%d\n", __FUNCTION__, __FILE__, __LINE__);
     errno = ENOSYS;
     return NULL;
 }
 
 int ttyname_r(int fd, char *buf, size_t buflen)
 {
-    fprintf(stderr, "FIX ME! implement ttyname_r() %s:%d\n", __FILE__, __LINE__);
+    fprintf(stderr, "FIX ME! implement %s() %s:%d\n", __FUNCTION__, __FILE__, __LINE__);
     errno = ENOSYS;
     return -ENOSYS;
 }
@@ -430,3 +431,16 @@ void endusershell(void)
     fprintf(stderr, "FIX ME! implement %s() %s:%d\n", __FUNCTION__, __FILE__, __LINE__);
 }
 
+ssize_t getdelim(char **lineptr, size_t *n, int delim, FILE *stream)
+{
+    fprintf(stderr, "FIX ME! implement %s() %s:%d\n", __FUNCTION__, __FILE__, __LINE__);
+    errno = ENOSYS;
+    return -1;
+}
+
+ssize_t getline(char **lineptr, size_t *n, FILE *stream)
+{
+    fprintf(stderr, "FIX ME! implement %s() %s:%d\n", __FUNCTION__, __FILE__, __LINE__);
+    errno = ENOSYS;
+    return -1;
+}
diff --git a/libc/include/stdio.h b/libc/include/stdio.h
index 7b636cf..b0fb57a 100644
--- a/libc/include/stdio.h
+++ b/libc/include/stdio.h
@@ -327,6 +327,11 @@ char	*tempnam(const char *, const char *);
 #endif
 __END_DECLS
 
+#if __POSIX_VISIBLE >= 200809
+ssize_t getline(char **lineptr, size_t *n, FILE *stream);
+ssize_t getdelim(char **lineptr, size_t *n, int delim, FILE *stream);
+#endif /* __POSIX_VISIBLE >= 200809 */
+
 #endif /* __BSD_VISIBLE || __POSIX_VISIBLE || __XPG_VISIBLE */
 
 /*
