commit 3ac6b6f1c8dce2b52f13b97d4d265b87783a99e9
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Thu May 12 21:31:55 2011 -0400

    debugging in linker, remove Android debuggerd sighandler

diff --git a/linker/Android.mk b/linker/Android.mk
index e2b7abb..b7325b6 100644
--- a/linker/Android.mk
+++ b/linker/Android.mk
@@ -34,7 +34,7 @@ LOCAL_CFLAGS += -DLINKER_AREA_SIZE=$(LINKER_AREA_SIZE)
 
 # Set LINKER_DEBUG to either 1 or 0
 #
-LOCAL_CFLAGS += -DLINKER_DEBUG=0
+LOCAL_CFLAGS += -DLINKER_DEBUG=1
 
 # we need to access the Bionic private header <bionic_tls.h>
 # in the linker; duplicate the HAVE_ARM_TLS_REGISTER definition
diff --git a/linker/debugger.c b/linker/debugger.c
index abb383c..24a0555 100644
--- a/linker/debugger.c
+++ b/linker/debugger.c
@@ -89,6 +89,7 @@ static int socket_abstract_client(const char *name, int type)
 
 void debugger_signal_handler(int n)
 {
+#if 0
     unsigned tid;
     int s;
 
@@ -118,10 +119,12 @@ void debugger_signal_handler(int n)
 
     /* remove our net so we fault for real when we return */
     signal(n, SIG_IGN);
+#endif /* 0 */
 }
 
 void debugger_init()
 {
+#if 0
     signal(SIGILL, debugger_signal_handler);
     signal(SIGABRT, debugger_signal_handler);
     signal(SIGBUS, debugger_signal_handler);
@@ -129,4 +132,5 @@ void debugger_init()
     signal(SIGSEGV, debugger_signal_handler);
     signal(SIGSTKFLT, debugger_signal_handler);
     signal(SIGPIPE, debugger_signal_handler);
+#endif /* 0 */
 }
