commit e79841d819d6ac60b06eeefeaf73c007191431c1
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Mon May 9 16:50:15 2011 -0400

    added asm/page.h to stdlib.h

diff --git a/libc/include/stdlib.h b/libc/include/stdlib.h
index 97d8e46..df163b9 100644
--- a/libc/include/stdlib.h
+++ b/libc/include/stdlib.h
@@ -42,6 +42,9 @@
 #include <strings.h>
 #include <memory.h>
 
+/* gcc relies on PAGE_SIZE being defined */
+#include <asm/page.h>
+
 __BEGIN_DECLS
 
 #define EXIT_FAILURE 1
