commit cd44f633ac533267986730acbeeedf01edf59add
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Fri Apr 29 17:36:18 2011 -0400

    fixed __{BEGIN,END}_DECL -> DECLS  in sys/sem.h

diff --git a/libc/include/sys/sem.h b/libc/include/sys/sem.h
index 7777f4b..72b29a1 100644
--- a/libc/include/sys/sem.h
+++ b/libc/include/sys/sem.h
@@ -3,13 +3,13 @@
 
 #include <linux/sem.h>
 
-__BEGIN_DECL
+__BEGIN_DECLS
 
 extern int semctl(int semid, int semnum, int cmd, ...);
 extern int semget(key_t key, int nsems, int semflg);
 extern int semop(int semid, struct sembuf *sops, size_t nsops);
 
-__END_DECL
+__END_DECLS
 
 #endif /* _HAVE_SYS_SEM_H_ */
 
