commit ce1fc0a41b39b81d7943c9e12c24676a0e14fd1e
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Thu Apr 28 06:52:47 2011 -0400

    added sys/swap.h

diff --git a/libc/include/sys/swap.h b/libc/include/sys/swap.h
new file mode 100644
index 0000000..97065c9
--- /dev/null
+++ b/libc/include/sys/swap.h
@@ -0,0 +1,7 @@
+#ifndef _SYS_SWAP_H_
+#define _SYS_SWAP_H_
+
+int swapon(const char *path, int swapflags);
+int swapoff(const char *path);
+
+#endif /* _SYS_SWAP_H_ */
