commit 12f786d8c6a55d489d78ec9077735e32ef763612
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Sun May 1 16:47:24 2011 -0400

    included sys/linux-syscalls.h in sys/syscall.h

diff --git a/libc/include/sys/syscall.h b/libc/include/sys/syscall.h
index 7055518..b7da416 100644
--- a/libc/include/sys/syscall.h
+++ b/libc/include/sys/syscall.h
@@ -33,6 +33,8 @@
 #include <sys/types.h>
 #include <asm/unistd.h>
 
+#include <sys/linux-syscalls.h>
+
 __BEGIN_DECLS
 
 int syscall(int number, ...);
