commit 22cd46ef4166e11125c2b347158323772fc78e73
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Thu Apr 28 00:28:43 2011 -0400

    copied safe macro decls for S_TYPEISMQ,etc

diff --git a/libc/include/sys/stat.h b/libc/include/sys/stat.h
index 091ee6d..2e5488a 100644
--- a/libc/include/sys/stat.h
+++ b/libc/include/sys/stat.h
@@ -37,6 +37,12 @@
 
 __BEGIN_DECLS
 
+/* POSIX.1b objects.  Note that these macros always evaluate to zero.  But
+   they do it by enforcing the correct use of the macros.  */
+#define S_TYPEISMQ(buf)  ((buf)->st_mode - (buf)->st_mode)
+#define S_TYPEISSEM(buf) ((buf)->st_mode - (buf)->st_mode)
+#define S_TYPEISSHM(buf) ((buf)->st_mode - (buf)->st_mode)
+
 /* really matches stat64 in the kernel, hence the padding
  * Note: The kernel zero's the padded region because glibc might read them
  * in the hope that the kernel has stretched to using larger sizes.

