commit 50631af7968ead88d454d0918fddddf6c45828f4
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Sun May 8 21:27:44 2011 -0400

    add /usr/lib to default so search path

diff --git a/linker/linker.c b/linker/linker.c
index 3a678f8..292fa27 100644
--- a/linker/linker.c
+++ b/linker/linker.c
@@ -606,8 +606,8 @@ static void dump(soinfo *si)
 #endif
 
 static const char *sopaths[] = {
-    "/vendor/lib",
     "/lib",
+    "/usr/lib",
     0
 };
 
