commit 6b4b34de34a9e4eed25880a57b6fe4a0f5585fe3
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Thu May 5 15:31:07 2011 -0400

    included linux/icmp.h in netinet/ip_icmp.h

diff --git a/libc/include/netinet/ip_icmp.h b/libc/include/netinet/ip_icmp.h
index 7510592..dbf90e7 100644
--- a/libc/include/netinet/ip_icmp.h
+++ b/libc/include/netinet/ip_icmp.h
@@ -38,6 +38,8 @@
 #include <netinet/ip.h>
 #include <sys/cdefs.h>
 
+#include <linux/icmp.h>
+
 __BEGIN_DECLS
 
 /*
