diff --git a/libc/Android.mk b/libc/Android.mk
index 9e6bdfb..b65e0a9 100644
--- a/libc/Android.mk
+++ b/libc/Android.mk
@@ -619,6 +619,8 @@ LOCAL_MODULE := libc_nomalloc
 LOCAL_WHOLE_STATIC_LIBRARIES := libc_common
 LOCAL_SYSTEM_SHARED_LIBRARIES :=
 
+LOCAL_LDLIBS := -lgcc
+
 include $(BUILD_STATIC_LIBRARY)
 
 
@@ -672,6 +674,7 @@ LOCAL_MODULE:= libc
 LOCAL_SHARED_LIBRARIES := libdl
 LOCAL_WHOLE_STATIC_LIBRARIES := libc_common
 LOCAL_SYSTEM_SHARED_LIBRARIES :=
+LOCAL_LDFLAGS := -lgcc
 
 include $(BUILD_SHARED_LIBRARY)
 
diff --git a/linker/Android.mk b/linker/Android.mk
index 27a6677..e2b7abb 100644
--- a/linker/Android.mk
+++ b/linker/Android.mk
@@ -62,6 +62,8 @@ LOCAL_MODULE:= linker
 
 LOCAL_STATIC_LIBRARIES := libc_nomalloc
 
+LOCAL_TAIL_MATTER := -lgcc
+
 #LOCAL_FORCE_STATIC_EXECUTABLE := true # not necessary when not including BUILD_EXECUTABLE
 
 #
