commit 461dde521ed4ed4df586bac1fce112f1195478ee
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Fri Apr 29 05:18:50 2011 -0400

    add struct arp to netinet/if_ether.h

diff --git a/libc/include/netinet/if_ether.h b/libc/include/netinet/if_ether.h
index 700b9db..8fad8f6 100644
--- a/libc/include/netinet/if_ether.h
+++ b/libc/include/netinet/if_ether.h
@@ -31,3 +31,25 @@
 #define ETHER_ADDR_LEN ETH_ALEN
 #include <net/ethertypes.h>
 #endif
+
+#include <sys/types.h>
+
+/*
+ * Ethernet Address Resolution Protocol.
+ *
+ * See RFC 826 for protocol description.  Structure below is adapted
+ * to resolving internet addresses.  Field names used correspond to
+ * RFC 826.
+ */
+struct  ether_arp {
+        struct  arphdr ea_hdr;  /* fixed-size header */
+        u_char  arp_sha[ETHER_ADDR_LEN];        /* sender hardware address */
+        u_char  arp_spa[4];     /* sender protocol address */
+        u_char  arp_tha[ETHER_ADDR_LEN];        /* target hardware address */
+        u_char  arp_tpa[4];     /* target protocol address */
+};
+#define arp_hrd ea_hdr.ar_hrd
+#define arp_pro ea_hdr.ar_pro
+#define arp_hln ea_hdr.ar_hln
+#define arp_pln ea_hdr.ar_pln
+#define arp_op  ea_hdr.ar_op
