commit e382e6806e7ec68ebce66cb3e9334682d80b34db
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Mon May 9 06:02:51 2011 -0400

    added stub for getpwent

diff --git a/libc/bionic/stubs.c b/libc/bionic/stubs.c
index acf724c..c3ed1b1 100644
--- a/libc/bionic/stubs.c
+++ b/libc/bionic/stubs.c
@@ -251,7 +251,6 @@ app_id_to_group(gid_t  gid, stubs_state_t*  state)
     return gr;
 }
 
-
 struct passwd*
 getpwuid(uid_t uid)
 {
@@ -314,6 +313,13 @@ int getlogin_r(char *buf, size_t bufsize)
     return -1;
 }
 
+struct passwd *
+getpwent(void) {
+    fprintf(stderr, "FIX ME! implement %s() %s:%d\n", __FUNCTION__, __FILE__, __LINE__);
+    errno = ENOSYS;
+    return NULL;
+}
+
 struct group*
 getgrgid(gid_t gid)
 {
