commit a61c0e0a897f04c72deefa88b6b0cca4c4a915a8
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Wed Apr 27 22:51:34 2011 -0400

    added strchrnul (used extensively by busybox)

diff --git a/libc/Android.mk b/libc/Android.mk
index b65e0a9..395b04d 100644
--- a/libc/Android.mk
+++ b/libc/Android.mk
@@ -186,6 +186,7 @@ libc_common_src_files := \
 	string/strcasestr.c \
 	string/strcat.c \
 	string/strchr.c \
+	string/strchrnul.c \
 	string/strcoll.c \
 	string/strcpy.c \
 	string/strcspn.c \
diff --git a/libc/include/string.h b/libc/include/string.h
index 613dcd7..011f41f 100644
--- a/libc/include/string.h
+++ b/libc/include/string.h
@@ -47,6 +47,7 @@ extern void   memswap(void *, void *, size_t);
 extern char*  index(const char *, int);
 extern char*  rindex(const char *, int);
 extern char*  strchr(const char *, int);
+extern char*  strchrnul(const char *, int);
 extern char*  strrchr(const char *, int);
 
 extern size_t strlen(const char *);
diff --git a/libc/string/strchrnul.c b/libc/string/strchrnul.c
new file mode 100644
index 0000000..8ca34ca
--- /dev/null
+++ b/libc/string/strchrnul.c
@@ -0,0 +1,43 @@
+/*	$OpenBSD: index.c,v 1.5 2005/08/08 08:05:37 espie Exp $ */
+/*-
+ * Copyright (c) 1990 The Regents of the University of California.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ * 3. Neither the name of the University nor the names of its contributors
+ *    may be used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE REGENTS AND CONTRIBUTORS ``AS IS'' AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED.  IN NO EVENT SHALL THE REGENTS OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+ * OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ */
+
+#include <string.h>
+
+char *
+strchrnul(const char *p, int ch)
+{
+	for (;; ++p) {
+		if (*p == ch)
+			return((char *)p);
+		if (!*p)
+			return(p);
+	}
+	/* NOTREACHED */
+}
