commit 270acc22a33f8410e200f50251fa0f07682979a7
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Fri Apr 29 02:54:49 2011 -0400

    made sethostname visible in unistd.h

diff --git a/libc/include/unistd.h b/libc/include/unistd.h
index c469f35..83df392 100644
--- a/libc/include/unistd.h
+++ b/libc/include/unistd.h
@@ -193,11 +193,12 @@ extern int execvpe(const char *, char * const *, char * const *);
 extern int execlpe(const char *, const char *, ...);
 extern int getfsuid(uid_t);
 extern int setfsuid(uid_t);
-extern int sethostname(const char *, size_t);
 extern int getdomainname(char *, size_t);
 extern int setdomainname(const char *, size_t);
 #endif /* MISSING */
 
+extern int sethostname(const char *, size_t);
+
 // FIXME: stubs
 extern pid_t  getsid(pid_t);
 extern int getlogin_r(char* name, size_t namesize);
