commit a8b7ad452752e04c2e5d3a834fce0763c1108286
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Thu Jan 12 10:34:44 2012 -0500

    force the raise to NULL to satisfy libgcc.a

diff --git a/linker/Android.mk b/linker/Android.mk
index b7325b6..fc53996 100644
--- a/linker/Android.mk
+++ b/linker/Android.mk
@@ -27,6 +27,7 @@ endif
 LINKER_AREA_SIZE := 0x01000000
 
 LOCAL_LDFLAGS := -Wl,-Ttext,$(LINKER_TEXT_BASE)
+LOCAL_LDFLAGS += -Wl,--defsym,raise=0
 
 LOCAL_CFLAGS += -DPRELINK
 LOCAL_CFLAGS += -DLINKER_TEXT_BASE=$(LINKER_TEXT_BASE)
