commit d0b63c6549e115f8cf505c6e459f00948c9d9f4b
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Tue May 10 03:26:04 2011 -0400

    size_t should be unsigned long and not unsigned int

diff --git a/libc/arch-x86/include/machine/_types.h b/libc/arch-x86/include/machine/_types.h
index be4f6e4..67d6405 100644
--- a/libc/arch-x86/include/machine/_types.h
+++ b/libc/arch-x86/include/machine/_types.h
@@ -38,11 +38,7 @@
 /* the kernel defines size_t as unsigned int, but g++ wants it to be unsigned long */
 #ifndef _SIZE_T
 #  define _SIZE_T
-#  ifdef ANDROID
-     typedef unsigned int  size_t;
-#  else
-     typedef unsigned long  size_t;
-#  endif
+typedef unsigned long  size_t;
 #endif
 #if !defined(_SSIZE_T) && !defined(_SSIZE_T_DEFINED_)
 #define _SSIZE_T
diff --git a/libc/include/sys/types.h b/libc/include/sys/types.h
index a6caf0d..c4364c6 100644
--- a/libc/include/sys/types.h
+++ b/libc/include/sys/types.h
@@ -84,7 +84,7 @@ typedef  .... pthread_t;
 
 #ifndef _SIZE_T_DEFINED_
 #define _SIZE_T_DEFINED_
-typedef unsigned int  size_t;
+typedef unsigned long  size_t;
 #endif
 
 /* size_t is defined by the GCC-specific <stddef.h> */
