commit c416670595cb7ab839e4ed4352477c4bb76b7f38
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Thu Apr 28 07:09:53 2011 -0400

    cleaned-up new syscall declarations

diff --git a/libc/include/sys/swap.h b/libc/include/sys/swap.h
index 97065c9..05897b0 100644
--- a/libc/include/sys/swap.h
+++ b/libc/include/sys/swap.h
@@ -1,7 +1,11 @@
 #ifndef _SYS_SWAP_H_
 #define _SYS_SWAP_H_
 
-int swapon(const char *path, int swapflags);
-int swapoff(const char *path);
+__BEGIN_DECLS
+
+extern int swapon(const char *path, int swapflags);
+extern int swapoff(const char *path);
+
+__END_DECLS
 
 #endif /* _SYS_SWAP_H_ */
diff --git a/libc/include/sys/timex.h b/libc/include/sys/timex.h
index 3426b34..a4e77d2 100644
--- a/libc/include/sys/timex.h
+++ b/libc/include/sys/timex.h
@@ -3,6 +3,10 @@
 
 #include <linux/timex.h>
 
-int adjtimex(struct timex *buf);
+__BEGIN_DECLS
+
+extern int adjtimex(struct timex *buf);
+
+__END_DECLS
 
 #endif /* _SYS_TIMEX_H_ */
diff --git a/libc/include/time.h b/libc/include/time.h
index 5d1a0e2..f8377d7 100644
--- a/libc/include/time.h
+++ b/libc/include/time.h
@@ -128,6 +128,8 @@ extern int  timer_settime(timer_t timerid, int flags, const struct itimerspec *v
 extern int  timer_gettime(timer_t timerid, struct itimerspec *value);
 extern int  timer_getoverrun(timer_t  timerid);
 
+extern int stime(time_t *t);
+
 __END_DECLS
 
 #endif /* _TIME_H_ */
