commit 065481f7a24fe8ce5a44f19b11e647b71633c395
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Thu Apr 28 07:58:15 2011 -0400

    added pututline, and getutid,getutline as stubs

diff --git a/libc/bionic/stubs.c b/libc/bionic/stubs.c
index 0c5ac40..bb9a697 100644
--- a/libc/bionic/stubs.c
+++ b/libc/bionic/stubs.c
@@ -458,3 +458,17 @@ char *hasmntopt(const struct mntent *mnt, const char *opt)
     errno = ENOSYS;
     return NULL;
 }
+
+struct utmp *getutid(struct utmp *ut)
+{
+    fprintf(stderr, "FIX ME! implement %s() %s:%d\n", __FUNCTION__, __FILE__, __LINE__);
+    errno = ENOSYS;
+    return NULL;
+}
+struct utmp *getutline(struct utmp *ut)
+{
+    fprintf(stderr, "FIX ME! implement %s() %s:%d\n", __FUNCTION__, __FILE__, __LINE__);
+    errno = ENOSYS;
+    return NULL;
+}
+
diff --git a/libc/include/utmp.h b/libc/include/utmp.h
index 0f36642..cf57a0b 100644
--- a/libc/include/utmp.h
+++ b/libc/include/utmp.h
@@ -93,6 +93,9 @@ __BEGIN_DECLS
 int utmpname(const char*);
 void setutent();
 struct utmp* getutent();
+void pututline(struct utmp* utmp);
+struct utmp *getutid(struct utmp *ut);
+struct utmp *getutline(struct utmp *ut);
 
 void setutent(void);
 void endutent(void);
