commit ddebd796e5a7d4f6592d1bfb7312abc8c4f55f06
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Thu Apr 28 12:50:49 2011 -0400

    added readahead

diff --git a/libc/SYSCALLS.TXT b/libc/SYSCALLS.TXT
index 16b1deb..e6749f2 100644
--- a/libc/SYSCALLS.TXT
+++ b/libc/SYSCALLS.TXT
@@ -130,6 +130,7 @@ int         mkdirat(int dirfd, const char *pathname, mode_t mode)  323,296
 int         fchownat(int dirfd, const char *path, uid_t owner, gid_t group, int flags)  325,298
 int         fchmodat(int dirfd, const char *path, mode_t mode, int flags)  333,306
 int         renameat(int olddirfd, const char *oldpath, int newdirfd, const char *newpath)  329,302
+ssize_t     readahead(int, off64_t, size_t) 225
 
 # file system
 int     link (const char*, const char*)  9
diff --git a/libc/include/fcntl.h b/libc/include/fcntl.h
index 7219dd7..92fc45c 100644
--- a/libc/include/fcntl.h
+++ b/libc/include/fcntl.h
@@ -48,6 +48,7 @@ extern int  openat(int fd, const char*  path, int  mode, ...);
 extern int  unlinkat(int dirfd, const char *pathname, int flags);
 extern int  fcntl(int   fd, int   command, ...);
 extern int  creat(const char*  path, mode_t  mode);
+extern ssize_t readahead(int fd, off64_t offset, size_t count);
 
 __END_DECLS
 
diff --git a/libc/include/sys/mtio.h b/libc/include/sys/mtio.h
new file mode 100644
index 0000000..8fb5655
--- /dev/null
+++ b/libc/include/sys/mtio.h
@@ -0,0 +1 @@
+#include <linux/mtio.h>
