commit 3fb9c5fded6ac61bc12ffc9c6777ba3021e70b01
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Wed Jan 11 14:19:25 2012 -0500

    remove dependency on soslim

diff --git a/core/definitions.mk b/core/definitions.mk
index 11cea3f..215ec99 100644
--- a/core/definitions.mk
+++ b/core/definitions.mk
@@ -1092,7 +1092,7 @@ endef
 define transform-to-stripped
 @mkdir -p $(dir $@)
 @echo "target Strip: $(PRIVATE_MODULE) ($@)"
-$(hide) $(SOSLIM) --strip --shady --quiet $< --outfile $@
+$(hide) cp $< $@
 endef
 
 define transform-to-prelinked
diff --git a/core/dynamic_binary.mk b/core/dynamic_binary.mk
index cfbe740..e6f688f 100644
--- a/core/dynamic_binary.mk
+++ b/core/dynamic_binary.mk
@@ -137,7 +137,7 @@ endif
 
 ifeq ($(LOCAL_STRIP_MODULE),true)
 # Strip the binary
-$(strip_output): $(strip_input) | $(SOSLIM)
+$(strip_output): $(strip_input)
 	$(transform-to-stripped)
 else
 # Don't strip the binary, just copy it.  We can't skip this step
