commit 4257879752f328d04ad3a01a302fa8c73b0ad38c
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Thu Jan 12 10:36:34 2012 -0500

    added libgcc_eh.a to resolve __aeabi_unwind_cppp_pr{0,1,2}

diff --git a/core/combo/TARGET_linux-arm.mk b/core/combo/TARGET_linux-arm.mk
index ae1997c..114fb35 100644
--- a/core/combo/TARGET_linux-arm.mk
+++ b/core/combo/TARGET_linux-arm.mk
@@ -147,12 +147,15 @@ libthread_db_root := bionic/libthread_db
 
 
 ## on some hosts, the target cross-compiler is not available so do not run this command
-ifneq ($(wildcard $(TARGET_CC)),)
 # We compile with the global cflags to ensure that
 # any flags which affect libgcc are correctly taken
 # into account.
-TARGET_LIBGCC := $(shell $(TARGET_CC) $(TARGET_GLOBAL_CFLAGS) -print-libgcc-file-name)
-endif
+TARGET_LIBGCC := \
+	$(shell $(TARGET_CC) $(TARGET_GLOBAL_CFLAGS) -print-libgcc-file-name)
+BLAH=$(shell $(TARGET_CC) $(TARGET_GLOBAL_CFLAGS) -print-file-name=libgcc_eh.a)
+ifneq ($(BLAH),libgcc_eh.a)
+TARGET_LIBGCC += $(BLAH)
+endif
 
 # Define FDO (Feedback Directed Optimization) options.
 
