commit 3f2707e3cb56589ff5c7178063d6adb8ff2a6b98
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Fri Apr 29 16:42:10 2011 -0400

    removed librpc/Android.mk

diff --git a/librpc/Android.mk b/librpc/Android.mk
deleted file mode 100644
index c04d955..0000000
--- a/librpc/Android.mk
+++ /dev/null
@@ -1,34 +0,0 @@
-LOCAL_PATH:= $(call my-dir)
-
-include $(CLEAR_VARS)
-
-LOCAL_SRC_FILES:= xdr.c rpc.c svc.c clnt.c ops.c svc_clnt_common.c
-
-LOCAL_C_INCLUDES:=$(LOCAL_PATH)
-
-LOCAL_CFLAGS:= -fno-short-enums
-
-LOCAL_CFLAGS+=-DRPC_OFFSET=0
-#LOCAL_CFLAGS+=-DDEBUG -DVERBOSE
-
-LOCAL_COPY_HEADERS_TO:= librpc/rpc
-LOCAL_COPY_HEADERS:= \
-	rpc/clnt.h \
-	rpc/pmap_clnt.h \
-	rpc/rpc.h \
-	rpc/rpc_router_ioctl.h \
-	rpc/svc.h \
-	rpc/types.h \
-	rpc/xdr.h
-
-LOCAL_MODULE:= librpc
-
-include $(BUILD_STATIC_LIBRARY)
-
-include $(CLEAR_VARS)
-LOCAL_MODULE := librpc
-# LOCAL_SHARED_LIBRARIES := liblog
-# LOCAL_STATIC_LIBRARIES := libpower
-LOCAL_WHOLE_STATIC_LIBRARIES := librpc
-# LOCAL_PRELINK_MODULE := false
-include $(BUILD_SHARED_LIBRARY)
