commit b74c3969ec1a5297363b3bccb96c57d9659e15f3
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Fri Apr 29 16:04:16 2011 -0400

    pruned all but librpc

diff --git a/Android.mk b/Android.mk
index 9b59dee..2f34efc 100644
--- a/Android.mk
+++ b/Android.mk
@@ -14,19 +14,4 @@
 # limitations under the License.
 #
 
-common_msm_dirs := libcopybit liblights libopencorehw librpc libstagefrighthw
-msm7k_dirs := $(common_msm_dirs) boot libgralloc libaudio
-qsd8k_dirs := $(common_msm_dirs) libgralloc-qsd8k libaudio-qsd8k dspcrashd
-msm7x30_dirs := liblights libgralloc-qsd8k librpc libaudio-qdsp5v2
-
-ifeq ($(TARGET_BOARD_PLATFORM),msm7k)
-  include $(call all-named-subdir-makefiles,$(msm7k_dirs))
-else
-  ifeq ($(TARGET_BOARD_PLATFORM),qsd8k)
-    include $(call all-named-subdir-makefiles,$(qsd8k_dirs))
-  else
-    ifeq ($(TARGET_BOARD_PLATFORM),msm7x30)
-      include $(call all-named-subdir-makefiles,$(msm7x30_dirs))
-    endif
-  endif
-endif
+include $(call all-named-subdir-makefiles,librpc)
