commit 35b16e63565dc7dbdcde28fed6a2f035500e3227
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Sat Apr 30 11:52:23 2011 -0400

    added struct client

diff --git a/librpc/rpc/clnt.h b/librpc/rpc/clnt.h
index fb7da91..d9be441 100644
--- a/librpc/rpc/clnt.h
+++ b/librpc/rpc/clnt.h
@@ -99,6 +99,24 @@ enum clnt_stat {
 
 struct CLIENT;
 typedef struct CLIENT CLIENT;
+struct CLIENT {
+  AUTH	*cl_auth;		 /* authenticator */
+  struct clnt_ops {
+    enum clnt_stat (*cl_call) (CLIENT *, u_long, xdrproc_t, caddr_t, xdrproc_t,
+			       caddr_t, struct timeval);
+			       	/* call remote procedure */
+    void (*cl_abort) (void);	/* abort a call */
+    void (*cl_geterr) (CLIENT *, struct rpc_err *);
+				/* get specific error code */
+    bool_t (*cl_freeres) (CLIENT *, xdrproc_t, caddr_t);
+				/* frees results */
+    void (*cl_destroy) (CLIENT *); /* destroy this structure */
+    bool_t (*cl_control) (CLIENT *, int, char *);
+				/* the ioctl() of rpc */
+  } *cl_ops;
+  caddr_t cl_private;		/* private stuff */
+};
+
 /* client call callback. 
  * Callback called when the reply is recieved or there is an error in
  * getting reply.
