commit d3aad10d50da793b0d6a1a04c848f32348d7a031
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Fri Apr 29 17:02:15 2011 -0400

    corrected <debug.h> -> "debug.h"

diff --git a/librpc/clnt.c b/librpc/clnt.c
index 8452b9f..aeca005 100644
--- a/librpc/clnt.c
+++ b/librpc/clnt.c
@@ -1,7 +1,7 @@
 #include <rpc/rpc.h>
 #include <arpa/inet.h>
 #include <rpc/rpc_router_ioctl.h>
-#include <debug.h>
+#include "debug.h"
 #include <pthread.h>
 #include <sys/select.h>
 
diff --git a/librpc/ops.c b/librpc/ops.c
index 66e1131..35ea977 100644
--- a/librpc/ops.c
+++ b/librpc/ops.c
@@ -1,6 +1,6 @@
 #include <rpc/rpc.h>
 #include <rpc/rpc_router_ioctl.h>
-#include <debug.h>
+#include "debug.h"
 
 #include <sys/types.h>   
 #include <sys/stat.h>     
diff --git a/librpc/svc.c b/librpc/svc.c
index aa70be6..634b0cd 100644
--- a/librpc/svc.c
+++ b/librpc/svc.c
@@ -45,7 +45,7 @@
 #include <unistd.h>
 #include <arpa/inet.h>
 #include <rpc/rpc_router_ioctl.h>
-#include <debug.h>
+#include "debug.h"
 #include <pthread.h>
 #include <stdlib.h>
 
diff --git a/librpc/svc_clnt_common.c b/librpc/svc_clnt_common.c
index 808cc6b..20eec3e 100644
--- a/librpc/svc_clnt_common.c
+++ b/librpc/svc_clnt_common.c
@@ -1,7 +1,7 @@
 #include <rpc/rpc.h>
 #include <arpa/inet.h>
 #include <errno.h>
-#include <debug.h>
+#include "debug.h"
 
 extern int r_open(const char *router);
 extern void r_close(int handle);
