commit 5f56e3519eedba308714913291f3911199a99db4
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Sat Apr 30 10:42:33 2011 -0400

    added types & prototypes to rpc/pmap_clnt.h

diff --git a/librpc/rpc/pmap_clnt.h b/librpc/rpc/pmap_clnt.h
index 7233dcb..d828368 100644
--- a/librpc/rpc/pmap_clnt.h
+++ b/librpc/rpc/pmap_clnt.h
@@ -46,6 +46,30 @@ extern "C" {
 #include <rpc/clnt.h>
 #include <rpc/svc.h>
 
+typedef bool_t (*resultproc_t) (caddr_t __resp, struct sockaddr_in *__raddr);
+
+extern bool_t pmap_set (__const u_long __program, __const u_long __vers,
+                        int __protocol, u_short __port);
+extern bool_t pmap_unset (__const u_long __program, __const u_long __vers);
+extern struct pmaplist *pmap_getmaps (struct sockaddr_in *__address);
+extern enum clnt_stat pmap_rmtcall (struct sockaddr_in *__addr,
+                                    __const u_long __prog,
+                                    __const u_long __vers,
+                                    __const u_long __proc,
+                                    xdrproc_t __xdrargs,
+                                    caddr_t __argsp, xdrproc_t __xdrres,
+                                    caddr_t __resp, struct timeval __tout,
+                                    u_long *__port_ptr);
+extern enum clnt_stat clnt_broadcast (__const u_long __prog,
+                                      __const u_long __vers,
+                                      __const u_long __proc, xdrproc_t __xargs,
+                                      caddr_t __argsp, xdrproc_t __xresults,
+                                      caddr_t __resultsp,
+                                      resultproc_t __eachresult);
+extern u_short pmap_getport (struct sockaddr_in *__address,
+                             __const u_long __program,
+                             __const u_long __version, u_int __protocol);
+
 #ifdef __cplusplus
 }
 #endif
