diff -uNrbB separate.gimp20.bak/Makefile separate.gimp20/Makefile
--- separate.gimp20.bak/Makefile	2004-01-21 01:11:41.000000000 +0800
+++ separate.gimp20/Makefile	2004-05-25 23:13:10.000000000 +0800
@@ -3,18 +3,21 @@
 
 GTK_INCLUDE = `pkg-config gtk+-2.0 --cflags`
 GTK_LIB = `pkg-config gtk+-2.0 --libs`
-GIMP_INCLUDE = `pkg-config gimp-1.3 --cflags` `pkg-config gimpui-1.3 --cflags`
-GIMP_LIB = `pkg-config gimp-1.3 --libs` `pkg-config gimpui-1.3 --libs`
+GIMP_INCLUDE = `pkg-config gimp-2.0 --cflags` `pkg-config gimpui-2.0 --cflags`
+GIMP_LIB = `pkg-config gimp-2.0 --libs` `pkg-config gimpui-2.0 --libs`
 LCMS_LIB = -llcms
 TIFF_LIB = -ltiff
 
+INCLUDES = $(GIMP_INCLUDE) $(GTK_INCLUDE)
+LIBRARIES = $(GIMP_LIB) $(GTK_LIB) $(LCMS_LIB) $(TIFF_LIB)
+
 VERSION = 0.1
 CC = gcc -g
 DEPEND = gccmakedep
 RM = rm -f
 
-CFLAGS = $(GIMP_INCLUDE) $(GTK_INCLUDE) -O2 -Wall
-LDFLAGS = $(GIMP_LIB) $(GTK_LIB) $(LCMS_LIB) $(TIFF_LIB)
+CFLAGS = -O2 -Wall
+LDFLAGS = 
 
 SOURCES = \
 	separate.c util.c tiff.c
@@ -26,13 +29,13 @@
 all:	 $(TARGETS)
 
 depend:
-	$(DEPEND) -- $(CFLAGS) -- $(SOURCES)
+	$(DEPEND) -- $(CFLAGS) $(INCLUDES) -- $(SOURCES)
 
 clean:
 	$(RM) $(OBJECTS) $(TARGETS) core *~
 
 separate: $(OBJECTS)
-	$(CC) separate.o util.o tiff.o -o $@ $(LDFLAGS)
+	$(CC) separate.o util.o tiff.o -o $@ $(LDFLAGS) $(LIBRARIES)
 
 install: separate
 	gimptool-2.0 --install-bin separate
@@ -43,5 +46,4 @@
 # Inference rules
 
 .c.o:
-	$(CC) $(CFLAGS) -c $<
-
+	$(CC) $(CFLAGS) $(INCLUDES) -c $<
diff -uNrbB separate.gimp20.bak/platform_unix.h separate.gimp20/platform_unix.h
--- separate.gimp20.bak/platform_unix.h	2004-01-24 21:49:11.000000000 +0800
+++ separate.gimp20/platform_unix.h	2004-05-25 23:13:57.000000000 +0800
@@ -1,8 +1,8 @@
 #ifndef SEPARATE_PLATFORM_H
 #define SEPARATE_PLATFORM_H
 
-#define DEFAULT_RGB_PROFILE "/usr/local/share/iccprofiles/sRGB Color Space Profile.icm"
-#define DEFAULT_CMYK_PROFILE "/usr/local/share/iccprofiles/EuroscaleCoated.icc"
+#define DEFAULT_RGB_PROFILE "/usr/share/lcms/profiles/sRGB Color Space Profile.icm"
+#define DEFAULT_CMYK_PROFILE "/usr/share/iccprofiles/Adobe ICC Profiles/CMYK Profiles/EuroscaleCoated.icc"
 
 #define DEBUG(x) fprintf(stdout,x); fflush(stdout)
 
diff -uNrbB separate.gimp20.bak/separate.c separate.gimp20/separate.c
--- separate.gimp20.bak/separate.c	2004-01-21 03:13:23.000000000 +0800
+++ separate.gimp20/separate.c	2004-05-25 23:13:10.000000000 +0800
@@ -27,7 +27,7 @@
 #include <libgimp/gimp.h>
 #include <libgimp/gimpui.h>
 
-#include <lcms.h>
+#include <lcms/lcms.h>
 
 #include "separate.h"
 
diff -uNrbB separate.gimp20.bak/util.c separate.gimp20/util.c
--- separate.gimp20.bak/util.c	2004-01-21 03:28:19.000000000 +0800
+++ separate.gimp20/util.c	2004-05-25 23:13:10.000000000 +0800
@@ -6,7 +6,7 @@
 #include <gtk/gtk.h>
 
 #include <libgimp/gimp.h>
-#include <lcms.h>
+#include <lcms/lcms.h>
 
 #include "platform.h"
 #include "util.h"
