--- clips-6.21/clipssrc/makefile.x.lib64	2003-09-26 20:03:22.000000000 +0200
+++ clips-6.21/clipssrc/makefile.x	2003-09-26 20:06:51.000000000 +0200
@@ -4,7 +4,8 @@
 # that xclips color are made.
 # -----------------------------------------------------
 
-WHERE_XLIBS_ARE =       /usr/X11R6/lib
+LIB =			lib
+WHERE_XLIBS_ARE =       /usr/X11R6/$(LIB)
 WHERE_INCL_FILES_ARE =  /usr/X11R6/include
 
 #--------------------------------------------------------------------
