diff -Nurp a/src/phobos/std/path.d b/src/phobos/std/path.d
--- a/src/phobos/std/path.d	2012-04-17 12:02:10.980873765 +0200
+++ b/src/phobos/std/path.d	2012-04-17 13:43:17.035037653 +0200
@@ -59,6 +59,7 @@ version(Posix)
     import core.stdc.errno;
     import core.sys.posix.pwd;
     import core.sys.posix.stdlib;
+    private import core.exception : onOutOfMemoryError;
 }
 
 
@@ -2795,7 +2796,6 @@ deprecated:
 version(Posix)
 {
     private import core.sys.posix.pwd;
-    private import core.exception : onOutOfMemoryError;
 }
 
 version(Windows)
