--- dmd2/src/dmd/posix.mak.orig	2011-09-08 05:10:14.000000000 +0200
+++ dmd2/src/dmd/posix.mak	2011-09-14 15:29:04.635180684 +0200
@@ -29,7 +29,7 @@ C=backend
 TK=tk
 ROOT=root
 
-MODEL=32
+MODEL?=32
 
 ifeq (OSX,$(TARGET))
     ## See: http://developer.apple.com/documentation/developertools/conceptual/cross_development/Using/chapter_3_section_2.html#//apple_ref/doc/uid/20002000-1114311-BABGCAAB
--- dmd2/src/druntime/posix.mak.orig	2011-09-01 21:22:06.000000000 +0200
+++ dmd2/src/druntime/posix.mak	2011-09-14 15:29:04.636180672 +0200
@@ -20,21 +20,21 @@ ifeq (,$(OS))
     endif
 endif
 
-DMD=dmd
+DMD?=dmd
 
 DOCDIR=doc
 IMPDIR=import
 
-MODEL=32
+MODEL?=32
 
 DFLAGS=-m$(MODEL) -O -release -inline -nofloat -w -d -Isrc -Iimport -property
 UDFLAGS=-m$(MODEL) -O -release -nofloat -w -d -Isrc -Iimport -property
 
 CFLAGS=-m$(MODEL) -O
 
-OBJDIR=obj
+OBJDIR=obj$(MODEL)
 DRUNTIME_BASE=druntime
-DRUNTIME=lib/lib$(DRUNTIME_BASE).a
+DRUNTIME=lib$(MODEL)/lib$(DRUNTIME_BASE).a
 
 DOCFMT=
 
--- dmd2/src/phobos/etc/c/zlib/gzio.c.orig	2010-09-16 22:27:48.000000000 +0200
+++ dmd2/src/phobos/etc/c/zlib/gzio.c	2011-09-14 15:29:29.412870920 +0200
@@ -8,6 +8,7 @@
 /* @(#) $Id$ */
 
 #include <stdio.h>
+#include <errno.h>
 
 #include "zutil.h"
 
--- dmd2/src/phobos/posix.mak.orig	2011-09-08 05:10:16.000000000 +0200
+++ dmd2/src/phobos/posix.mak	2011-09-14 15:29:04.638180648 +0200
@@ -48,7 +48,7 @@ endif
 DRUNTIME_PATH = ../druntime
 ZIPFILE = phobos.zip
 ROOT_OF_THEM_ALL = generated
-ROOT = $(ROOT_OF_THEM_ALL)/$(OS)/$(BUILD)/$(MODEL)
+ROOT = $(ROOT_OF_THEM_ALL)/$(OS)/$(BUILD)$(MODEL)
 # Documentation-related stuff
 DOCSRC = ../d-programming-language.org
 WEBSITE_DIR = ../web
@@ -61,8 +61,6 @@ DDOCFLAGS=-m$(MODEL) -d -c -o- -version=
 
 # Variable defined in an OS-dependent manner (see below)
 CC =
-DMD =
-DDOC =
 CFLAGS =
 DFLAGS =
 
@@ -75,7 +73,7 @@ MAKEFILE:=$(lastword $(MAKEFILE_LIST))
 
 # Set DRUNTIME name and full path
 ifeq (,$(findstring win,$(OS)))
-	DRUNTIME = $(DRUNTIME_PATH)/lib/libdruntime.a
+	DRUNTIME = $(DRUNTIME_PATH)/lib$(MODEL)/libdruntime.a
 else
 	DRUNTIME = $(DRUNTIME_PATH)/lib/druntime.lib
 endif
@@ -83,14 +81,14 @@ endif
 # Set CC and DMD
 ifeq ($(OS),win32wine)
 	CC = wine dmc.exe
-	DMD = wine dmd.exe
+	DMD ?= wine dmd.exe
 	RUN = wine
 else
 	ifeq ($(OS),win32remote)
-		DMD = ssh 206.125.170.138 "cd code/dmd/phobos && dmd"
+		DMD ?= ssh 206.125.170.138 "cd code/dmd/phobos && dmd"
 		CC = ssh 206.125.170.138 "cd code/dmd/phobos && dmc"
 	else
-		DMD = dmd
+		DMD ?= dmd
 		ifeq ($(OS),win32)
 			CC = dmc
 		else
@@ -141,7 +139,7 @@ else
 endif
 
 # Set DDOC, the documentation generator
-DDOC=dmd
+DDOC?=$(DMD)
 
 # Set LIB, the ultimate target
 ifeq (,$(findstring win,$(OS)))
