--- dmd2/src/phobos/etc/c/zlib/gzio.c.orig	2011-02-19 09:44:45.000000000 +0200
+++ dmd2/src/phobos/etc/c/zlib/gzio.c	2011-02-19 09:45:42.456666733 +0200
@@ -8,6 +8,7 @@
 /* @(#) $Id$ */
 
 #include <stdio.h>
+#include <errno.h>
 
 #include "zutil.h"
 
--- dmd2/src/druntime/posix.mak.orig	2011-02-19 10:06:22.000000000 +0200
+++ dmd2/src/druntime/posix.mak	2011-02-19 10:38:16.673333400 +0200
@@ -8,16 +8,16 @@
 DOCDIR=doc
 IMPDIR=import
 
-MODEL=32
+MODEL?=32
 
 DFLAGS=-m$(MODEL) -O -release -inline -nofloat -w -d -Isrc -Iimport 
 UDFLAGS=-m$(MODEL) -O -release -nofloat -w -d -Isrc -Iimport 
 
 CFLAGS=-m$(MODEL) -O
 
-OBJDIR=obj
+OBJDIR=obj$(MODEL)
 DRUNTIME_BASE=druntime
-DRUNTIME=lib/lib$(DRUNTIME_BASE).a
+DRUNTIME=lib$(MODEL)/lib$(DRUNTIME_BASE).a
 
 DOCFMT=
 
--- dmd2/src/dmd/linux.mak.orig	2011-02-20 10:45:13.000000000 +0200
+++ dmd2/src/dmd/linux.mak	2011-02-20 10:57:32.993333337 +0200
@@ -3,9 +3,9 @@
 TK=tk
 ROOT=root
 
-MODEL=-m32
+MODEL?=32
 
-CC=g++ $(MODEL)
+CC=g++ -m$(MODEL)
 
 #OPT=-g -g3
 #OPT=-O2
@@ -94,7 +94,7 @@
 all: dmd
 
 dmd: $(DMD_OBJS)
-	gcc $(MODEL) -lstdc++ -lpthread $(COV) $(DMD_OBJS) -o dmd
+	gcc -m$(MODEL) -lstdc++ -lpthread $(COV) $(DMD_OBJS) -o dmd
 
 clean:
 	rm -f $(DMD_OBJS) dmd optab.o id.o impcnvgen idgen id.c id.h \
@@ -467,7 +467,7 @@
 	$(CC) -c $(GFLAGS) -I$(ROOT) $<
 
 strtold.o: $C/strtold.c
-	gcc $(MODEL) -c $C/strtold.c
+	gcc -m$(MODEL) -c $C/strtold.c
 
 struct.o: struct.c
 	$(CC) -c $(CFLAGS) $<
--- dmd2/src/phobos/posix.mak.orig	2011-02-20 10:45:14.000000000 +0200
+++ dmd2/src/phobos/posix.mak	2011-02-20 10:47:36.923333335 +0200
@@ -48,7 +48,7 @@
 DRUNTIME_PATH = ../druntime
 ZIPFILE = phobos.zip
 ROOT_OF_THEM_ALL = generated
-ROOT = $(ROOT_OF_THEM_ALL)/$(OS)/$(BUILD)/$(MODEL)
+ROOT = $(ROOT_OF_THEM_ALL)/$(OS)/$(BUILD)$(MODEL)
 # Documentation-related stuff
 DOCSRC = ../d-programming-language.org
 WEBSITE_DIR = ../web
@@ -73,7 +73,7 @@
 
 # Set DRUNTIME name and full path
 ifeq (,$(findstring win,$(OS)))
-	DRUNTIME = $(DRUNTIME_PATH)/lib/libdruntime.a
+	DRUNTIME = $(DRUNTIME_PATH)/lib$(MODEL)/libdruntime.a
 else
 	DRUNTIME = $(DRUNTIME_PATH)/lib/druntime.lib
 endif
--- dmd2/src/druntime/posix.mak.orig	2011-02-20 11:35:02.000000000 +0200
+++ dmd2/src/druntime/posix.mak	2011-02-20 11:41:04.153333339 +0200
@@ -3,7 +3,7 @@
 #    pkg_add -r gmake
 # and then run as gmake rather than make.
 
-DMD=dmd
+DMD?=dmd
 
 DOCDIR=doc
 IMPDIR=import
--- dmd2/src/phobos/posix.mak.orig	2011-02-20 11:35:02.000000000 +0200
+++ dmd2/src/phobos/posix.mak	2011-02-20 11:43:31.790000006 +0200
@@ -59,8 +59,6 @@
 
 # Variable defined in an OS-dependent manner (see below)
 CC =
-DMD =
-DDOC =
 CFLAGS =
 DFLAGS =
 
@@ -81,14 +79,14 @@
 # Set CC and DMD
 ifeq ($(OS),win32wine)
 	CC = wine $(HOME)/dmc/bin/dmc.exe
-	DMD = wine $(HOME)/dmd2/windows/bin/dmd.exe
+	DMD ?= wine $(HOME)/dmd2/windows/bin/dmd.exe
 	RUN = wine
 else
 	ifeq ($(OS),win32remote)
-		DMD = ssh 206.125.170.138 "cd code/dmd/phobos && dmd"
+		DMD ?= ssh 206.125.170.138 "cd code/dmd/phobos && dmd"
 		CC = ssh 206.125.170.138 "cd code/dmd/phobos && dmc"
 	else
-		DMD = dmd
+		DMD ?= dmd
 		ifeq ($(OS),win32)
 			CC = dmc
 		else
@@ -139,7 +137,7 @@
 endif
 
 # Set DDOC, the documentation generator
-DDOC=dmd
+DDOC?=$(DMD)
 
 # Set LIB, the ultimate target
 ifeq (,$(findstring win,$(OS)))
