diff -Nur a/cmake/developer_package/packaging/archive.cmake b/cmake/developer_package/packaging/archive.cmake
--- a/cmake/developer_package/packaging/archive.cmake	2024-11-08 22:14:37.000000000 -0800
+++ b/cmake/developer_package/packaging/archive.cmake	2024-12-12 02:31:52.106226700 -0800
@@ -4,19 +4,6 @@
 
 include(GNUInstallDirs)
 
-if(APPLE)
-    # on macOS versions with SIP enabled, we need to use @rpath
-    # because DYLD_LIBRARY_PATH is ignored
-    set(CMAKE_SKIP_INSTALL_RPATH_DEFAULT OFF)
-else()
-    # we don't need RPATHs, because setupvars.sh is used
-    set(CMAKE_SKIP_INSTALL_RPATH_DEFAULT ON)
-endif()
-
-if(NOT DEFINED CMAKE_SKIP_INSTALL_RPATH)
-    set(CMAKE_SKIP_INSTALL_RPATH ${CMAKE_SKIP_INSTALL_RPATH_DEFAULT})
-endif()
-
 #
 # ov_archive_cpack_set_dirs()
 #
@@ -25,19 +12,16 @@
 macro(ov_archive_cpack_set_dirs)
     # common "archive" package locations
     # TODO: move current variables to OpenVINO specific locations
-    set(OV_CPACK_INCLUDEDIR runtime/include)
-    set(OV_CPACK_OPENVINO_CMAKEDIR runtime/cmake)
-    set(OV_CPACK_DOCDIR docs)
-    set(OV_CPACK_LICENSESDIR licenses)
-    set(OV_CPACK_SAMPLESDIR samples)
+    set(OV_CPACK_INCLUDEDIR ${CMAKE_INSTALL_INCLUDEDIR})
+    set(OV_CPACK_OPENVINO_CMAKEDIR ${CMAKE_INSTALL_LIBDIR}/cmake/openvino${OpenVINO_VERSION})
+    set(OV_CPACK_DOCDIR ${CMAKE_INSTALL_DATADIR}/doc/openvino-${OpenVINO_VERSION})
+    set(OV_CPACK_LICENSESDIR ${OV_CPACK_DOCDIR}/licenses)
+    set(OV_CPACK_SAMPLESDIR ${OV_CPACK_DOCDIR}/samples)
     set(OV_CPACK_WHEELSDIR wheels)
     set(OV_CPACK_DEVREQDIR tools)
-    set(OV_CPACK_PYTHONDIR python)
-
-    if(USE_BUILD_TYPE_SUBFOLDER)
-        set(build_type ${CMAKE_BUILD_TYPE})
-    else()
-        set(build_type $<CONFIG>)
+    ov_get_pyversion(pyversion)
+    if(pyversion)
+        set(OV_CPACK_PYTHONDIR ${CMAKE_INSTALL_LIBDIR}/${pyversion}/site-packages)
     endif()
 
     if(WIN32)
@@ -49,11 +33,11 @@
         set(OV_CPACK_RUNTIMEDIR runtime/lib/${ARCH_FOLDER}/${build_type})
         set(OV_CPACK_ARCHIVEDIR runtime/lib/${ARCH_FOLDER}/${build_type})
     else()
-        set(OV_CPACK_LIBRARYDIR runtime/lib/${ARCH_FOLDER})
-        set(OV_CPACK_RUNTIMEDIR runtime/lib/${ARCH_FOLDER})
-        set(OV_CPACK_ARCHIVEDIR runtime/lib/${ARCH_FOLDER})
+        set(OV_CPACK_LIBRARYDIR ${CMAKE_INSTALL_LIBDIR})
+        set(OV_CPACK_RUNTIMEDIR ${CMAKE_INSTALL_LIBDIR})
+        set(OV_CPACK_ARCHIVEDIR ${CAMKE_INSTALL_LIBDIR})
     endif()
-    set(OV_CPACK_PLUGINSDIR ${OV_CPACK_RUNTIMEDIR})
+    set(OV_CPACK_PLUGINSDIR ${CMAKE_INSTALL_LIBDIR}/openvino-${OpenVINO_VERSION})
 endmacro()
 
 ov_archive_cpack_set_dirs()
@@ -91,9 +75,6 @@
     unset(OV_CPACK_COMP_OPENVINO_REQ_FILES_EXCLUDE_ALL)
     # nodejs
     set(OV_CPACK_COMP_NPM_EXCLUDE_ALL EXCLUDE_FROM_ALL)
-    # scripts
-    unset(OV_CPACK_COMP_INSTALL_DEPENDENCIES_EXCLUDE_ALL)
-    unset(OV_CPACK_COMP_SETUPVARS_EXCLUDE_ALL)
     # pkgconfig
     set(OV_CPACK_COMP_PKG_CONFIG_EXCLUDE_ALL ${OV_CPACK_COMP_CORE_DEV_EXCLUDE_ALL})
     # symbolic links
diff -Nur a/cmake/developer_package/packaging/packaging.cmake b/cmake/developer_package/packaging/packaging.cmake
--- a/cmake/developer_package/packaging/packaging.cmake	2024-11-08 22:14:37.000000000 -0800
+++ b/cmake/developer_package/packaging/packaging.cmake	2024-12-12 02:31:19.442892214 -0800
@@ -173,9 +173,6 @@
     set(OV_CPACK_COMP_OPENVINO_REQ_FILES "openvino_req_files")
     # nodejs
     set(OV_CPACK_COMP_NPM "ov_node_addon")
-    # scripts
-    set(OV_CPACK_COMP_INSTALL_DEPENDENCIES "install_dependencies")
-    set(OV_CPACK_COMP_SETUPVARS "setupvars")
     # symbolic_links
     set(OV_CPACK_COMP_LINKS "core_dev_links")
     # pkgconfig
diff -Nur a/scripts/CMakeLists.txt b/scripts/CMakeLists.txt
--- a/scripts/CMakeLists.txt	2024-11-08 22:14:37.000000000 -0800
+++ b/scripts/CMakeLists.txt	2024-12-12 03:07:46.886302602 -0800
@@ -39,18 +39,3 @@
     list(APPEND _setupvars_files "${CMAKE_CURRENT_SOURCE_DIR}/${_setupvars_file_pwsh}")
 endif()
 
-install(PROGRAMS ${_setupvars_files}
-        DESTINATION .
-        COMPONENT ${OV_CPACK_COMP_SETUPVARS}
-        ${OV_CPACK_COMP_SETUPVARS_EXCLUDE_ALL})
-
-# install install_dependencies
-
-if(LINUX)
-    ov_cpack_add_component(${OV_CPACK_COMP_INSTALL_DEPENDENCIES} HIDDEN)
-
-    install(PROGRAMS "${CMAKE_CURRENT_SOURCE_DIR}/install_dependencies/install_openvino_dependencies.sh"
-            DESTINATION install_dependencies/
-            COMPONENT ${OV_CPACK_COMP_INSTALL_DEPENDENCIES}
-            ${OV_CPACK_COMP_INSTALL_DEPENDENCIES_EXCLUDE_ALL})
-endif()
