diff -Nur a/thirdparty/dependencies.cmake b/thirdparty/dependencies.cmake
--- a/thirdparty/dependencies.cmake	2024-11-08 22:14:37.000000000 -0800
+++ b/thirdparty/dependencies.cmake	2024-12-13 01:17:21.302457262 -0800
@@ -18,7 +18,7 @@
 # cmake older than 3.18 cannot create an alias for imported non-GLOBAL targets
 # so, we have to use 'IMPORTED_GLOBAL' property
 if(CMAKE_VERSION VERSION_LESS 3.18)
-    set(OV_PkgConfig_VISILITY GLOBAL)
+    set(OV_PkgConfig_VISIBILITY GLOBAL)
 endif()
 
 if(SUGGEST_OVERRIDE_SUPPORTED)
@@ -69,9 +69,16 @@
 #
 
 if(ENABLE_INTEL_NPU)
-    add_subdirectory(thirdparty/level_zero EXCLUDE_FROM_ALL)
-
-    add_library(LevelZero::LevelZero ALIAS ze_loader)
+    if(ENABLE_SYSTEM_LEVEL_ZERO)
+        pkg_search_module(level_zero QUIET
+                          IMPORTED_TARGET
+                          ${OV_PkgConfig_VISIBILITY}
+                          level-zero)
+        if(level_zero_FOUND)
+            add_library(LevelZero::LevelZero ALIAS PkgConfig::level_zero)
+            message(STATUS "${PKG_CONFIG_EXECUTABLE}: level_zero (${level_zero_VERSION}) is found at ${level_zero_PREFIX}")
+        endif()
+    endif()
 endif()
 
 #
@@ -190,7 +197,7 @@
         # Ubuntu 18.04 case when cmake interface is not available
         pkg_search_module(pugixml QUIET
                           IMPORTED_TARGET
-                          ${OV_PkgConfig_VISILITY}
+                          ${OV_PkgConfig_VISIBILITY}
                           pugixml)
         if(pugixml_FOUND)
             set(pugixml_target PkgConfig::pugixml)
@@ -249,7 +256,7 @@
             message(FATAL_ERROR "Debian | RPM package build requires shared Pugixml library")
         endif()
 
-        if(OV_PkgConfig_VISILITY)
+        if(OV_PkgConfig_VISIBILITY)
             # need to set GLOBAL visibility in order to create ALIAS for this target
             set_target_properties(${pugixml_target} PROPERTIES IMPORTED_GLOBAL ON)
         endif()
@@ -299,7 +306,7 @@
 
     if(GTest_FOUND)
         foreach(gtest_target gtest gtest_main gmock gmock_main)
-            if(OV_PkgConfig_VISILITY)
+            if(OV_PkgConfig_VISIBILITY)
                 # need to set GLOBAL visibility in order to create ALIAS for this target
                 set_target_properties(GTest::${gtest_target} PROPERTIES IMPORTED_GLOBAL ON)
             endif()
@@ -448,7 +455,7 @@
             set(ov_snappy_lib Snappy::snappy-static)
         endif()
 
-        if(OV_PkgConfig_VISILITY)
+        if(OV_PkgConfig_VISIBILITY)
             # need to set GLOBAL visibility in order to create ALIAS for this target
             set_target_properties(${ov_snappy_lib} PROPERTIES IMPORTED_GLOBAL ON)
         endif()
