diff -Nur a/src/bindings/python/src/pyopenvino/core/core.cpp b/src/bindings/python/src/pyopenvino/core/core.cpp
--- a/src/bindings/python/src/pyopenvino/core/core.cpp
+++ b/src/bindings/python/src/pyopenvino/core/core.cpp
@@ -4,6 +4,7 @@
 
 #include "pyopenvino/core/core.hpp"
 
+#include <pybind11/pybind11.h>
 #include <pybind11/stl.h>
 #include <pybind11/typing.h>
 
@@ -769,8 +770,7 @@
             )");
 
     cls.def_property_readonly("available_devices",
-                              &ov::Core::get_available_devices,
-                              py::call_guard<py::gil_scoped_release>(),
+                    py::cpp_function(&ov::Core::get_available_devices, py::call_guard<py::gil_scoped_release>()),
                               R"(
                                     Returns devices available for inference Core objects goes over all registered plugins.
 
diff -Nur a/src/bindings/python/src/pyopenvino/core/infer_request.cpp b/src/bindings/python/src/pyopenvino/core/infer_request.cpp
--- a/src/bindings/python/src/pyopenvino/core/infer_request.cpp
+++ b/src/bindings/python/src/pyopenvino/core/infer_request.cpp
@@ -819,10 +819,11 @@
 
     cls.def_property_readonly(
         "profiling_info",
+        py::cpp_function(
         [](InferRequestWrapper& self) {
             return self.m_request.get_profiling_info();
         },
-        py::call_guard<py::gil_scoped_release>(),
+        py::call_guard<py::gil_scoped_release>()),
         R"(
             Performance is measured per layer to get feedback on the most time-consuming operation.
             Not all plugins provide meaningful data!
