diff -Nur a/src/bindings/python/CMakeLists.txt b/src/bindings/python/CMakeLists.txt
--- a/src/bindings/python/CMakeLists.txt	2026-03-25 07:35:23.000000000 -0700
+++ b/src/bindings/python/CMakeLists.txt	2026-04-02 14:44:01.345533090 -0700
@@ -278,10 +278,8 @@
     list(APPEND openvino_py_files ${openvino_pyi_files})
 
     list(APPEND ov_setup_py_deps
-        ${openvino_py_files}
         "${OpenVINO_SOURCE_DIR}/pyproject.toml"
         "${OpenVINO_SOURCE_DIR}/setup.py"
-        "${OpenVINOPython_SOURCE_DIR}/requirements.txt"
         "${OpenVINOPython_SOURCE_DIR}/wheel/readme.txt"
         "${OpenVINO_SOURCE_DIR}/LICENSE"
         "${OpenVINO_SOURCE_DIR}/licensing/onednn_third-party-programs.txt"
diff -Nur a/src/bindings/python/src/pyopenvino/CMakeLists.txt b/src/bindings/python/src/pyopenvino/CMakeLists.txt
--- a/src/bindings/python/src/pyopenvino/CMakeLists.txt	2026-03-25 07:35:23.000000000 -0700
+++ b/src/bindings/python/src/pyopenvino/CMakeLists.txt	2026-04-02 14:32:24.187230375 -0700
@@ -100,7 +100,6 @@
 add_custom_command(
     OUTPUT ${PYTHON_COPY_STAMP_FILE}
     COMMAND ${CMAKE_COMMAND} -E copy_directory ${OpenVINOPython_SOURCE_DIR}/src/openvino ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}
-    COMMAND ${CMAKE_COMMAND} -E copy ${OpenVINOPython_SOURCE_DIR}/requirements.txt ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/../requirements.txt
     COMMAND ${CMAKE_COMMAND} -E touch ${PYTHON_COPY_STAMP_FILE}
     DEPENDS ${OPENVINO_PYTHON_SOURCE_FILES} ${OpenVINOPython_SOURCE_DIR}/requirements.txt
     COMMENT "Copying Python source files to build directory"
@@ -144,21 +143,6 @@
 
 ov_cpack_add_component(${OV_CPACK_COMP_OPENVINO_REQ_FILES} HIDDEN)
 
-install(FILES ${OpenVINOPython_SOURCE_DIR}/requirements.txt
-        DESTINATION ${OV_CPACK_PYTHONDIR}
-        COMPONENT ${OV_CPACK_COMP_OPENVINO_REQ_FILES}
-        ${OV_CPACK_COMP_OPENVINO_REQ_FILES_EXCLUDE_ALL})
-
-install(FILES ${OpenVINOPython_SOURCE_DIR}/src/openvino/preprocess/torchvision/requirements.txt
-        DESTINATION ${OV_CPACK_PYTHONDIR}/openvino/preprocess/torchvision
-        COMPONENT ${OV_CPACK_COMP_OPENVINO_REQ_FILES}
-        ${OV_CPACK_COMP_OPENVINO_REQ_FILES_EXCLUDE_ALL})
-
-install(DIRECTORY ${OpenVINOPython_SOURCE_DIR}/tests
-        DESTINATION tests/${PROJECT_NAME}
-        COMPONENT tests
-        EXCLUDE_FROM_ALL)
-
 if(TARGET ie_wheel)
     add_dependencies(ie_wheel ${PROJECT_NAME})
 endif()
