diff -Nur a/src/plugins/intel_cpu/CMakeLists.txt b/src/plugins/intel_cpu/CMakeLists.txt
--- a/src/plugins/intel_cpu/CMakeLists.txt	2026-02-18 07:33:17.000000000 -0800
+++ b/src/plugins/intel_cpu/CMakeLists.txt	2026-03-02 03:09:39.026932077 -0800
@@ -272,7 +272,7 @@
 target_include_directories(${TARGET_NAME} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/src)
 if (ENABLE_MLAS_FOR_CPU)
     target_link_libraries(${TARGET_NAME} PRIVATE mlas)
-    target_include_directories(${TARGET_NAME} SYSTEM PRIVATE $<TARGET_PROPERTY:mlas,INCLUDE_DIRECTORIES>)
+    #target_include_directories(${TARGET_NAME} SYSTEM PRIVATE $<TARGET_PROPERTY:mlas,INCLUDE_DIRECTORIES>)
     add_definitions(-DOV_CPU_WITH_MLAS)
 endif()
 if (ENABLE_SNIPPETS_LIBXSMM_TPP)
@@ -419,7 +419,7 @@
 
     target_include_directories(${TARGET_NAME}_obj SYSTEM PRIVATE $<TARGET_PROPERTY:openvino::runtime::dev,INTERFACE_INCLUDE_DIRECTORIES>)
     if(ENABLE_MLAS_FOR_CPU)
-        target_include_directories(${TARGET_NAME}_obj SYSTEM PUBLIC $<TARGET_PROPERTY:mlas,INCLUDE_DIRECTORIES>)
+        #target_include_directories(${TARGET_NAME}_obj SYSTEM PUBLIC $<TARGET_PROPERTY:mlas,INCLUDE_DIRECTORIES>)
     endif()
 
     if(ENABLE_SHL_FOR_CPU)
diff -Nur a/src/plugins/intel_cpu/thirdparty/CMakeLists.txt b/src/plugins/intel_cpu/thirdparty/CMakeLists.txt
--- a/src/plugins/intel_cpu/thirdparty/CMakeLists.txt	2026-02-18 07:33:17.000000000 -0800
+++ b/src/plugins/intel_cpu/thirdparty/CMakeLists.txt	2026-03-02 03:09:59.363599458 -0800
@@ -182,10 +182,10 @@
     ov_install_static_lib(libxsmm ${OV_CPACK_COMP_CORE})
 endif()
 
-if(ENABLE_MLAS_FOR_CPU)
-    add_subdirectory(mlas EXCLUDE_FROM_ALL)
-    ov_install_static_lib(mlas ${OV_CPACK_COMP_CORE})
-endif()
+#if(ENABLE_MLAS_FOR_CPU)
+#    add_subdirectory(mlas EXCLUDE_FROM_ALL)
+#    ov_install_static_lib(mlas ${OV_CPACK_COMP_CORE})
+#endif()
 
 if(ENABLE_SHL_FOR_CPU)
     function(ov_build_shl_static)
