diff -Nur a/thirdparty/dependencies.cmake b/thirdparty/dependencies.cmake
--- a/thirdparty/dependencies.cmake	2024-11-08 22:14:37.000000000 -0800
+++ b/thirdparty/dependencies.cmake	2024-12-14 13:55:32.910456023 -0800
@@ -284,10 +291,10 @@
 # Gflags
 #
 
-if(ENABLE_SAMPLES OR ENABLE_TESTS OR ENABLE_INTEL_NPU_INTERNAL)
-    add_subdirectory(thirdparty/gflags EXCLUDE_FROM_ALL)
-    ov_developer_package_export_targets(TARGET gflags)
-endif()
+#if(ENABLE_SAMPLES OR ENABLE_TESTS OR ENABLE_INTEL_NPU_INTERNAL)
+#    add_subdirectory(thirdparty/gflags EXCLUDE_FROM_ALL)
+#    ov_developer_package_export_targets(TARGET gflags)
+#endif()
 
 #
 # Google Tests framework
