diff -Naur gapcmon-0.8.3/gapcmon.desktop gapcmon-0.8.3-gentoo/gapcmon.desktop
--- gapcmon-0.8.3/gapcmon.desktop	2006-05-24 13:46:22.000000000 -0400
+++ gapcmon-0.8.3-gentoo/gapcmon.desktop	2006-12-15 22:55:16.916027448 -0500
@@ -4,8 +4,8 @@
 Version=1.0
 Exec=gapcmon
 Icon=apcupsd.png
-Name=APCUPSD Monitor
+Name=APC UPS Monitor
 GenericName=gapcmon
-Comment=Gtk/GUI Monitor program for (U)ninterruptible (P)ower (S)upply under the management of the APCUPSD.sourceforge.net package, also available at gapcmon.sourceforge.net
+Comment=Status monitor for APC Uninterruptible Power Supplies
 Terminal=false
 Categories=GTK;Application;System;
diff -Naur gapcmon-0.8.3/src/gapcmon.c gapcmon-0.8.3-gentoo/src/gapcmon.c
--- gapcmon-0.8.3/src/gapcmon.c	2006-07-03 17:39:13.000000000 -0400
+++ gapcmon-0.8.3-gentoo/src/gapcmon.c	2006-12-15 21:46:01.911256000 -0500
@@ -4040,7 +4040,7 @@
    gtk_widget_show(frame);
 
    label = gtk_label_new("<span foreground=\"blue\">"
-      "<i>double-click a columns value to change it.</i>" "</span>");
+      "<i>double-click a column's value to change it.</i>" "</span>");
    gtk_label_set_use_markup(GTK_LABEL(label), TRUE);
    gtk_label_set_justify(GTK_LABEL(label), GTK_JUSTIFY_CENTER);
 
@@ -4150,7 +4150,7 @@
       g_strdup_printf("<b><big>%s</big>\nVersion %s</b>\n", pch_pname, pch_pversion);
 
    about_msg =
-      g_strdup_printf("<b>gui monitor for UPSs under the management"
+      g_strdup_printf("<b>GUI monitor for UPSs under the management"
       " of the APCUPSD.sourceforge.net package</b>\n"
       "<i>http://gapcmon.sourceforge.net/</i>\n\n"
       "Copyright \xC2\xA9 2006 James Scott, Jr.\n"
@@ -5250,7 +5250,7 @@
      * These must be set before the first drawing_area configure event 
      */
     lg_graph_set_chart_title  (plg, "Waiting for Update");
-    lg_graph_set_y_label_text (plg, "Precentage of 100% normal");
+    lg_graph_set_y_label_text (plg, "Percentage of normal");
     lg_graph_set_x_label_text (plg, "Waiting for Update");
 
     g_snprintf (plg->ch_tooltip_text, sizeof (plg->ch_tooltip_text), "%s",
@@ -5378,9 +5378,9 @@
    gtk_box_pack_end(GTK_BOX(pbox), rbox, TRUE, TRUE, 0);
    gtk_widget_show(rbox);
 
-   label = gtk_label_new("Selftest running\n" "Number of transfers\n"
-      "Reason last transfer\n" "Last transfer to battery\n"
-      "Last transfer off battery\n" "Time on battery\n" "Cummulative on battery");
+   label = gtk_label_new("Self-test running\n" "Number of transfers\n"
+      "Reason for last transfer\n" "Last transfer to battery\n"
+      "Last transfer off battery\n" "Time on battery\n" "Cumulative on battery");
    gtk_label_set_use_markup(GTK_LABEL(label), TRUE);
    gtk_label_set_justify(GTK_LABEL(label), GTK_JUSTIFY_RIGHT);
    gtk_label_set_line_wrap(GTK_LABEL(label), TRUE);
