--- oggconvert-0.3.3.ebuild	2011-04-17 17:37:52.903269293 +0200
+++ oggconvert-0.3.3-r1.ebuild	2011-04-17 18:19:11.170270176 +0200
@@ -2,10 +2,12 @@
 # Distributed under the terms of the GNU General Public License v2
 # $Header: $
 
-EAPI="2"
+EAPI="3"
 PYTHON_DEPEND="2"
+SUPPORT_PYTHON_ABIS="1"
+RESTRICT_PYTHON_ABIS="3.*"
 
-inherit distutils
+inherit distutils eutils
 
 DESCRIPTION="Utility to convert media into the Vorbis audio and the Theora/Dirac/WebM video formats."
 HOMEPAGE="http://oggconvert.tristanb.net/"
@@ -16,6 +18,11 @@
 KEYWORDS="~amd64"
 IUSE="dirac vpx theora"
 
+src_prepare() {
+	epatch "${FILESDIR}/${P}-webm-fix.patch"
+	distutils_src_prepare
+}
+
 RDEPEND=">=media-libs/gst-plugins-base-0.10.11
 	media-plugins/gst-plugins-vorbis
 	dirac? ( >=media-plugins/gst-plugins-schroedinger-0.10.14 )
