diff -ur oggconvert-0.3.3/OggConvert/ocv_constants.py oggconvert-0.3.3.new/OggConvert/ocv_constants.py
--- oggconvert-0.3.3/OggConvert/ocv_constants.py	2010-05-23 18:40:53.000000000 +0200
+++ oggconvert-0.3.3.new/OggConvert/ocv_constants.py	2011-04-17 18:05:28.069269882 +0200
@@ -117,3 +117,4 @@
 
 WEBM_OPTS = { "writing-app" : "OggConvert " + version,
               "doctype" : "webm" }
+WEBMMUX_OPTS = {}
\ Geen regeleindeteken (LF) aan einde van bestand
diff -ur oggconvert-0.3.3/OggConvert/ocv_transcoder.py oggconvert-0.3.3.new/OggConvert/ocv_transcoder.py
--- oggconvert-0.3.3/OggConvert/ocv_transcoder.py	2010-05-23 17:09:40.000000000 +0200
+++ oggconvert-0.3.3.new/OggConvert/ocv_transcoder.py	2011-04-17 18:06:10.841269898 +0200
@@ -64,6 +64,10 @@
             self._mux = gst.element_factory_make("oggmux")
             for key in ocv_constants.OGGMUX_OPTS:
                 self._mux.set_property(key, ocv_constants.OGGMUX_OPTS[key])
+        elif fformat == "WEBM":
+            self._mux = gst.element_factory_make("webmmux")
+            for key in ocv_constants.WEBMMUX_OPTS:
+                self._mux.set_property(key, ocv_constants.WEBMMUX_OPTS[key])
         else:
             self._mux = gst.element_factory_make("matroskamux")
             for key in ocv_constants.MATROSKAMUX_OPTS:
