commit 6632ca79d59e02217c31a6e5e88569cbf475a960
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Tue May 10 05:25:21 2011 -0400

    let ICU_CONFIG be specified by the environment

diff --git a/configure b/configure
index a0080e7..fa2da38 100755
--- a/configure
+++ b/configure
@@ -14477,7 +14477,9 @@ WITH_ICU=0
 if test "$with_icu" != "yes" ; then
     echo Disabling ICU support
 else
-    ICU_CONFIG=icu-config
+    if [ "$ICU_CONFIG" = "" ]; then
+        ICU_CONFIG=icu-config
+    fi
     if ${ICU_CONFIG} --cflags >/dev/null 2>&1
     then
         ICU_LIBS=`icu-config --ldflags`
diff --git a/configure.in b/configure.in
index a1d2c89..4c2d5cd 100644
--- a/configure.in
+++ b/configure.in
@@ -1327,7 +1327,9 @@ WITH_ICU=0
 if test "$with_icu" != "yes" ; then
     echo Disabling ICU support
 else
-    ICU_CONFIG=icu-config
+    if [ "$ICU_CONFIG" = "" ]; then
+        ICU_CONFIG=icu-config
+    fi
     if ${ICU_CONFIG} --cflags >/dev/null 2>&1
     then
         ICU_LIBS=`icu-config --ldflags`
