commit 5e7826d06ed3870ee72ea0db7baf95bea32aa871
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Mon May 9 06:20:13 2011 -0400

    remove test targets

diff --git a/Makefile.am b/Makefile.am
index 300154b..9d8bcef 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -8,10 +8,7 @@ DIST_SUBDIRS = include . doc example python xstc
 
 INCLUDES = -I$(top_builddir)/include -I@srcdir@/include @THREAD_CFLAGS@ @Z_CFLAGS@ 
 
-noinst_PROGRAMS=testSchemas testRelax testSAX testHTML testXPath testURI \
-                testThreads testC14N testAutomata testRegexp \
-                testReader testapi testModule runtest runsuite testchar \
-		testdict runxmlconf testrecurse
+noinst_PROGRAMS=
 
 bin_PROGRAMS = xmllint xmlcatalog
 
diff --git a/Makefile.in b/Makefile.in
index df1af58..d48e88d 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -37,13 +37,7 @@ PRE_UNINSTALL = :
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
-noinst_PROGRAMS = testSchemas$(EXEEXT) testRelax$(EXEEXT) \
-	testSAX$(EXEEXT) testHTML$(EXEEXT) testXPath$(EXEEXT) \
-	testURI$(EXEEXT) testThreads$(EXEEXT) testC14N$(EXEEXT) \
-	testAutomata$(EXEEXT) testRegexp$(EXEEXT) testReader$(EXEEXT) \
-	testapi$(EXEEXT) testModule$(EXEEXT) runtest$(EXEEXT) \
-	runsuite$(EXEEXT) testchar$(EXEEXT) testdict$(EXEEXT) \
-	runxmlconf$(EXEEXT) testrecurse$(EXEEXT)
+noinst_PROGRAMS = 
 bin_PROGRAMS = xmllint$(EXEEXT) xmlcatalog$(EXEEXT)
 subdir = .
 DIST_COMMON = README $(am__configure_deps) $(srcdir)/Makefile.am \
