commit f86c9aec9c4a96ed5673280891e42e7d8b4304cd
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Mon May 9 20:30:25 2011 -0400

    use getDynamicClassID instead of typeid

diff --git a/common/ustrenum.cpp b/common/ustrenum.cpp
index 8ae7d7e..db43005 100644
--- a/common/ustrenum.cpp
+++ b/common/ustrenum.cpp
@@ -115,7 +115,7 @@ StringEnumeration::setChars(const char *s, int32_t length, UErrorCode &status) {
 }
 UBool 
 StringEnumeration::operator==(const StringEnumeration& that)const {
-    return typeid(*this) == typeid(that); 
+    return getDynamicClassID() == that.getDynamicClassID();
 }
 
 UBool
