commit ae6b9ea079ec74242ebb6b67a0fd6afd0e25080a
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Mon May 9 23:34:37 2011 -0400

    change typeid to getDynamicClassID()

diff --git a/common/uchriter.cpp b/common/uchriter.cpp
index f287881..ce296ce 100644
--- a/common/uchriter.cpp
+++ b/common/uchriter.cpp
@@ -68,7 +68,7 @@ UCharCharacterIterator::operator==(const ForwardCharacterIterator& that) const {
     if (this == &that) {
         return TRUE;
     }
-    if (typeid(*this) != typeid(that)) {
+    if ( getDynamicClassID() != that.getDynamicClassID() ) {
         return FALSE;
     }
 
