commit 9c42e809db35b072ad3956b833c263a0217dffe1
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Mon May 9 23:36:59 2011 -0400

    remove dynamic cast from uniset_props.cpp

diff --git a/common/uniset_props.cpp b/common/uniset_props.cpp
index 6f82dfb..e143800 100644
--- a/common/uniset_props.cpp
+++ b/common/uniset_props.cpp
@@ -616,7 +616,7 @@ void UnicodeSet::applyPattern(RuleCharacterIterator& chars,
             } else if (symbols != 0) {
                 const UnicodeFunctor *m = symbols->lookupMatcher(c);
                 if (m != 0) {
-                    const UnicodeSet *ms = dynamic_cast<const UnicodeSet *>(m);
+                    const UnicodeSet *ms = (const UnicodeSet *)(m);
                     if (ms == NULL) {
                         ec = U_MALFORMED_SET;
                         return;
