commit d98c86cfa20c1f3cf1b232c48bacbeaae73cd699
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Mon May 9 23:39:12 2011 -0400

    use getDynamicClassID() instead of typeid in rbbi.cpp

diff --git a/common/rbbi.cpp b/common/rbbi.cpp
index 2615a4b..2ef2622 100644
--- a/common/rbbi.cpp
+++ b/common/rbbi.cpp
@@ -292,7 +292,7 @@ RuleBasedBreakIterator::clone(void) const {
  */
 UBool
 RuleBasedBreakIterator::operator==(const BreakIterator& that) const {
-    if (typeid(*this) != typeid(that)) {
+    if ( getDynamicClassID() != that.getDynamicClassID() ) {
         return FALSE;
     }
 
