commit 8c5266a0dfcc15a980ccd77f3f9ac4aff254f5ea
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Mon May 9 23:49:53 2011 -0400

    remove dynamic_cast

diff --git a/i18n/numfmt.cpp b/i18n/numfmt.cpp
index dcd24a7..8a69d9c 100644
--- a/i18n/numfmt.cpp
+++ b/i18n/numfmt.cpp
@@ -387,7 +387,7 @@ ArgExtractor::ArgExtractor(const NumberFormat& nf, const Formattable& obj, UErro
 
     const UObject* o = obj.getObject(); // most commonly o==NULL
     const CurrencyAmount* amt;
-    if (o != NULL && (amt = dynamic_cast<const CurrencyAmount*>(o)) != NULL) {
+    if (o != NULL && (amt = (const CurrencyAmount*)(o)) != NULL) {
         // getISOCurrency() returns a pointer to internal storage, so we
         // copy it to retain it across the call to setCurrency().
         const UChar* curr = amt->getISOCurrency();
