commit 14d489fe89baac38e187ac01c7e42eabde932ebb
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Mon May 9 23:54:54 2011 -0400

    remove dynamic_cast

diff --git a/i18n/reldtfmt.cpp b/i18n/reldtfmt.cpp
index c135c1d..c931f42 100644
--- a/i18n/reldtfmt.cpp
+++ b/i18n/reldtfmt.cpp
@@ -288,7 +288,7 @@ RelativeDateFormat::toPatternDate(UnicodeString& result, UErrorCode& status) con
     if (!U_FAILURE(status)) {
         result.remove();
         if ( fDateFormat ) {
-            SimpleDateFormat* sdtfmt = dynamic_cast<SimpleDateFormat*>(fDateFormat);
+            SimpleDateFormat* sdtfmt = (SimpleDateFormat*)(fDateFormat);
             if (sdtfmt != NULL) {
                 sdtfmt->toPattern(result);
             } else {
@@ -305,7 +305,7 @@ RelativeDateFormat::toPatternTime(UnicodeString& result, UErrorCode& status) con
     if (!U_FAILURE(status)) {
         result.remove();
         if ( fTimeFormat ) {
-            SimpleDateFormat* sdtfmt = dynamic_cast<SimpleDateFormat*>(fTimeFormat);
+            SimpleDateFormat* sdtfmt = (SimpleDateFormat*)(fTimeFormat);
             if (sdtfmt != NULL) {
                 sdtfmt->toPattern(result);
             } else {
@@ -322,11 +322,11 @@ RelativeDateFormat::applyPatterns(const UnicodeString& datePattern, const Unicod
     if (!U_FAILURE(status)) {
         SimpleDateFormat* sdtfmt = NULL;
         SimpleDateFormat* stmfmt = NULL;
-        if (fDateFormat && (sdtfmt = dynamic_cast<SimpleDateFormat*>(fDateFormat)) == NULL) {
+        if (fDateFormat && (sdtfmt = (SimpleDateFormat*)(fDateFormat)) == NULL) {
             status = U_UNSUPPORTED_ERROR;
             return;
         }
-        if (fTimeFormat && (stmfmt = dynamic_cast<SimpleDateFormat*>(fTimeFormat)) == NULL) {
+        if (fTimeFormat && (stmfmt = (SimpleDateFormat*)(fTimeFormat)) == NULL) {
             status = U_UNSUPPORTED_ERROR;
             return;
         }
