commit 43ab5219f1ff47ef04d089827bd50f8270e0ae91
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Tue May 10 00:39:50 2011 -0400

    initial commit

diff --git a/i18n/udat.cpp b/i18n/udat.cpp
index 758a6a7..85771cb 100644
--- a/i18n/udat.cpp
+++ b/i18n/udat.cpp
@@ -33,7 +33,7 @@ U_NAMESPACE_USE
  */
 static void verifyIsSimpleDateFormat(const UDateFormat* fmt, UErrorCode *status) {
    if(U_SUCCESS(*status) &&
-       dynamic_cast<const SimpleDateFormat*>(reinterpret_cast<const DateFormat*>(fmt))==NULL) {
+       (const SimpleDateFormat*>(reinterpret_cast<const DateFormat*)(fmt))==NULL) {
        *status = U_ILLEGAL_ARGUMENT_ERROR;
    }
 }
@@ -338,14 +338,14 @@ udat_toPattern(    const   UDateFormat     *fmt,
     }
 
     const DateFormat *df=reinterpret_cast<const DateFormat *>(fmt);
-    const SimpleDateFormat *sdtfmt=dynamic_cast<const SimpleDateFormat *>(df);
+    const SimpleDateFormat *sdtfmt=(const SimpleDateFormat *)(df);
     const RelativeDateFormat *reldtfmt;
     if (sdtfmt!=NULL) {
         if(localized)
             sdtfmt->toLocalizedPattern(res, *status);
         else
             sdtfmt->toPattern(res);
-    } else if (!localized && (reldtfmt=dynamic_cast<const RelativeDateFormat *>(df))!=NULL) {
+    } else if (!localized && (reldtfmt=(const RelativeDateFormat *)(df))!=NULL) {
         reldtfmt->toPattern(res, *status);
     } else {
         *status = U_ILLEGAL_ARGUMENT_ERROR;
@@ -900,7 +900,7 @@ udat_getLocaleByType(const UDateFormat *fmt,
  */
 static void verifyIsRelativeDateFormat(const UDateFormat* fmt, UErrorCode *status) {
    if(U_SUCCESS(*status) &&
-       dynamic_cast<const RelativeDateFormat*>(reinterpret_cast<const DateFormat*>(fmt))==NULL) {
+       (const RelativeDateFormat*>(reinterpret_cast<const DateFormat*)(fmt))==NULL) {
        *status = U_ILLEGAL_ARGUMENT_ERROR;
    }
 }
