commit 1de4a63301af59669d60385bf380ee3641c1d6ca
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Mon May 9 23:58:23 2011 -0400

    remove dynamic_cast

diff --git a/i18n/udat.cpp b/i18n/udat.cpp
index 85771cb..d46cbc6 100644
--- a/i18n/udat.cpp
+++ b/i18n/udat.cpp
@@ -33,7 +33,7 @@ U_NAMESPACE_USE
  */
 static void verifyIsSimpleDateFormat(const UDateFormat* fmt, UErrorCode *status) {
    if(U_SUCCESS(*status) &&
-       (const SimpleDateFormat*>(reinterpret_cast<const DateFormat*)(fmt))==NULL) {
+       (const SimpleDateFormat*)(reinterpret_cast<const DateFormat*>(fmt))==NULL) {
        *status = U_ILLEGAL_ARGUMENT_ERROR;
    }
 }
@@ -900,7 +900,7 @@ udat_getLocaleByType(const UDateFormat *fmt,
  */
 static void verifyIsRelativeDateFormat(const UDateFormat* fmt, UErrorCode *status) {
    if(U_SUCCESS(*status) &&
-       (const RelativeDateFormat*>(reinterpret_cast<const DateFormat*)(fmt))==NULL) {
+       (const RelativeDateFormat*)(reinterpret_cast<const DateFormat*>(fmt))==NULL) {
        *status = U_ILLEGAL_ARGUMENT_ERROR;
    }
 }
