commit f992253bc90fe028c342140a379693708088b860
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Tue May 10 01:14:49 2011 -0400

    typeid -> getDynamicClassID

diff --git a/i18n/rbnf.cpp b/i18n/rbnf.cpp
index 93d013e..1a36898 100644
--- a/i18n/rbnf.cpp
+++ b/i18n/rbnf.cpp
@@ -874,7 +874,7 @@ RuleBasedNumberFormat::operator==(const Format& other) const
         return TRUE;
     }
 
-    if (typeid(*this) == typeid(other)) {
+    if ( getDynamicClassID() == other.getDynamicClassID() ) {
         const RuleBasedNumberFormat& rhs = (const RuleBasedNumberFormat&)other;
         if (locale == rhs.locale &&
             lenient == rhs.lenient &&
