commit 1f0e4c0f5d3fc678af447474894c26e8da61880c
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Tue May 10 01:40:59 2011 -0400

    removed typeid

diff --git a/i18n/ucal.cpp b/i18n/ucal.cpp
index 23bac27..9a1b1fb 100644
--- a/i18n/ucal.cpp
+++ b/i18n/ucal.cpp
@@ -251,7 +251,7 @@ ucal_setGregorianChange(UCalendar *cal, UDate date, UErrorCode *pErrorCode) {
     // Not if(gregocal == NULL) {
     // because we really want to work only with a GregorianCalendar, not with
     // its subclasses like BuddhistCalendar.
-    if(typeid(*cpp_cal) != typeid(GregorianCalendar)) {
+    if(cpp_cal == NULL || cpp_cal->getDynamicClassID() != GregorianCalendar::getStaticClassID()) {
         *pErrorCode = U_UNSUPPORTED_ERROR;
         return;
     }
@@ -267,7 +267,7 @@ ucal_getGregorianChange(const UCalendar *cal, UErrorCode *pErrorCode) {
     const GregorianCalendar *gregocal = (const GregorianCalendar *)(cpp_cal);
     // Not if(gregocal == NULL) {
     // see comments in ucal_setGregorianChange().
-    if(typeid(*cpp_cal) != typeid(GregorianCalendar)) {
+    if(cpp_cal == NULL || cpp_cal->getDynamicClassID() != GregorianCalendar::getStaticClassID()) {
         *pErrorCode = U_UNSUPPORTED_ERROR;
         return (UDate)0;
     }
