commit 49f9f91d43e615aef2281b4da3c992ecb6b725ec
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Tue May 10 02:23:13 2011 -0400

    typeid

diff --git a/i18n/calendar.cpp b/i18n/calendar.cpp
index eea4c20..577358f 100644
--- a/i18n/calendar.cpp
+++ b/i18n/calendar.cpp
@@ -909,7 +909,7 @@ Calendar::operator==(const Calendar& that) const
 UBool 
 Calendar::isEquivalentTo(const Calendar& other) const
 {
-    return typeid(*this) == typeid(other) &&
+    return getDynamicClassID() == other.getDynamicClassID() &&
         fLenient                == other.fLenient &&
         fFirstDayOfWeek         == other.fFirstDayOfWeek &&
         fMinimalDaysInFirstWeek == other.fMinimalDaysInFirstWeek &&
