commit d8045060b80c9be1b70d97db9bfcdffbc13ae0b8
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Tue May 10 02:24:56 2011 -0400

    typeid

diff --git a/i18n/timezone.cpp b/i18n/timezone.cpp
index 777bee7..bca2c27 100644
--- a/i18n/timezone.cpp
+++ b/i18n/timezone.cpp
@@ -384,7 +384,7 @@ TimeZone::operator=(const TimeZone &right)
 UBool
 TimeZone::operator==(const TimeZone& that) const
 {
-    return typeid(*this) == typeid(that) &&
+    return getDynamicClassID() == that.getDynamicClassID() &&
         fID == that.fID;
 }
 
