commit bb0b724d9078d3b25cabd1c5bda4dd51cc9ba008
Author: Christopher Friedt <chrisfriedt@gmail.com>
Date:   Tue May 10 02:27:17 2011 -0400

    typeid

diff --git a/i18n/simpletz.cpp b/i18n/simpletz.cpp
index fb960bb..9de1410 100644
--- a/i18n/simpletz.cpp
+++ b/i18n/simpletz.cpp
@@ -231,7 +231,7 @@ UBool
 SimpleTimeZone::operator==(const TimeZone& that) const
 {
     return ((this == &that) ||
-            (typeid(*this) == typeid(that) &&
+            (getDynamicClassID() == that.getDynamicClassID() &&
             TimeZone::operator==(that) &&
             hasSameRules(that)));
 }
@@ -743,7 +743,7 @@ UBool
 SimpleTimeZone::hasSameRules(const TimeZone& other) const
 {
     if (this == &other) return TRUE;
-    if (typeid(*this) != typeid(other)) return FALSE;
+    if (getDynamicClassID() != other.getDynamicClassID()) return FALSE;
     SimpleTimeZone *that = (SimpleTimeZone*)&other;
     return rawOffset     == that->rawOffset &&
         useDaylight     == that->useDaylight &&
