diff --git a/common/rbbi.cpp b/common/rbbi.cpp
index 2ef2622..9c4b27d 100644
--- a/common/rbbi.cpp
+++ b/common/rbbi.cpp
@@ -10,7 +10,6 @@
 //                   class RuleBasedBreakIterator
 //
 
-#include <typeinfo>  // for 'typeid' to work
 
 #include "unicode/utypes.h"
 
diff --git a/common/schriter.cpp b/common/schriter.cpp
index ba9c188..3c10c85 100644
--- a/common/schriter.cpp
+++ b/common/schriter.cpp
@@ -13,7 +13,6 @@
 ******************************************************************************
 */
 
-#include <typeinfo>  // for 'typeid' to work
 
 #include "unicode/chariter.h"
 #include "unicode/schriter.h"
diff --git a/common/uchriter.cpp b/common/uchriter.cpp
index ce296ce..590361b 100644
--- a/common/uchriter.cpp
+++ b/common/uchriter.cpp
@@ -5,7 +5,6 @@
 ******************************************************************************
 */
 
-#include <typeinfo>  // for 'typeid' to work
 
 #include "unicode/uchriter.h"
 #include "unicode/ustring.h"
diff --git a/common/ustrenum.cpp b/common/ustrenum.cpp
index db43005..7da0cd7 100644
--- a/common/ustrenum.cpp
+++ b/common/ustrenum.cpp
@@ -8,7 +8,6 @@
 * Since: ICU 2.4
 **********************************************************************
 */
-#include <typeinfo>  // for 'typeid' to work 
 
 #include "unicode/ustring.h"
 #include "unicode/strenum.h"
diff --git a/i18n/calendar.cpp b/i18n/calendar.cpp
index 577358f..ad4b479 100644
--- a/i18n/calendar.cpp
+++ b/i18n/calendar.cpp
@@ -24,7 +24,6 @@
 *******************************************************************************
 */
 
-#include <typeinfo>  // for 'typeid' to work 
 
 #include "unicode/utypes.h"
 
diff --git a/i18n/currfmt.cpp b/i18n/currfmt.cpp
index b395d51..7f22051 100644
--- a/i18n/currfmt.cpp
+++ b/i18n/currfmt.cpp
@@ -8,7 +8,6 @@
 * Since: ICU 3.0
 **********************************************************************
 */
-#include <typeinfo>  // for 'typeid' to work
 
 #include "unicode/utypes.h"
 
diff --git a/i18n/currunit.cpp b/i18n/currunit.cpp
index cac3177..adda3b9 100644
--- a/i18n/currunit.cpp
+++ b/i18n/currunit.cpp
@@ -8,7 +8,6 @@
 * Since: ICU 3.0
 **********************************************************************
 */
-#include <typeinfo>  // for 'typeid' to work
 
 #include "unicode/utypes.h"
 
diff --git a/i18n/dtitvfmt.cpp b/i18n/dtitvfmt.cpp
index fc73184..560efc0 100644
--- a/i18n/dtitvfmt.cpp
+++ b/i18n/dtitvfmt.cpp
@@ -8,7 +8,6 @@
 *******************************************************************************
 */
 
-#include <typeinfo>  // for 'typeid' to work
 
 #include "unicode/dtitvfmt.h"
 
diff --git a/i18n/dtrule.cpp b/i18n/dtrule.cpp
index 6a28cd5..895c9f1 100644
--- a/i18n/dtrule.cpp
+++ b/i18n/dtrule.cpp
@@ -5,7 +5,6 @@
 *******************************************************************************
 */
 
-#include <typeinfo>  // for 'typeid' to work
 
 #include "unicode/utypes.h"
 
diff --git a/i18n/format.cpp b/i18n/format.cpp
index f26a062..0456590 100644
--- a/i18n/format.cpp
+++ b/i18n/format.cpp
@@ -19,7 +19,6 @@
 // This file was generated from the java source file Format.java
 // *****************************************************************************
 
-#include <typeinfo>  // for 'typeid' to work
 
 #include "unicode/utypes.h"
 
diff --git a/i18n/measure.cpp b/i18n/measure.cpp
index 5cb4ac5..6541576 100644
--- a/i18n/measure.cpp
+++ b/i18n/measure.cpp
@@ -8,7 +8,6 @@
 * Since: ICU 3.0
 **********************************************************************
 */
-#include <typeinfo>  // for 'typeid' to work
 
 #include "unicode/utypes.h"
 
diff --git a/i18n/nfsubs.cpp b/i18n/nfsubs.cpp
index 6490348..c1da0f8 100644
--- a/i18n/nfsubs.cpp
+++ b/i18n/nfsubs.cpp
@@ -14,7 +14,6 @@
 */
 
 #include <stdio.h>
-#include <typeinfo>  // for 'typeid' to work
 
 #include "nfsubs.h"
 #include "digitlst.h"
diff --git a/i18n/olsontz.cpp b/i18n/olsontz.cpp
index bec0a0d..3267a3e 100644
--- a/i18n/olsontz.cpp
+++ b/i18n/olsontz.cpp
@@ -9,7 +9,6 @@
 **********************************************************************
 */
 
-#include <typeinfo>  // for 'typeid' to work
 
 #include "olsontz.h"
 
diff --git a/i18n/rbnf.cpp b/i18n/rbnf.cpp
index 1a36898..9a8e3e4 100644
--- a/i18n/rbnf.cpp
+++ b/i18n/rbnf.cpp
@@ -5,7 +5,6 @@
 *******************************************************************************
 */
 
-#include <typeinfo>  // for 'typeid' to work
 
 #include "unicode/rbnf.h"
 
diff --git a/i18n/rbtz.cpp b/i18n/rbtz.cpp
index 7d66c50..7050ba6 100644
--- a/i18n/rbtz.cpp
+++ b/i18n/rbtz.cpp
@@ -5,7 +5,6 @@
 *******************************************************************************
 */
 
-#include <typeinfo>  // for 'typeid' to work
 
 #include "unicode/utypes.h"
 
diff --git a/i18n/selfmt.cpp b/i18n/selfmt.cpp
index 04974aa..a6a2311 100755
--- a/i18n/selfmt.cpp
+++ b/i18n/selfmt.cpp
@@ -14,7 +14,6 @@
  *   11/16/09    kirtig      Improved version
  ********************************************************************/
 
-#include <typeinfo>  // for 'typeid' to work
 
 #include "unicode/utypes.h"
 #include "unicode/ustring.h"
diff --git a/i18n/simpletz.cpp b/i18n/simpletz.cpp
index 9de1410..cc71d1d 100644
--- a/i18n/simpletz.cpp
+++ b/i18n/simpletz.cpp
@@ -21,7 +21,6 @@
  ********************************************************************************
  */
 
-#include <typeinfo>  // for 'typeid' to work
 
 #include "unicode/utypes.h"
 
diff --git a/i18n/tblcoll.cpp b/i18n/tblcoll.cpp
index a39c80f..79469f3 100644
--- a/i18n/tblcoll.cpp
+++ b/i18n/tblcoll.cpp
@@ -54,7 +54,6 @@
  * 01/29/01     synwee      Modified into a C++ wrapper calling C APIs (ucol.h)
  */
 
-#include <typeinfo>  // for 'typeid' to work
 
 #include "unicode/utypes.h"
 
diff --git a/i18n/timezone.cpp b/i18n/timezone.cpp
index bca2c27..ca6ea3a 100644
--- a/i18n/timezone.cpp
+++ b/i18n/timezone.cpp
@@ -35,7 +35,6 @@
 *                           available IDs code.  Misc. cleanup.
 *********************************************************************************/
 
-#include <typeinfo>  // for 'typeid' to work
 
 #include "unicode/utypes.h"
 #include "unicode/ustring.h"
diff --git a/i18n/tmunit.cpp b/i18n/tmunit.cpp
index 9198e8b..92d9c55 100644
--- a/i18n/tmunit.cpp
+++ b/i18n/tmunit.cpp
@@ -5,7 +5,6 @@
  *******************************************************************************
  */
 
-#include <typeinfo>  // for 'typeid' to work
 
 #include "unicode/tmunit.h"
 
diff --git a/i18n/tmutfmt.cpp b/i18n/tmutfmt.cpp
index b497aaf..6e3c25b 100644
--- a/i18n/tmutfmt.cpp
+++ b/i18n/tmutfmt.cpp
@@ -5,7 +5,6 @@
  *******************************************************************************
  */
 
-#include <typeinfo>  // for 'typeid' to work
 
 #include "unicode/tmutfmt.h"
 
diff --git a/i18n/translit.cpp b/i18n/translit.cpp
index 825ec35..a644adf 100644
--- a/i18n/translit.cpp
+++ b/i18n/translit.cpp
@@ -8,7 +8,6 @@
  **********************************************************************
  */
 
-#include <typeinfo>  // for 'typeid' to work
 
 #include "unicode/utypes.h"
 
diff --git a/i18n/tzrule.cpp b/i18n/tzrule.cpp
index d661144..c4adc81 100644
--- a/i18n/tzrule.cpp
+++ b/i18n/tzrule.cpp
@@ -5,7 +5,6 @@
 *******************************************************************************
 */
 
-#include <typeinfo>  // for 'typeid' to work
 
 #include "unicode/utypes.h"
 
diff --git a/i18n/tztrans.cpp b/i18n/tztrans.cpp
index 473b470..76e9f69 100644
--- a/i18n/tztrans.cpp
+++ b/i18n/tztrans.cpp
@@ -5,7 +5,6 @@
 *******************************************************************************
 */
 
-#include <typeinfo>  // for 'typeid' to work
 
 #include "unicode/utypes.h"
 
diff --git a/i18n/ucal.cpp b/i18n/ucal.cpp
index 9a1b1fb..264600a 100644
--- a/i18n/ucal.cpp
+++ b/i18n/ucal.cpp
@@ -5,7 +5,6 @@
 *******************************************************************************
 */
 
-#include <typeinfo>  // for 'typeid' to work
 
 #include "unicode/utypes.h"
 
diff --git a/i18n/vtzone.cpp b/i18n/vtzone.cpp
index 0129b82..cfe5168 100644
--- a/i18n/vtzone.cpp
+++ b/i18n/vtzone.cpp
@@ -5,7 +5,6 @@
 *******************************************************************************
 */
 
-#include <typeinfo>  // for 'typeid' to work
 
 #include "unicode/utypes.h"
 
diff --git a/test/intltest/citrtest.cpp b/test/intltest/citrtest.cpp
index d542b13..d1dbfe2 100644
--- a/test/intltest/citrtest.cpp
+++ b/test/intltest/citrtest.cpp
@@ -9,7 +9,6 @@
  ****************************************************************************************/
 
 #include <string.h>
-#include <typeinfo>  // for 'typeid' to work
 
 #include "unicode/chariter.h"
 #include "unicode/ustring.h"
diff --git a/test/intltest/icusvtst.cpp b/test/intltest/icusvtst.cpp
index 1eb6e14..017c729 100644
--- a/test/intltest/icusvtst.cpp
+++ b/test/intltest/icusvtst.cpp
@@ -5,7 +5,6 @@
  *******************************************************************************
  */
 
-#include <typeinfo>  // for 'typeid' to work
 
 #include "unicode/utypes.h"
 
diff --git a/test/intltest/rbbitst.cpp b/test/intltest/rbbitst.cpp
index 43f5bf5..3a529a5 100644
--- a/test/intltest/rbbitst.cpp
+++ b/test/intltest/rbbitst.cpp
@@ -9,7 +9,6 @@
 *   01/12/2000  Madhu        Updated for changed API and added new tests
 ************************************************************************/
 
-#include <typeinfo>  // for 'typeid' to work
 
 #include "unicode/utypes.h"
 
diff --git a/test/intltest/uobjtest.cpp b/test/intltest/uobjtest.cpp
index fd57cd8..1638f75 100644
--- a/test/intltest/uobjtest.cpp
+++ b/test/intltest/uobjtest.cpp
@@ -7,7 +7,6 @@
 
 #include <stdio.h>
 #include <string.h>
-#include <typeinfo>  // for 'typeid' to work
 
 #include "uobjtest.h"
 #include "cmemory.h" // UAlignedMemory
