diff --git a/i18n/digitlst.cpp b/i18n/digitlst.cpp
index 4a146f4..8ca096e 100644
--- a/i18n/digitlst.cpp
+++ b/i18n/digitlst.cpp
@@ -37,7 +37,6 @@
 #include <limits.h>
 #include <string.h>
 #include <stdio.h>
-#include <limits>
 
 // ***************************************************************************
 // class DigitList
@@ -413,11 +412,8 @@ DigitList::getDouble() const
             nonConstThis->fDouble /= -1;
         }
     } else if (isInfinite()) {
-        if (std::numeric_limits<double>::has_infinity) {
-            nonConstThis->fDouble = std::numeric_limits<double>::infinity();
-        } else {
-            nonConstThis->fDouble = std::numeric_limits<double>::max();
-        }
+	// std::numeric_limits<double>::infinity()
+        nonConstThis->fDouble = 0x7ff0000000000000;
         if (!isPositive()) {
             nonConstThis->fDouble = -fDouble;
         } 
