diff -ur aserve.orig/main.cl aserve/main.cl
--- aserve.orig/main.cl	2005-08-26 02:47:49.000000000 -0500
+++ aserve/main.cl	2005-09-05 14:21:05.000000000 -0500
@@ -215,7 +215,7 @@
 )
 
 #+(and clisp unix)
-(defun getpid () (unix:getpid))
+(defun getpid () (sys::process-id))
 
 #+ (and sbcl unix)
 (progn
@@ -1519,8 +1519,12 @@
 		  (if* error-obj
 		     then (brief-logmess 
 			   (format nil "While reading http request~:_ from ~a:~:_ ~a" 
+				   #+clisp
+				   (acl-compat.socket:ipaddr-to-dotted 
+				     (acl-compat.socket::remote-host sock))
+                                   #-clisp
 				   (socket:ipaddr-to-dotted 
-				    (socket::remote-host sock))
+				     (socket::remote-host sock))
 				   error-obj)))
 
 		  ; notify the client if it's still listening
@@ -2777,6 +2781,7 @@
   data	 ; list of buffers
   create ; create new object for the buffer
   init	 ; optional - used to init buffers taken off the free list
+  #-clisp
   (lock  (acl-compat.mp:make-process-lock))
   )
 
